<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.189">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Anna Elisabeth Furtjes">
<meta name="dcterms.date" content="2024-11-04">

<title>Brain Atrophy Genetics - Genome-wide association study</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Brain Atrophy Genetics</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html">Overview</a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#regenie-step-1" id="toc-regenie-step-1" class="nav-link active" data-scroll-target="#regenie-step-1">Regenie: Step 1</a></li>
  <li><a href="#regenie-step-2" id="toc-regenie-step-2" class="nav-link" data-scroll-target="#regenie-step-2">Regenie: Step 2</a></li>
  <li><a href="#regenie-interaction-models-step-2" id="toc-regenie-interaction-models-step-2" class="nav-link" data-scroll-target="#regenie-interaction-models-step-2">Regenie interaction models: step 2</a></li>
  <li><a href="#format-gwas-data" id="toc-format-gwas-data" class="nav-link" data-scroll-target="#format-gwas-data">Format GWAS data</a></li>
  <li><a href="#manhattan-plot" id="toc-manhattan-plot" class="nav-link" data-scroll-target="#manhattan-plot">Manhattan plot</a>
  <ul class="collapse">
  <li><a href="#alter-manhattan-function-for-better-looking-annotation" id="toc-alter-manhattan-function-for-better-looking-annotation" class="nav-link" data-scroll-target="#alter-manhattan-function-for-better-looking-annotation">Alter Manhattan function for better looking annotation</a></li>
  <li><a href="#plot-gwas-associations" id="toc-plot-gwas-associations" class="nav-link" data-scroll-target="#plot-gwas-associations">Plot GWAS associations</a></li>
  <li><a href="#miami-plots" id="toc-miami-plots" class="nav-link" data-scroll-target="#miami-plots">Miami plots</a></li>
  <li><a href="#examine-qq-plots" id="toc-examine-qq-plots" class="nav-link" data-scroll-target="#examine-qq-plots">Examine QQ plots</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Genome-wide association study</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Anna Elisabeth Furtjes </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">November 4, 2024</p>
    </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Doi</div>
    <div class="quarto-title-meta-contents">
      <p class="doi">
        <a href="https://doi.org/test">test</a>
      </p>
    </div>
  </div>
  </div>
  

</header>

<hr>
<p>GWAS analysis was performed with the <a href="https://rgcgithub.github.io/regenie/">REGENIE software</a>.</p>
<p>Ensure that the <code>resid_stand</code> variable was calculated in the subsample with non-missing data that is analysed in the GWAS analysis.</p>
<section id="regenie-step-1" class="level2">
<h2 class="anchored" data-anchor-id="regenie-step-1">Regenie: Step 1</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="ex">regenie_v3.4</span> <span class="dt">\</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>--step 1 <span class="dt">\</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>--bed <span class="va">$genotype</span>/ukb_neuroimaging_brainAtrophy_GWASinput <span class="dt">\</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>--phenoFile <span class="va">$phenotype</span>/UKB_CrossNeuroIDP_noOutliers.txt <span class="dt">\</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>--phenoColList <span class="va">${trait}</span>  <span class="dt">\</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>--covarFile <span class="va">$phenotype</span>/UKB_covarGWAS.txt <span class="dt">\</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>--catCovarList sex,assessmentMonth,site,array,batch <span class="dt">\</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>--bsize 100 <span class="dt">\</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>--threads 30 <span class="dt">\</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>--maxCatLevels 106 <span class="dt">\</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>--lowmem <span class="dt">\</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>--lowmem-prefix <span class="va">${wd}</span>/tmp_noOutliers <span class="dt">\</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>--iid-only <span class="dt">\</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>--out <span class="va">${wd}</span>/Step1_out_all_noOutliers</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="regenie-step-2" class="level2">
<h2 class="anchored" data-anchor-id="regenie-step-2">Regenie: Step 2</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="ex">regenie_v3.4</span> <span class="dt">\</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>--step 2 <span class="dt">\</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>--bgen <span class="va">$imputed</span>/ukb_chr<span class="va">${CHR}</span>_clean_v3.bgen <span class="dt">\</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>--sample <span class="va">$sampleFile</span>/ukb10279_imp_chr<span class="va">${CHR}</span>_v3_s487395.sample <span class="dt">\</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>--keep <span class="va">$genotype</span>/ukb_neuroimaging_brainAtrophy_GWASinput.fam <span class="dt">\</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>--covarFile <span class="va">$phenotype</span>/UKB_covarGWAS.txt <span class="dt">\</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>--catCovarList sex,assessmentMonth,site,array,batch <span class="dt">\</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>--maxCatLevels 106 <span class="dt">\</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>--phenoFile <span class="va">$phenotype</span>/UKB_CrossNeuroIDP_noOutliers.txt <span class="dt">\</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>--phenoColList <span class="va">${trait}</span> <span class="dt">\</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>--bsize 400 <span class="dt">\</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>--minINFO 0.4 <span class="dt">\</span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>--minMAC 5 <span class="dt">\</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>--threads 5 <span class="dt">\</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>--pred <span class="va">${step1}</span>/Step1_out_all_pred_noOutliers.list <span class="dt">\</span></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>--out <span class="va">$wd</span>/GWAS_<span class="va">${trait}</span>_chr<span class="va">${CHR}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="regenie-interaction-models-step-2" class="level2">
<h2 class="anchored" data-anchor-id="regenie-interaction-models-step-2">Regenie interaction models: step 2</h2>
<p>As unintuitive as it may seem, you must standardise the age variable before running this script. Regenie calculates a product-wise variable for the interaction between phenotype and age and if age is indicated in months or even days, the variable will be so large that regenie cannot handle it, and results will be massively inflated (results are complete nonsense with every SNP being extremely significant).</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="co"># note that this script only runs when linked to the bgen sample file </span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="va">trait</span><span class="op">=</span><span class="va">$1</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> CHR <span class="kw">in</span> <span class="dt">{</span><span class="dv">1</span><span class="dt">..</span><span class="dv">22</span><span class="dt">}</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="ex">regenie_v3.4</span> <span class="dt">\</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>--step 2 <span class="dt">\</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>--bgen <span class="va">$imputed</span>/ukb_chr<span class="va">${CHR}</span>_clean_v3.bgen <span class="dt">\</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>--ref-first <span class="dt">\</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>--sample <span class="va">$sampleFile</span>/ukb10279_imp_chr<span class="va">${CHR}</span>_v3_s487395.sample <span class="dt">\</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>--keep <span class="va">$genotype</span>/ukb_neuroimaging_brainAtrophy_GWASinput.fam <span class="dt">\</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>--covarFile <span class="va">$phenotype</span>/UKB_covarGWAS.txt <span class="dt">\</span></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>--catCovarList sex,assessmentMonth,site,array,batch <span class="dt">\</span></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>--maxCatLevels 106 <span class="dt">\</span></span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>--phenoFile <span class="va">$phenotype</span>/UKB_CrossNeuroIDP_noOutliers.txt <span class="dt">\</span></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>--phenoColList <span class="va">${trait}</span> <span class="dt">\</span></span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>--bsize 400 <span class="dt">\</span></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>--minINFO 0.4 <span class="dt">\</span></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>--minMAC 5 <span class="dt">\</span></span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>--threads 5 <span class="dt">\</span></span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>--gz <span class="dt">\</span></span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>--interaction age <span class="dt">\</span></span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>--pred <span class="va">${step1}</span>/Step1_out_all_noOutliers_pred.list <span class="dt">\</span></span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>--out <span class="va">$wd</span>/GWAS_<span class="va">${trait}</span>_chr<span class="va">${CHR}</span>_interaction</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="format-gwas-data" class="level2">
<h2 class="anchored" data-anchor-id="format-gwas-data">Format GWAS data</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in the files for each area and merge them </span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>all_traits<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">"resid_stand"</span>,<span class="st">"ratio_stand"</span>,<span class="st">"diff_stand"</span>,<span class="st">"TBVstand"</span>,<span class="st">"ICVstand"</span>,<span class="st">"CSFstand"</span>)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> all_traits){</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>    <span class="co"># list all files that belong to this trait</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>    fileNames <span class="ot">=</span> <span class="fu">list.files</span>(<span class="at">pattern =</span> i) </span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>    fileNames <span class="ot">=</span> fileNames[<span class="fu">grepl</span>(<span class="st">".regenie.gz"</span>, fileNames)]</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>    <span class="co"># exclude the 'interaction' GWAS</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>    fileNames <span class="ot">=</span> fileNames[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"_interaction_"</span>, fileNames)]</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>    <span class="co"># include the 'interaction' GWAS</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>    <span class="co">#fileNames = fileNames[grepl("_interaction_", fileNames)]</span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>    <span class="co"># sex split analyses</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>    <span class="co">#fileNames&lt;-fileNames[grepl("_males", fileNames)]</span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>    <span class="co">#i = paste0(i,"_males")</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># count number of chromosomes</span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(fileNames) <span class="sc">!=</span> <span class="dv">22</span>){<span class="fu">warning</span>(<span class="fu">paste0</span>(i, <span class="st">" GWAS has not got exactly 22 chromosome files!"</span>))}</span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>    <span class="co"># set object to count number of rows to 0</span></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>    count_rows<span class="ot">&lt;-</span><span class="dv">0</span></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>    <span class="co"># object to hold GWAS</span></span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>    dat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()    </span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a>     <span class="cf">for</span>(j <span class="cf">in</span> fileNames){</span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a>       <span class="fu">print</span>(j)</span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a>        <span class="co"># read in file</span></span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a>       file<span class="ot">&lt;-</span><span class="fu">fread</span>(j,<span class="at">header=</span>T,<span class="at">data.table=</span>F)</span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a>        <span class="co"># if this is an interaction test, only keep interaction row for each SNP</span></span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">grepl</span>(<span class="st">"interaction"</span>, j)){</span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a>            <span class="co"># pull out the interaction effect from the full model</span></span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a>            file <span class="ot">&lt;-</span> file[<span class="fu">grepl</span>(<span class="st">"SNPxage"</span>, file<span class="sc">$</span>TEST),] </span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a>            <span class="co"># if we're interested to look at attenuation of SNP effects, we can pull out ADD_INT_SNP which should represent the main effect of the tested SNP in the interaction model</span></span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a>            <span class="co">#file &lt;- file[grepl("INT_SNP", file$TEST),] </span></span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a>            <span class="co"># remove SNPxage</span></span>
<span id="cb4-48"><a href="#cb4-48" aria-hidden="true" tabindex="-1"></a>            <span class="co">#file &lt;- file[!grepl("SNPxage", file$TEST),]</span></span>
<span id="cb4-49"><a href="#cb4-49" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb4-50"><a href="#cb4-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-51"><a href="#cb4-51" aria-hidden="true" tabindex="-1"></a>        <span class="co"># merge</span></span>
<span id="cb4-52"><a href="#cb4-52" aria-hidden="true" tabindex="-1"></a>        dat <span class="ot">&lt;-</span> <span class="fu">rbind</span>(dat, file) </span>
<span id="cb4-53"><a href="#cb4-53" aria-hidden="true" tabindex="-1"></a>       </span>
<span id="cb4-54"><a href="#cb4-54" aria-hidden="true" tabindex="-1"></a>        <span class="co"># count number of rows </span></span>
<span id="cb4-55"><a href="#cb4-55" aria-hidden="true" tabindex="-1"></a>        count_rows<span class="ot">&lt;-</span>count_rows<span class="sc">+</span><span class="fu">nrow</span>(file)</span>
<span id="cb4-56"><a href="#cb4-56" aria-hidden="true" tabindex="-1"></a>     }</span>
<span id="cb4-57"><a href="#cb4-57" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-58"><a href="#cb4-58" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">"Done merging chromosome files"</span>)</span>
<span id="cb4-59"><a href="#cb4-59" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-60"><a href="#cb4-60" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Sanity checks: Does the resulting file have the expected dimensions?</span></span>
<span id="cb4-61"><a href="#cb4-61" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">nrow</span>(dat)<span class="sc">!=</span> count_rows){<span class="fu">print</span>(<span class="st">"Resulting merged file does not match the expected numeber of rows"</span>);<span class="cf">break</span>}</span>
<span id="cb4-62"><a href="#cb4-62" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-63"><a href="#cb4-63" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">ncol</span>(dat) <span class="sc">!=</span> <span class="dv">14</span>){<span class="fu">print</span>(<span class="st">"Resulting merged files does not match the expected number of columns"</span>); <span class="cf">break</span>}</span>
<span id="cb4-64"><a href="#cb4-64" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-65"><a href="#cb4-65" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">"File has dimensions as expected"</span>)</span>
<span id="cb4-66"><a href="#cb4-66" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">dim</span>(dat))</span>
<span id="cb4-67"><a href="#cb4-67" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-68"><a href="#cb4-68" aria-hidden="true" tabindex="-1"></a>    <span class="co"># transform pvalues from log transformation to regular</span></span>
<span id="cb4-69"><a href="#cb4-69" aria-hidden="true" tabindex="-1"></a>    <span class="co"># https://www.biostars.org/p/358046/</span></span>
<span id="cb4-70"><a href="#cb4-70" aria-hidden="true" tabindex="-1"></a>    dat<span class="sc">$</span>P<span class="ot">&lt;-</span><span class="dv">10</span><span class="sc">^</span>(<span class="sc">-</span>dat<span class="sc">$</span>LOG10P)</span>
<span id="cb4-71"><a href="#cb4-71" aria-hidden="true" tabindex="-1"></a>    <span class="co">#if(min(dat$P,na.rm=T)&lt;=0 | max(dat$P,na.rm=T) &gt;= 1){"Transformed p-value is out of bounds"; break}</span></span>
<span id="cb4-72"><a href="#cb4-72" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summary</span>(dat<span class="sc">$</span>P)</span>
<span id="cb4-73"><a href="#cb4-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-74"><a href="#cb4-74" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">"Done transform p-value column, and p-values are between 0 and 1"</span>)</span>
<span id="cb4-75"><a href="#cb4-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-76"><a href="#cb4-76" aria-hidden="true" tabindex="-1"></a>    <span class="co"># create MAF column</span></span>
<span id="cb4-77"><a href="#cb4-77" aria-hidden="true" tabindex="-1"></a>    dat<span class="sc">$</span>MAF<span class="ot">&lt;-</span><span class="fu">ifelse</span>(dat<span class="sc">$</span>A1FREQ <span class="sc">&lt;</span> <span class="fl">0.5</span>, dat<span class="sc">$</span>A1FREQ, <span class="dv">1</span><span class="sc">-</span>dat<span class="sc">$</span>A1FREQ)</span>
<span id="cb4-78"><a href="#cb4-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-79"><a href="#cb4-79" aria-hidden="true" tabindex="-1"></a>    <span class="co"># remove EXTRA column because it's empty</span></span>
<span id="cb4-80"><a href="#cb4-80" aria-hidden="true" tabindex="-1"></a>    dat<span class="sc">$</span>EXTRA <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb4-81"><a href="#cb4-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-82"><a href="#cb4-82" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"This is the file head for "</span>,i))</span>
<span id="cb4-83"><a href="#cb4-83" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">head</span>(dat))</span>
<span id="cb4-84"><a href="#cb4-84" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-85"><a href="#cb4-85" aria-hidden="true" tabindex="-1"></a>    <span class="co"># define file name to save the merged file</span></span>
<span id="cb4-86"><a href="#cb4-86" aria-hidden="true" tabindex="-1"></a>    fileName <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"/CCACE_Shared/Anna_F/BrainAtrophy/data/regenie_out/finalGWAS/GWAS_brainAtrophy_"</span>, i,<span class="st">"_N43110.gz"</span>)</span>
<span id="cb4-87"><a href="#cb4-87" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-88"><a href="#cb4-88" aria-hidden="true" tabindex="-1"></a>    <span class="co"># if this is an interaction run, name accordingly}               </span></span>
<span id="cb4-89"><a href="#cb4-89" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">grepl</span>(<span class="st">"interaction"</span>, j)){</span>
<span id="cb4-90"><a href="#cb4-90" aria-hidden="true" tabindex="-1"></a>        <span class="co">#fileName &lt;- paste0(wd,"/GWAS_brainAtrophy_", i,"_attenuatedMainEffects_N43110.gz")</span></span>
<span id="cb4-91"><a href="#cb4-91" aria-hidden="true" tabindex="-1"></a>        fileName <span class="ot">&lt;-</span> <span class="fu">paste0</span>(wd,<span class="st">"/GWAS_brainAtrophy_"</span>, i,<span class="st">"_SNPxage_N43110.gz"</span>)</span>
<span id="cb4-92"><a href="#cb4-92" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb4-93"><a href="#cb4-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-94"><a href="#cb4-94" aria-hidden="true" tabindex="-1"></a>    <span class="co"># save merged file</span></span>
<span id="cb4-95"><a href="#cb4-95" aria-hidden="true" tabindex="-1"></a>    <span class="fu">fwrite</span>(dat, fileName, <span class="at">na =</span> <span class="st">"NA"</span>, <span class="at">quote =</span> F, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-96"><a href="#cb4-96" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-97"><a href="#cb4-97" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Done writing file for "</span>, i))</span>
<span id="cb4-98"><a href="#cb4-98" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-99"><a href="#cb4-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-100"><a href="#cb4-100" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Done all traits"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="manhattan-plot" class="level2">
<h2 class="anchored" data-anchor-id="manhattan-plot">Manhattan plot</h2>
<p>The scripts below were used to make Figure 4 in the main paper, and Supplementary Plots 15-19.</p>
<section id="alter-manhattan-function-for-better-looking-annotation" class="level3">
<h3 class="anchored" data-anchor-id="alter-manhattan-function-for-better-looking-annotation">Alter Manhattan function for better looking annotation</h3>
<p>This function was saved in a file called <code>manhattan_big.R</code> to be read in with <code>source(manhattan_big.R)</code> in the next step.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>manhattan_big <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">chr=</span><span class="st">"CHR"</span>, <span class="at">bp=</span><span class="st">"BP"</span>, <span class="at">p=</span><span class="st">"P"</span>, <span class="at">snp=</span><span class="st">"SNP"</span>, </span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>                      <span class="at">col=</span><span class="fu">c</span>(<span class="st">"gray10"</span>, <span class="st">"gray60"</span>), <span class="at">chrlabs=</span><span class="cn">NULL</span>,</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">suggestiveline=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">1e-5</span>), <span class="at">genomewideline=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">5e-8</span>), </span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">highlight=</span><span class="cn">NULL</span>, <span class="at">logp=</span><span class="cn">TRUE</span>, <span class="at">annotatePval =</span> <span class="cn">NULL</span>, <span class="at">annotateTop =</span> <span class="cn">TRUE</span>, ...) {</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Not sure why, but package check will warn without this.</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>    CHR<span class="ot">=</span>BP<span class="ot">=</span>P<span class="ot">=</span>index<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Check for sensible dataset</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Make sure you have chr, bp and p columns.</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span>(chr <span class="sc">%in%</span> <span class="fu">names</span>(x))) <span class="fu">stop</span>(<span class="fu">paste</span>(<span class="st">"Column"</span>, chr, <span class="st">"not found!"</span>))</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span>(bp <span class="sc">%in%</span> <span class="fu">names</span>(x))) <span class="fu">stop</span>(<span class="fu">paste</span>(<span class="st">"Column"</span>, bp, <span class="st">"not found!"</span>))</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span>(p <span class="sc">%in%</span> <span class="fu">names</span>(x))) <span class="fu">stop</span>(<span class="fu">paste</span>(<span class="st">"Column"</span>, p, <span class="st">"not found!"</span>))</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>    <span class="do">## warn if you don't have a snp column</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span>(snp <span class="sc">%in%</span> <span class="fu">names</span>(x))) <span class="fu">warning</span>(<span class="fu">paste</span>(<span class="st">"No SNP column found. OK unless you're trying to highlight."</span>))</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>    <span class="do">## make sure chr, bp, and p columns are numeric.</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.numeric</span>(x[[chr]])) <span class="fu">stop</span>(<span class="fu">paste</span>(chr, <span class="st">"column should be numeric. Do you have 'X', 'Y', 'MT', etc? If so change to numbers and try again."</span>))</span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.numeric</span>(x[[bp]])) <span class="fu">stop</span>(<span class="fu">paste</span>(bp, <span class="st">"column should be numeric."</span>))</span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.numeric</span>(x[[p]])) <span class="fu">stop</span>(<span class="fu">paste</span>(p, <span class="st">"column should be numeric."</span>))</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create a new data.frame with columns called CHR, BP, and P.</span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>    <span class="co"># d=data.frame(CHR=x[[chr]], BP=x[[bp]], P=x[[p]], pos = NA, index = NA) # with millions of SNPs, create dataframe at once </span></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>                                                             <span class="co">#  rather than dynamically allocated(see line 72-73, and remove line 87 and line 91 )</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>    <span class="co"># If the input data frame has a SNP column, add it to the new data frame you're creating.</span></span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(x[[snp]])) d <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">CHR=</span>x[[chr]], <span class="at">BP=</span>x[[bp]], <span class="at">P=</span>x[[p]], <span class="at">pos =</span> <span class="cn">NA</span>, <span class="at">index =</span> <span class="cn">NA</span> ,<span class="at">SNP=</span>x[[snp]], <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="cf">else</span> </span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a>        d <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">CHR=</span>x[[chr]], <span class="at">BP=</span>x[[bp]], <span class="at">P=</span>x[[p]], <span class="at">pos =</span> <span class="cn">NA</span>, <span class="at">index =</span> <span class="cn">NA</span>)</span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set positions, ticks, and labels for plotting</span></span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Sort and keep only values where is numeric.</span></span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a>    <span class="co">#d &lt;- subset(d[order(d$CHR, d$BP), ], (P&gt;0 &amp; P&lt;=1 &amp; is.numeric(P)))</span></span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a>    <span class="co">#  d &lt;- subset(d, (is.numeric(CHR) &amp; is.numeric(BP) &amp; is.numeric(P)))       ## unused, all three variables are numeric, line:63-65 </span></span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a>    d <span class="ot">&lt;-</span> d[<span class="fu">order</span>(d<span class="sc">$</span>CHR, d<span class="sc">$</span>BP), ]</span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a>    <span class="co">#d$logp &lt;- ifelse(logp, yes=-log10(d$P), no=d$P)</span></span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (logp) {</span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a>        d<span class="sc">$</span>logp <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(d<span class="sc">$</span>P)</span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true" tabindex="-1"></a>        d<span class="sc">$</span>logp <span class="ot">&lt;-</span> d<span class="sc">$</span>P</span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true" tabindex="-1"></a>   <span class="co"># d$pos=NA</span></span>
<span id="cb5-42"><a href="#cb5-42" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb5-43"><a href="#cb5-43" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb5-44"><a href="#cb5-44" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Fixes the bug where one chromosome is missing by adding a sequential index column.</span></span>
<span id="cb5-45"><a href="#cb5-45" aria-hidden="true" tabindex="-1"></a>   <span class="co"># d$index=NA</span></span>
<span id="cb5-46"><a href="#cb5-46" aria-hidden="true" tabindex="-1"></a>   <span class="co"># ind = 0</span></span>
<span id="cb5-47"><a href="#cb5-47" aria-hidden="true" tabindex="-1"></a>   <span class="co"># for (i in unique(d$CHR)){</span></span>
<span id="cb5-48"><a href="#cb5-48" aria-hidden="true" tabindex="-1"></a>   <span class="co">#     ind = ind + 1</span></span>
<span id="cb5-49"><a href="#cb5-49" aria-hidden="true" tabindex="-1"></a>   <span class="co">#     d[d$CHR==i,]$index = ind</span></span>
<span id="cb5-50"><a href="#cb5-50" aria-hidden="true" tabindex="-1"></a>   <span class="co"># }</span></span>
<span id="cb5-51"><a href="#cb5-51" aria-hidden="true" tabindex="-1"></a>   d<span class="sc">$</span>index <span class="ot">=</span> <span class="fu">rep.int</span>(<span class="fu">seq_along</span>(<span class="fu">unique</span>(d<span class="sc">$</span>CHR)), <span class="at">times =</span> <span class="fu">tapply</span>(d<span class="sc">$</span>SNP,d<span class="sc">$</span>CHR,length))  <span class="co"># replcace the for loop of line 92-96 to improve efficiency</span></span>
<span id="cb5-52"><a href="#cb5-52" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb5-53"><a href="#cb5-53" aria-hidden="true" tabindex="-1"></a>    <span class="co"># This section sets up positions and ticks. Ticks should be placed in the</span></span>
<span id="cb5-54"><a href="#cb5-54" aria-hidden="true" tabindex="-1"></a>    <span class="co"># middle of a chromosome. The a new pos column is added that keeps a running</span></span>
<span id="cb5-55"><a href="#cb5-55" aria-hidden="true" tabindex="-1"></a>    <span class="co"># sum of the positions of each successive chromsome. For example:</span></span>
<span id="cb5-56"><a href="#cb5-56" aria-hidden="true" tabindex="-1"></a>    <span class="co"># chr bp pos</span></span>
<span id="cb5-57"><a href="#cb5-57" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 1   1  1</span></span>
<span id="cb5-58"><a href="#cb5-58" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 1   2  2</span></span>
<span id="cb5-59"><a href="#cb5-59" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 2   1  3</span></span>
<span id="cb5-60"><a href="#cb5-60" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 2   2  4</span></span>
<span id="cb5-61"><a href="#cb5-61" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 3   1  5</span></span>
<span id="cb5-62"><a href="#cb5-62" aria-hidden="true" tabindex="-1"></a>    nchr <span class="ot">=</span> <span class="fu">length</span>(<span class="fu">unique</span>(d<span class="sc">$</span>CHR))</span>
<span id="cb5-63"><a href="#cb5-63" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (nchr<span class="sc">==</span><span class="dv">1</span>) { <span class="do">## For a single chromosome</span></span>
<span id="cb5-64"><a href="#cb5-64" aria-hidden="true" tabindex="-1"></a>        <span class="do">## Uncomment the next two linex to plot single chr results in Mb</span></span>
<span id="cb5-65"><a href="#cb5-65" aria-hidden="true" tabindex="-1"></a>        <span class="co">#options(scipen=999)</span></span>
<span id="cb5-66"><a href="#cb5-66" aria-hidden="true" tabindex="-1"></a>        <span class="co">#d$pos=d$BP/1e6</span></span>
<span id="cb5-67"><a href="#cb5-67" aria-hidden="true" tabindex="-1"></a>        d<span class="sc">$</span>pos<span class="ot">=</span>d<span class="sc">$</span>BP</span>
<span id="cb5-68"><a href="#cb5-68" aria-hidden="true" tabindex="-1"></a>      <span class="co">#  ticks=floor(length(d$pos))/2+1          ## unused, from code line: 169</span></span>
<span id="cb5-69"><a href="#cb5-69" aria-hidden="true" tabindex="-1"></a>        xlabel <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">'Chromosome'</span>,<span class="fu">unique</span>(d<span class="sc">$</span>CHR),<span class="st">'position'</span>)</span>
<span id="cb5-70"><a href="#cb5-70" aria-hidden="true" tabindex="-1"></a>      <span class="co">#  labs = ticks          ## unused, from code line: 169</span></span>
<span id="cb5-71"><a href="#cb5-71" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> { <span class="do">## For multiple chromosomes</span></span>
<span id="cb5-72"><a href="#cb5-72" aria-hidden="true" tabindex="-1"></a>        lastbase<span class="ot">=</span><span class="dv">0</span></span>
<span id="cb5-73"><a href="#cb5-73" aria-hidden="true" tabindex="-1"></a>        ticks<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb5-74"><a href="#cb5-74" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">unique</span>(d<span class="sc">$</span>index)) {</span>
<span id="cb5-75"><a href="#cb5-75" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> (i<span class="sc">==</span><span class="dv">1</span>) {</span>
<span id="cb5-76"><a href="#cb5-76" aria-hidden="true" tabindex="-1"></a>                d[d<span class="sc">$</span>index<span class="sc">==</span>i, ]<span class="sc">$</span>pos<span class="ot">=</span>d[d<span class="sc">$</span>index<span class="sc">==</span>i, ]<span class="sc">$</span>BP</span>
<span id="cb5-77"><a href="#cb5-77" aria-hidden="true" tabindex="-1"></a>            } <span class="cf">else</span> {</span>
<span id="cb5-78"><a href="#cb5-78" aria-hidden="true" tabindex="-1"></a>        <span class="do">## chromosome position maybe not start at 1, eg. 9999. So gaps may be produced. </span></span>
<span id="cb5-79"><a href="#cb5-79" aria-hidden="true" tabindex="-1"></a>        lastbase <span class="ot">=</span> lastbase <span class="sc">+</span><span class="fu">max</span>(d[d<span class="sc">$</span>index<span class="sc">==</span>(i<span class="dv">-1</span>),<span class="st">"BP"</span>])   <span class="co"># replace line 128</span></span>
<span id="cb5-80"><a href="#cb5-80" aria-hidden="true" tabindex="-1"></a>        d[d<span class="sc">$</span>index <span class="sc">==</span> i,<span class="st">"BP"</span>] <span class="ot">=</span> d[d<span class="sc">$</span>index <span class="sc">==</span> i,<span class="st">"BP"</span>]<span class="sc">-</span><span class="fu">min</span>(d[d<span class="sc">$</span>index<span class="sc">==</span>i,<span class="st">"BP"</span>]) <span class="sc">+</span><span class="dv">1</span></span>
<span id="cb5-81"><a href="#cb5-81" aria-hidden="true" tabindex="-1"></a>        d[d<span class="sc">$</span>index <span class="sc">==</span> i, <span class="st">"pos"</span>] <span class="ot">=</span> d[d<span class="sc">$</span>index <span class="sc">==</span> i,<span class="st">"BP"</span>] <span class="sc">+</span> lastbase    <span class="co"># replace line 129</span></span>
<span id="cb5-82"><a href="#cb5-82" aria-hidden="true" tabindex="-1"></a>                <span class="co"># lastbase=lastbase+tail(subset(d,index==i-1)$BP, 1)</span></span>
<span id="cb5-83"><a href="#cb5-83" aria-hidden="true" tabindex="-1"></a>                <span class="co"># d[d$index==i, ]$pos=d[d$index==i, ]$BP+lastbase</span></span>
<span id="cb5-84"><a href="#cb5-84" aria-hidden="true" tabindex="-1"></a>           </span>
<span id="cb5-85"><a href="#cb5-85" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb5-86"><a href="#cb5-86" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Old way: assumes SNPs evenly distributed</span></span>
<span id="cb5-87"><a href="#cb5-87" aria-hidden="true" tabindex="-1"></a>            <span class="co"># ticks=c(ticks, d[d$index==i, ]$pos[floor(length(d[d$index==i, ]$pos)/2)+1])</span></span>
<span id="cb5-88"><a href="#cb5-88" aria-hidden="true" tabindex="-1"></a>            <span class="co"># New way: doesn't make that assumption</span></span>
<span id="cb5-89"><a href="#cb5-89" aria-hidden="true" tabindex="-1"></a>           <span class="co"># ticks = c(ticks, (min(d[d$index == i,]$pos) + max(d[d$index == i,]$pos))/2 + 1)  # see line 136, to reduce the burden of for loop </span></span>
<span id="cb5-90"><a href="#cb5-90" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb5-91"><a href="#cb5-91" aria-hidden="true" tabindex="-1"></a>    ticks <span class="ot">&lt;-</span><span class="fu">tapply</span>(d<span class="sc">$</span>pos,d<span class="sc">$</span>index,quantile,<span class="at">probs=</span><span class="fl">0.5</span>)   <span class="co"># replace line 135</span></span>
<span id="cb5-92"><a href="#cb5-92" aria-hidden="true" tabindex="-1"></a>        xlabel <span class="ot">=</span> <span class="st">'Chromosome'</span></span>
<span id="cb5-93"><a href="#cb5-93" aria-hidden="true" tabindex="-1"></a>        <span class="co">#labs = append(unique(d$CHR),'') ## I forgot what this was here for... if seems to work, remove.</span></span>
<span id="cb5-94"><a href="#cb5-94" aria-hidden="true" tabindex="-1"></a>        labs <span class="ot">&lt;-</span> <span class="fu">unique</span>(d<span class="sc">$</span>CHR)</span>
<span id="cb5-95"><a href="#cb5-95" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb5-96"><a href="#cb5-96" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb5-97"><a href="#cb5-97" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Initialize plot</span></span>
<span id="cb5-98"><a href="#cb5-98" aria-hidden="true" tabindex="-1"></a>    xmax <span class="ot">=</span> <span class="fu">ceiling</span>(<span class="fu">max</span>(d<span class="sc">$</span>pos) <span class="sc">*</span> <span class="fl">1.03</span>)</span>
<span id="cb5-99"><a href="#cb5-99" aria-hidden="true" tabindex="-1"></a>    xmin <span class="ot">=</span> <span class="fu">floor</span>(<span class="fu">max</span>(d<span class="sc">$</span>pos) <span class="sc">*</span> <span class="sc">-</span><span class="fl">0.03</span>)</span>
<span id="cb5-100"><a href="#cb5-100" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb5-101"><a href="#cb5-101" aria-hidden="true" tabindex="-1"></a>    <span class="co"># The old way to initialize the plot</span></span>
<span id="cb5-102"><a href="#cb5-102" aria-hidden="true" tabindex="-1"></a>    <span class="co"># plot(NULL, xaxt='n', bty='n', xaxs='i', yaxs='i', xlim=c(xmin,xmax), ylim=c(ymin,ymax),</span></span>
<span id="cb5-103"><a href="#cb5-103" aria-hidden="true" tabindex="-1"></a>    <span class="co">#      xlab=xlabel, ylab=expression(-log[10](italic(p))), las=1, pch=20, ...)</span></span>
<span id="cb5-104"><a href="#cb5-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-105"><a href="#cb5-105" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb5-106"><a href="#cb5-106" aria-hidden="true" tabindex="-1"></a>    <span class="co"># The new way to initialize the plot.</span></span>
<span id="cb5-107"><a href="#cb5-107" aria-hidden="true" tabindex="-1"></a>    <span class="do">## See http://stackoverflow.com/q/23922130/654296</span></span>
<span id="cb5-108"><a href="#cb5-108" aria-hidden="true" tabindex="-1"></a>    <span class="do">## First, define your default arguments</span></span>
<span id="cb5-109"><a href="#cb5-109" aria-hidden="true" tabindex="-1"></a>    def_args <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">xaxt=</span><span class="st">'n'</span>, <span class="at">bty=</span><span class="st">'n'</span>, <span class="at">xaxs=</span><span class="st">'i'</span>, <span class="at">yaxs=</span><span class="st">'i'</span>, <span class="at">las=</span><span class="dv">1</span>, <span class="at">pch=</span><span class="dv">20</span>,</span>
<span id="cb5-110"><a href="#cb5-110" aria-hidden="true" tabindex="-1"></a>                     <span class="at">xlim=</span><span class="fu">c</span>(xmin,xmax), <span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fu">ceiling</span>(<span class="fu">max</span>(d<span class="sc">$</span>logp))),</span>
<span id="cb5-111"><a href="#cb5-111" aria-hidden="true" tabindex="-1"></a>                     <span class="at">xlab=</span>xlabel, <span class="at">ylab=</span><span class="fu">expression</span>(<span class="sc">-</span>log[<span class="dv">10</span>](<span class="fu">italic</span>(p))))</span>
<span id="cb5-112"><a href="#cb5-112" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Next, get a list of ... arguments</span></span>
<span id="cb5-113"><a href="#cb5-113" aria-hidden="true" tabindex="-1"></a>    <span class="co">#dotargs &lt;- as.list(match.call())[-1L]</span></span>
<span id="cb5-114"><a href="#cb5-114" aria-hidden="true" tabindex="-1"></a>    dotargs <span class="ot">&lt;-</span> <span class="fu">list</span>(...)</span>
<span id="cb5-115"><a href="#cb5-115" aria-hidden="true" tabindex="-1"></a>    <span class="do">## And call the plot function passing NA, your ... arguments, and the default</span></span>
<span id="cb5-116"><a href="#cb5-116" aria-hidden="true" tabindex="-1"></a>    <span class="do">## arguments that were not defined in the ... arguments.</span></span>
<span id="cb5-117"><a href="#cb5-117" aria-hidden="true" tabindex="-1"></a>    <span class="fu">do.call</span>(<span class="st">"plot"</span>, <span class="fu">c</span>(<span class="cn">NA</span>, dotargs, def_args[<span class="sc">!</span><span class="fu">names</span>(def_args) <span class="sc">%in%</span> <span class="fu">names</span>(dotargs)]))</span>
<span id="cb5-118"><a href="#cb5-118" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb5-119"><a href="#cb5-119" aria-hidden="true" tabindex="-1"></a>    <span class="co"># If manually specifying chromosome labels, ensure a character vector and number of labels matches number chrs.</span></span>
<span id="cb5-120"><a href="#cb5-120" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(chrlabs)) {</span>
<span id="cb5-121"><a href="#cb5-121" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">is.character</span>(chrlabs)) {</span>
<span id="cb5-122"><a href="#cb5-122" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> (<span class="fu">length</span>(chrlabs)<span class="sc">==</span><span class="fu">length</span>(labs)) {</span>
<span id="cb5-123"><a href="#cb5-123" aria-hidden="true" tabindex="-1"></a>                labs <span class="ot">&lt;-</span> chrlabs</span>
<span id="cb5-124"><a href="#cb5-124" aria-hidden="true" tabindex="-1"></a>            } <span class="cf">else</span> {</span>
<span id="cb5-125"><a href="#cb5-125" aria-hidden="true" tabindex="-1"></a>                <span class="fu">warning</span>(<span class="st">"You're trying to specify chromosome labels but the number of labels != number of chromosomes."</span>)</span>
<span id="cb5-126"><a href="#cb5-126" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb5-127"><a href="#cb5-127" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> {</span>
<span id="cb5-128"><a href="#cb5-128" aria-hidden="true" tabindex="-1"></a>            <span class="fu">warning</span>(<span class="st">"If you're trying to specify chromosome labels, chrlabs must be a character vector"</span>)</span>
<span id="cb5-129"><a href="#cb5-129" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb5-130"><a href="#cb5-130" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb5-131"><a href="#cb5-131" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb5-132"><a href="#cb5-132" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Add an axis. </span></span>
<span id="cb5-133"><a href="#cb5-133" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (nchr<span class="sc">==</span><span class="dv">1</span>) { <span class="co">#If single chromosome, ticks and labels automatic.</span></span>
<span id="cb5-134"><a href="#cb5-134" aria-hidden="true" tabindex="-1"></a>        <span class="fu">axis</span>(<span class="dv">1</span>, ...)</span>
<span id="cb5-135"><a href="#cb5-135" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> { <span class="co"># if multiple chrs, use the ticks and labels you created above.</span></span>
<span id="cb5-136"><a href="#cb5-136" aria-hidden="true" tabindex="-1"></a>        <span class="fu">axis</span>(<span class="dv">1</span>, <span class="at">at=</span>ticks, <span class="at">labels=</span>labs, ...)</span>
<span id="cb5-137"><a href="#cb5-137" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb5-138"><a href="#cb5-138" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb5-139"><a href="#cb5-139" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create a vector of alternatiting colors</span></span>
<span id="cb5-140"><a href="#cb5-140" aria-hidden="true" tabindex="-1"></a>    <span class="co">#col=rep(col, max(d$CHR))  # replaced by line 187</span></span>
<span id="cb5-141"><a href="#cb5-141" aria-hidden="true" tabindex="-1"></a>    col <span class="ot">=</span> <span class="fu">rep_len</span>(col, <span class="fu">max</span>(d<span class="sc">$</span>index))  <span class="do">## mean this one?  the results are same</span></span>
<span id="cb5-142"><a href="#cb5-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-143"><a href="#cb5-143" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Add points to the plot</span></span>
<span id="cb5-144"><a href="#cb5-144" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (nchr<span class="sc">==</span><span class="dv">1</span>) {</span>
<span id="cb5-145"><a href="#cb5-145" aria-hidden="true" tabindex="-1"></a>        <span class="fu">with</span>(d, <span class="fu">points</span>(pos, logp, <span class="at">pch=</span><span class="dv">20</span>, <span class="at">col=</span>col[<span class="dv">1</span>], ...))</span>
<span id="cb5-146"><a href="#cb5-146" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb5-147"><a href="#cb5-147" aria-hidden="true" tabindex="-1"></a>        <span class="co"># if multiple chromosomes, need to alternate colors and increase the color index (icol) each chr.</span></span>
<span id="cb5-148"><a href="#cb5-148" aria-hidden="true" tabindex="-1"></a>        icol<span class="ot">=</span><span class="dv">1</span></span>
<span id="cb5-149"><a href="#cb5-149" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">unique</span>(d<span class="sc">$</span>index)) {</span>
<span id="cb5-150"><a href="#cb5-150" aria-hidden="true" tabindex="-1"></a>            <span class="co">#with(d[d$index==unique(d$index)[i], ], points(pos, logp, col=col[icol], pch=20, ...))</span></span>
<span id="cb5-151"><a href="#cb5-151" aria-hidden="true" tabindex="-1"></a>        <span class="fu">points</span>(d[d<span class="sc">$</span>index<span class="sc">==</span>i,<span class="st">"pos"</span>], d[d<span class="sc">$</span>index<span class="sc">==</span>i,<span class="st">"logp"</span>], <span class="at">col=</span>col[icol], <span class="at">pch=</span><span class="dv">20</span>, ...)</span>
<span id="cb5-152"><a href="#cb5-152" aria-hidden="true" tabindex="-1"></a>            icol<span class="ot">=</span>icol<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb5-153"><a href="#cb5-153" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb5-154"><a href="#cb5-154" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb5-155"><a href="#cb5-155" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb5-156"><a href="#cb5-156" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Add suggestive and genomewide lines</span></span>
<span id="cb5-157"><a href="#cb5-157" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (suggestiveline) <span class="fu">abline</span>(<span class="at">h=</span>suggestiveline, <span class="at">col=</span><span class="st">"blue"</span>)</span>
<span id="cb5-158"><a href="#cb5-158" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (genomewideline) <span class="fu">abline</span>(<span class="at">h=</span>genomewideline, <span class="at">col=</span><span class="st">"red"</span>)</span>
<span id="cb5-159"><a href="#cb5-159" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb5-160"><a href="#cb5-160" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Highlight snps from a character vector</span></span>
<span id="cb5-161"><a href="#cb5-161" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(highlight)) {</span>
<span id="cb5-162"><a href="#cb5-162" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">any</span>(<span class="sc">!</span>(highlight <span class="sc">%in%</span> d<span class="sc">$</span>SNP))) <span class="fu">warning</span>(<span class="st">"You're trying to highlight SNPs that don't exist in your results."</span>)</span>
<span id="cb5-163"><a href="#cb5-163" aria-hidden="true" tabindex="-1"></a>        d.highlight<span class="ot">=</span>d[<span class="fu">which</span>(d<span class="sc">$</span>SNP <span class="sc">%in%</span> highlight), ]</span>
<span id="cb5-164"><a href="#cb5-164" aria-hidden="true" tabindex="-1"></a>        <span class="fu">with</span>(d.highlight, <span class="fu">points</span>(pos, logp, <span class="at">col=</span><span class="st">"green3"</span>, <span class="at">pch=</span><span class="dv">20</span>, ...)) </span>
<span id="cb5-165"><a href="#cb5-165" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb5-166"><a href="#cb5-166" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb5-167"><a href="#cb5-167" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Highlight top SNPs</span></span>
<span id="cb5-168"><a href="#cb5-168" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(annotatePval)) {</span>
<span id="cb5-169"><a href="#cb5-169" aria-hidden="true" tabindex="-1"></a>        <span class="co"># extract top SNPs at given p-val</span></span>
<span id="cb5-170"><a href="#cb5-170" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (logp) {</span>
<span id="cb5-171"><a href="#cb5-171" aria-hidden="true" tabindex="-1"></a>            topHits <span class="ot">=</span> <span class="fu">subset</span>(d, P <span class="sc">&lt;=</span> annotatePval)</span>
<span id="cb5-172"><a href="#cb5-172" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span></span>
<span id="cb5-173"><a href="#cb5-173" aria-hidden="true" tabindex="-1"></a>            topHits <span class="ot">=</span> <span class="fu">subset</span>(d, P <span class="sc">&gt;=</span> annotatePval)</span>
<span id="cb5-174"><a href="#cb5-174" aria-hidden="true" tabindex="-1"></a>        <span class="fu">par</span>(<span class="at">xpd =</span> <span class="cn">TRUE</span>)</span>
<span id="cb5-175"><a href="#cb5-175" aria-hidden="true" tabindex="-1"></a>        <span class="co"># annotate these SNPs</span></span>
<span id="cb5-176"><a href="#cb5-176" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (annotateTop <span class="sc">==</span> <span class="cn">FALSE</span>) {</span>
<span id="cb5-177"><a href="#cb5-177" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span> (logp) {</span>
<span id="cb5-178"><a href="#cb5-178" aria-hidden="true" tabindex="-1"></a>              <span class="fu">with</span>(<span class="fu">subset</span>(d, P <span class="sc">&lt;=</span> annotatePval), </span>
<span id="cb5-179"><a href="#cb5-179" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">textxy</span>(pos, <span class="sc">-</span><span class="fu">log10</span>(P), <span class="at">offset =</span> <span class="fl">0.625</span>, <span class="at">labs =</span> topHits<span class="sc">$</span>SNP, <span class="at">cex =</span> <span class="dv">1</span>), ...)</span>
<span id="cb5-180"><a href="#cb5-180" aria-hidden="true" tabindex="-1"></a>          } <span class="cf">else</span></span>
<span id="cb5-181"><a href="#cb5-181" aria-hidden="true" tabindex="-1"></a>              <span class="fu">with</span>(<span class="fu">subset</span>(d, P <span class="sc">&gt;=</span> annotatePval), </span>
<span id="cb5-182"><a href="#cb5-182" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">textxy</span>(pos, P, <span class="at">offset =</span> <span class="fl">0.625</span>, <span class="at">labs =</span> topHits<span class="sc">$</span>SNP, <span class="at">cex =</span> <span class="dv">1</span>), ...)</span>
<span id="cb5-183"><a href="#cb5-183" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb5-184"><a href="#cb5-184" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span> {</span>
<span id="cb5-185"><a href="#cb5-185" aria-hidden="true" tabindex="-1"></a>            <span class="co"># could try alternative, annotate top SNP of each sig chr</span></span>
<span id="cb5-186"><a href="#cb5-186" aria-hidden="true" tabindex="-1"></a>            topHits <span class="ot">&lt;-</span> topHits[<span class="fu">order</span>(topHits<span class="sc">$</span>P),]</span>
<span id="cb5-187"><a href="#cb5-187" aria-hidden="true" tabindex="-1"></a>            topSNPs <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb5-188"><a href="#cb5-188" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb5-189"><a href="#cb5-189" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">unique</span>(topHits<span class="sc">$</span>CHR)) {</span>
<span id="cb5-190"><a href="#cb5-190" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb5-191"><a href="#cb5-191" aria-hidden="true" tabindex="-1"></a>                chrSNPs <span class="ot">&lt;-</span> topHits[topHits<span class="sc">$</span>CHR <span class="sc">==</span> i,]</span>
<span id="cb5-192"><a href="#cb5-192" aria-hidden="true" tabindex="-1"></a>                topSNPs <span class="ot">&lt;-</span> <span class="fu">rbind</span>(topSNPs, chrSNPs[<span class="dv">1</span>,])</span>
<span id="cb5-193"><a href="#cb5-193" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb5-194"><a href="#cb5-194" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb5-195"><a href="#cb5-195" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> (logp ){</span>
<span id="cb5-196"><a href="#cb5-196" aria-hidden="true" tabindex="-1"></a>                <span class="fu">textxy</span>(topSNPs<span class="sc">$</span>pos, <span class="sc">-</span><span class="fu">log10</span>(topSNPs<span class="sc">$</span>P), <span class="at">offset =</span> <span class="fl">0.625</span>, <span class="at">labs =</span> topSNPs<span class="sc">$</span>SNP, <span class="at">cex =</span> <span class="dv">1</span>, ...)</span>
<span id="cb5-197"><a href="#cb5-197" aria-hidden="true" tabindex="-1"></a>            } <span class="cf">else</span></span>
<span id="cb5-198"><a href="#cb5-198" aria-hidden="true" tabindex="-1"></a>              <span class="fu">textxy</span>(topSNPs<span class="sc">$</span>pos, topSNPs<span class="sc">$</span>P, <span class="at">offset =</span> <span class="fl">0.625</span>, <span class="at">labs =</span> topSNPs<span class="sc">$</span>SNP, <span class="at">cex =</span> <span class="dv">1</span>, ...)</span>
<span id="cb5-199"><a href="#cb5-199" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb5-200"><a href="#cb5-200" aria-hidden="true" tabindex="-1"></a>    }  </span>
<span id="cb5-201"><a href="#cb5-201" aria-hidden="true" tabindex="-1"></a>    <span class="fu">par</span>(<span class="at">xpd =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-202"><a href="#cb5-202" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="plot-gwas-associations" class="level3">
<h3 class="anchored" data-anchor-id="plot-gwas-associations">Plot GWAS associations</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="do">#######</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages("R.utils") # to read .gz files</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages("qqman")</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages("calibrate")</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(qqman)</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(calibrate)</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Manhattan plot</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="co"># loop through all traits</span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>all_traits <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"resid_stand"</span>, <span class="st">"ratio_stand"</span>, <span class="st">"diff_stand"</span>, <span class="st">"TBVstand"</span>, <span class="st">"ICVstand"</span>, <span class="st">"CSFstand"</span>)</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(trait_name <span class="cf">in</span> all_traits){</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>    dat <span class="ot">=</span> <span class="fu">fread</span>(<span class="fu">list.files</span>(<span class="at">pattern =</span> <span class="fu">paste0</span>(trait_name, <span class="st">"_N"</span>)), <span class="at">data.table =</span> F)</span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>    <span class="do">## remove MAF below 0.01</span></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>    dat <span class="ot">=</span> dat[dat<span class="sc">$</span>MAF <span class="sc">&gt;</span> <span class="fl">0.01</span>,]</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>    <span class="do">## remove INFO below 0.8</span></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>    dat <span class="ot">=</span> dat[dat<span class="sc">$</span>INFO <span class="sc">&gt;</span> <span class="fl">0.8</span>,]</span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>    <span class="do">## for some reason the function plots too many SNP names below indicated threshold - reomve names </span></span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>    dat[<span class="fu">which</span>(dat<span class="sc">$</span>P <span class="sc">&gt;</span> <span class="fl">5e-8</span>), <span class="st">"ID"</span>] <span class="ot">=</span> <span class="cn">NA</span></span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">source</span>(<span class="fu">paste0</span>(wd,<span class="st">"manhattan_big.R"</span>))</span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">png</span>(<span class="at">filename =</span> <span class="fu">paste0</span>(wd, <span class="st">"/Manhattan"</span>,trait_name,<span class="st">"_clean.png"</span>), <span class="at">width =</span> <span class="dv">1575</span>, <span class="at">height=</span><span class="dv">700</span>, <span class="at">units=</span><span class="st">"px"</span>)</span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a>        layout_matrix <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="at">nrow =</span> <span class="dv">1</span>, <span class="at">ncol=</span><span class="dv">2</span>)</span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a>        <span class="fu">layout</span>(layout_matrix,  <span class="at">widths =</span> <span class="dv">2</span><span class="sc">:</span><span class="dv">1</span>) <span class="co">#heights = 1.5:1,</span></span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a>        <span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">2</span>))</span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a>        <span class="fu">par</span>(<span class="at">oma =</span> <span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">3</span>))</span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a>    <span class="co"># determine color for each of the comp methods</span></span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(trait_name <span class="sc">==</span> <span class="st">"resid_stand"</span>){col <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"#004D40"</span>, <span class="st">"#00e7c0"</span>);pretty_name <span class="ot">=</span> <span class="st">"Lifetime atrophy (Residual score)"</span>}</span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(trait_name <span class="sc">==</span> <span class="st">"ratio_stand"</span>){col <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"#FFC107"</span>, <span class="st">"#FFE7a1"</span>); pretty_name <span class="ot">=</span> <span class="st">"Lifetime atrophy (Ratio score)"</span>}</span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(trait_name <span class="sc">==</span> <span class="st">"diff_stand"</span>){col <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"#D81B60"</span>, <span class="st">"#F29ABA"</span>); pretty_name <span class="ot">=</span> <span class="st">"Lifetime atrophy (Difference score)"</span>}</span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-40"><a href="#cb6-40" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Make MANHATTAN PLOT</span></span>
<span id="cb6-41"><a href="#cb6-41" aria-hidden="true" tabindex="-1"></a>    main <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">"Manhattan plot: "</span>, pretty_name)</span>
<span id="cb6-42"><a href="#cb6-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">manhattan_big</span>(dat, </span>
<span id="cb6-43"><a href="#cb6-43" aria-hidden="true" tabindex="-1"></a>                <span class="at">main =</span> main, </span>
<span id="cb6-44"><a href="#cb6-44" aria-hidden="true" tabindex="-1"></a>                <span class="at">col =</span> col, </span>
<span id="cb6-45"><a href="#cb6-45" aria-hidden="true" tabindex="-1"></a>                <span class="at">chr=</span><span class="st">"CHROM"</span>,</span>
<span id="cb6-46"><a href="#cb6-46" aria-hidden="true" tabindex="-1"></a>                <span class="at">bp=</span><span class="st">"GENPOS"</span>,</span>
<span id="cb6-47"><a href="#cb6-47" aria-hidden="true" tabindex="-1"></a>                <span class="at">snp=</span><span class="st">"ID"</span>,</span>
<span id="cb6-48"><a href="#cb6-48" aria-hidden="true" tabindex="-1"></a>                <span class="at">p=</span><span class="st">"P"</span>, </span>
<span id="cb6-49"><a href="#cb6-49" aria-hidden="true" tabindex="-1"></a>                <span class="at">annotatePval=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">5e-12</span>),</span>
<span id="cb6-50"><a href="#cb6-50" aria-hidden="true" tabindex="-1"></a>                <span class="at">cex.axis =</span> <span class="fl">1.5</span>)</span>
<span id="cb6-51"><a href="#cb6-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-52"><a href="#cb6-52" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Assess systmeatic bias using Genomic Inflation factor</span></span>
<span id="cb6-53"><a href="#cb6-53" aria-hidden="true" tabindex="-1"></a>    <span class="co"># the genomic inflation factor is defined as the ratio of the median of the empirically observed distribution of test statistics to the expected mean, thus quantifying the extent of the bulk inflation and the excess false positive rate</span></span>
<span id="cb6-54"><a href="#cb6-54" aria-hidden="true" tabindex="-1"></a>    chisq <span class="ot">&lt;-</span> <span class="fu">qchisq</span>(<span class="dv">1</span><span class="sc">-</span>dat<span class="sc">$</span>P,<span class="dv">1</span>)</span>
<span id="cb6-55"><a href="#cb6-55" aria-hidden="true" tabindex="-1"></a>    lambda <span class="ot">&lt;-</span> <span class="fu">median</span>(chisq)<span class="sc">/</span><span class="fu">qchisq</span>(<span class="fl">0.5</span>,<span class="dv">1</span>)</span>
<span id="cb6-56"><a href="#cb6-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-57"><a href="#cb6-57" aria-hidden="true" tabindex="-1"></a>    <span class="co"># make qq plot</span></span>
<span id="cb6-58"><a href="#cb6-58" aria-hidden="true" tabindex="-1"></a>    main <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"QQ plot: "</span>, pretty_name, <span class="st">" (Lambda = "</span>, <span class="fu">round</span>(lambda, <span class="at">digits =</span> <span class="dv">3</span>), <span class="st">")"</span>)</span>
<span id="cb6-59"><a href="#cb6-59" aria-hidden="true" tabindex="-1"></a>    <span class="fu">qq</span>(dat<span class="sc">$</span>P, <span class="at">main =</span> main, <span class="at">cex =</span> <span class="dv">1</span>, <span class="at">cex.axis =</span> <span class="fl">1.5</span>)</span>
<span id="cb6-60"><a href="#cb6-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-61"><a href="#cb6-61" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dev.off</span>()</span>
<span id="cb6-62"><a href="#cb6-62" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/Manhattanratio_stand.png" class="img-fluid figure-img"></p>
</figure>
</div>
</section>
<section id="miami-plots" class="level3">
<h3 class="anchored" data-anchor-id="miami-plots">Miami plots</h3>
<p>This script was used to generate Supplementary Plots 20-23 and 28-32.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Aim is to contrast main effects with the main effects that controlled for interaction effects for each of the three lifetime atrophy phenotypes</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. Read in main effects</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. Read in attenuated main effects</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="co"># 3. Create dummy columns to indicate which is which (will be used for split_by column)</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co"># 4. Merge the two GWAS</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 5. Call function to contrast the two</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(miamiplot)</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>all_traits <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"resid_stand"</span>, <span class="st">"ratio_stand"</span>, <span class="st">"diff_stand"</span>,<span class="st">"TBVstand"</span>)</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="co"># set INFO filter</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>INFOfilter <span class="ot">=</span> <span class="fl">0.9</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a><span class="co"># keep only diallelic SNPs</span></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>diallelic <span class="ot">=</span> <span class="cn">TRUE</span></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a><span class="co"># MAF filter</span></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>MAFfilter <span class="ot">=</span> <span class="fl">0.01</span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a><span class="co">#i = all_traits[1]</span></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> all_traits){</span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 1. Read in main effects</span></span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>    main <span class="ot">=</span> <span class="fu">fread</span>(<span class="fu">list.files</span>(<span class="at">pattern =</span> <span class="fu">paste0</span>(i, <span class="st">"_N"</span>)), <span class="at">data.table =</span> F)</span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a>    <span class="co"># dummy var</span></span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a>    main<span class="sc">$</span>split <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 2. Read in attenuated main effects</span></span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a>    <span class="co">#att = fread(list.files(pattern = paste0(i, "_attenuatedMainEffects")), data.table = F)</span></span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a>    att <span class="ot">=</span> <span class="fu">fread</span>(<span class="fu">list.files</span>(<span class="at">pattern =</span> <span class="fu">paste0</span>(i, <span class="st">"_SNPxage"</span>)), <span class="at">data.table =</span> F)</span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a>    <span class="co"># dummy var</span></span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a>    att<span class="sc">$</span>split <span class="ot">=</span> <span class="sc">-</span><span class="dv">1</span></span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a>    <span class="co"># some associations have a P value of 0 which will give an infinite value when transforming it into log</span></span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a>    <span class="co"># this is the case for 18,794 SNPs in resid GWAS for example, will exclude them here as plot otherwise does not work</span></span>
<span id="cb7-42"><a href="#cb7-42" aria-hidden="true" tabindex="-1"></a>    <span class="co">#att &lt;- att[att$P != 0,]</span></span>
<span id="cb7-43"><a href="#cb7-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-44"><a href="#cb7-44" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 4. Merge the two GWAS</span></span>
<span id="cb7-45"><a href="#cb7-45" aria-hidden="true" tabindex="-1"></a>    merged <span class="ot">=</span> <span class="fu">rbind</span>(main, att)</span>
<span id="cb7-46"><a href="#cb7-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-47"><a href="#cb7-47" aria-hidden="true" tabindex="-1"></a>    <span class="co"># if there is an info filter, filter for more reliable SNPs</span></span>
<span id="cb7-48"><a href="#cb7-48" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">any</span>(<span class="fu">ls</span>() <span class="sc">==</span> <span class="st">"INFOfilter"</span>)){merged <span class="ot">&lt;-</span> merged[merged<span class="sc">$</span>INFO <span class="sc">&gt;</span> INFOfilter,]}</span>
<span id="cb7-49"><a href="#cb7-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-50"><a href="#cb7-50" aria-hidden="true" tabindex="-1"></a>    <span class="co"># if there is an MAF filter, filter for MAF</span></span>
<span id="cb7-51"><a href="#cb7-51" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">any</span>(<span class="fu">ls</span>() <span class="sc">==</span> <span class="st">"MAFfilter"</span>)){merged <span class="ot">&lt;-</span> merged[merged<span class="sc">$</span>MAF <span class="sc">&gt;</span> MAFfilter,]}</span>
<span id="cb7-52"><a href="#cb7-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-53"><a href="#cb7-53" aria-hidden="true" tabindex="-1"></a>    <span class="co"># if diallelic has been set to TRUE, filter for diallelic SNPs only</span></span>
<span id="cb7-54"><a href="#cb7-54" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">any</span>(<span class="fu">ls</span>() <span class="sc">==</span> <span class="st">"diallelic"</span>)){   </span>
<span id="cb7-55"><a href="#cb7-55" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(diallelic <span class="sc">==</span> T){</span>
<span id="cb7-56"><a href="#cb7-56" aria-hidden="true" tabindex="-1"></a>            merged <span class="ot">&lt;-</span> merged[<span class="fu">which</span>(<span class="fu">nchar</span>(merged<span class="sc">$</span>ALLELE0) <span class="sc">==</span> <span class="dv">1</span>),]</span>
<span id="cb7-57"><a href="#cb7-57" aria-hidden="true" tabindex="-1"></a>            merged <span class="ot">&lt;-</span> merged[<span class="fu">which</span>(<span class="fu">nchar</span>(merged<span class="sc">$</span>ALLELE1) <span class="sc">==</span> <span class="dv">1</span>),]</span>
<span id="cb7-58"><a href="#cb7-58" aria-hidden="true" tabindex="-1"></a>    }}</span>
<span id="cb7-59"><a href="#cb7-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-60"><a href="#cb7-60" aria-hidden="true" tabindex="-1"></a>    <span class="co">#merged = merged[merged$CHROM == 1,] # for testing</span></span>
<span id="cb7-61"><a href="#cb7-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-62"><a href="#cb7-62" aria-hidden="true" tabindex="-1"></a>    <span class="co"># summary(merged$P[which(merged$split == -1)])</span></span>
<span id="cb7-63"><a href="#cb7-63" aria-hidden="true" tabindex="-1"></a>    <span class="co"># summary(merged$P[which(merged$split == 1)])</span></span>
<span id="cb7-64"><a href="#cb7-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-65"><a href="#cb7-65" aria-hidden="true" tabindex="-1"></a>    <span class="co"># clean up big files</span></span>
<span id="cb7-66"><a href="#cb7-66" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">c</span>(<span class="st">"att"</span>, <span class="st">"main"</span>))</span>
<span id="cb7-67"><a href="#cb7-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-68"><a href="#cb7-68" aria-hidden="true" tabindex="-1"></a>    <span class="co"># plot</span></span>
<span id="cb7-69"><a href="#cb7-69" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">grepl</span>(<span class="st">"diff"</span>, i)){prettyName <span class="ot">=</span> <span class="st">"Difference score"</span>; col <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"#D81B60"</span>, <span class="st">"#F29ABA"</span>)}</span>
<span id="cb7-70"><a href="#cb7-70" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">grepl</span>(<span class="st">"ratio"</span>, i)){prettyName <span class="ot">=</span> <span class="st">"Ratio score"</span>; col <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"#FFC107"</span>, <span class="st">"#FFE7a1"</span>)}</span>
<span id="cb7-71"><a href="#cb7-71" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">grepl</span>(<span class="st">"resid"</span>, i)){prettyName <span class="ot">=</span> <span class="st">"Residual score"</span>; col <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"#004D40"</span>, <span class="st">"#00e7c0"</span>)}</span>
<span id="cb7-72"><a href="#cb7-72" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">grepl</span>(<span class="st">"TBV"</span>, i)){prettyName <span class="ot">=</span> <span class="st">"TBV"</span>; col <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"#82A0D8"</span>,<span class="st">"#8DDFCB"</span>)}</span>
<span id="cb7-73"><a href="#cb7-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-74"><a href="#cb7-74" aria-hidden="true" tabindex="-1"></a>    ulabel <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">"Main SNP effects</span><span class="sc">\n</span><span class="st">(Original GWAS)"</span>)</span>
<span id="cb7-75"><a href="#cb7-75" aria-hidden="true" tabindex="-1"></a>    <span class="co">#llabel = paste0( "Main SNP effects\n(GWAS including SNP-by-age interaction term)")</span></span>
<span id="cb7-76"><a href="#cb7-76" aria-hidden="true" tabindex="-1"></a>    llabel <span class="ot">=</span> <span class="fu">paste0</span>( <span class="st">"SNP-by-age effects (Interaction GWAS)"</span>)</span>
<span id="cb7-77"><a href="#cb7-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-78"><a href="#cb7-78" aria-hidden="true" tabindex="-1"></a>    p <span class="ot">=</span> <span class="fu">ggmiami</span>(<span class="at">data =</span> merged, </span>
<span id="cb7-79"><a href="#cb7-79" aria-hidden="true" tabindex="-1"></a>            <span class="at">chr =</span> <span class="st">"CHROM"</span>,</span>
<span id="cb7-80"><a href="#cb7-80" aria-hidden="true" tabindex="-1"></a>            <span class="at">pos =</span> <span class="st">"GENPOS"</span>,</span>
<span id="cb7-81"><a href="#cb7-81" aria-hidden="true" tabindex="-1"></a>            <span class="at">p =</span> <span class="st">"P"</span>,</span>
<span id="cb7-82"><a href="#cb7-82" aria-hidden="true" tabindex="-1"></a>            <span class="at">split_by =</span> <span class="st">"split"</span>, <span class="at">split_at =</span> <span class="dv">0</span>,</span>
<span id="cb7-83"><a href="#cb7-83" aria-hidden="true" tabindex="-1"></a>            <span class="at">upper_ylab =</span> ulabel,</span>
<span id="cb7-84"><a href="#cb7-84" aria-hidden="true" tabindex="-1"></a>            <span class="at">lower_ylab =</span> llabel,</span>
<span id="cb7-85"><a href="#cb7-85" aria-hidden="true" tabindex="-1"></a>            <span class="at">chr_colors =</span> col)</span>
<span id="cb7-86"><a href="#cb7-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-87"><a href="#cb7-87" aria-hidden="true" tabindex="-1"></a>    <span class="co"># save plot</span></span>
<span id="cb7-88"><a href="#cb7-88" aria-hidden="true" tabindex="-1"></a>    fileName <span class="ot">=</span> <span class="fu">paste0</span>(wd,<span class="st">"/miami_SNPxage_"</span>,i)</span>
<span id="cb7-89"><a href="#cb7-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-90"><a href="#cb7-90" aria-hidden="true" tabindex="-1"></a>    <span class="co"># if INFO filter was used, write into file name</span></span>
<span id="cb7-91"><a href="#cb7-91" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">any</span>(<span class="fu">ls</span>() <span class="sc">==</span> <span class="st">"INFOfilter"</span>)){fileName <span class="ot">=</span> <span class="fu">paste0</span>(fileName,<span class="st">"_INFO"</span>, INFOfilter)}</span>
<span id="cb7-92"><a href="#cb7-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-93"><a href="#cb7-93" aria-hidden="true" tabindex="-1"></a>    <span class="co"># if INFO filter was used, write into file name</span></span>
<span id="cb7-94"><a href="#cb7-94" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">any</span>(<span class="fu">ls</span>() <span class="sc">==</span> <span class="st">"MAFfilter"</span>)){fileName <span class="ot">=</span> <span class="fu">paste0</span>(fileName,<span class="st">"_MAF"</span>, MAFfilter)}</span>
<span id="cb7-95"><a href="#cb7-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-96"><a href="#cb7-96" aria-hidden="true" tabindex="-1"></a>    <span class="co"># if only diallelic SNPs used, say in the name</span></span>
<span id="cb7-97"><a href="#cb7-97" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">any</span>(<span class="fu">ls</span>() <span class="sc">==</span> <span class="st">"diallelic"</span>)){</span>
<span id="cb7-98"><a href="#cb7-98" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(diallelic <span class="sc">==</span> T){</span>
<span id="cb7-99"><a href="#cb7-99" aria-hidden="true" tabindex="-1"></a>            fileName <span class="ot">=</span> <span class="fu">paste0</span>(fileName, <span class="st">"_diallelic"</span>)</span>
<span id="cb7-100"><a href="#cb7-100" aria-hidden="true" tabindex="-1"></a>    }}</span>
<span id="cb7-101"><a href="#cb7-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-102"><a href="#cb7-102" aria-hidden="true" tabindex="-1"></a>    fileName <span class="ot">=</span> <span class="fu">paste0</span>(fileName, <span class="st">"_2.png"</span>)</span>
<span id="cb7-103"><a href="#cb7-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-104"><a href="#cb7-104" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggsave</span>(fileName, <span class="at">plot =</span> p, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">10</span>, <span class="at">dpi =</span> <span class="dv">200</span>)</span>
<span id="cb7-105"><a href="#cb7-105" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/miami_resid_stand_2.png" class="img-fluid figure-img"></p>
</figure>
</div>
</section>
<section id="examine-qq-plots" class="level3">
<h3 class="anchored" data-anchor-id="examine-qq-plots">Examine QQ plots</h3>
<p>This was used to generate Supplementary Plot 33.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="do">### examine why the qqplot seems so inflated</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="co"># plot qq all, MAF filter &amp; INFO filter, hm3 SNPs (MAF &gt; 0.01; INFO &gt; 0.9), 1000G SNPs (MAF &gt; 0.01; INFO &gt; 0.9)</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(qqman)</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(calibrate)</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a><span class="co"># loop through all three traits</span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>all_traits <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"resid_stand"</span>, <span class="st">"ratio_stand"</span>, <span class="st">"diff_stand"</span>)</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(trait_name <span class="cf">in</span> all_traits){</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>    dat <span class="ot">=</span> <span class="fu">fread</span>(<span class="fu">list.files</span>(<span class="at">pattern =</span> <span class="fu">paste0</span>(trait_name, <span class="st">"_N"</span>)), <span class="at">data.table =</span> F)<span class="co">#, nrows = 200000</span></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>    <span class="co">#dat = dat[1:20000,]</span></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>    <span class="co">#dat = dat[dat$CHROM == 22,]</span></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>    <span class="co"># determine color for each of the comp methods</span></span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(trait_name <span class="sc">==</span> <span class="st">"resid_stand"</span>){pretty_name <span class="ot">=</span> <span class="st">"Residual score"</span>}</span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(trait_name <span class="sc">==</span> <span class="st">"ratio_stand"</span>){pretty_name <span class="ot">=</span> <span class="st">"Ratio score"</span>}</span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(trait_name <span class="sc">==</span> <span class="st">"diff_stand"</span>){pretty_name <span class="ot">=</span> <span class="st">"Difference score"</span>}</span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># set up to save plot</span></span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a>    fileName <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"/CCACE_Shared/Anna_F/BrainAtrophy/data/QQeval_"</span>,trait_name,<span class="st">".png"</span>)</span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">png</span>(<span class="at">filename =</span> fileName, <span class="at">width =</span> <span class="dv">1000</span>, <span class="at">height=</span><span class="dv">1000</span>, <span class="at">units=</span><span class="st">"px"</span>)</span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a>        layout_matrix <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">5</span>,<span class="dv">2</span>,<span class="dv">4</span>,<span class="dv">6</span>), <span class="at">ncol=</span><span class="dv">2</span>, <span class="at">nrow=</span><span class="dv">3</span>)</span>
<span id="cb8-28"><a href="#cb8-28" aria-hidden="true" tabindex="-1"></a>        <span class="fu">layout</span>(layout_matrix,  <span class="at">widths =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">1</span><span class="sc">:</span><span class="dv">1</span><span class="sc">:</span><span class="dv">1</span><span class="sc">:</span><span class="dv">1</span><span class="sc">:</span><span class="dv">1</span>) <span class="co">#heights = 1.5:1,</span></span>
<span id="cb8-29"><a href="#cb8-29" aria-hidden="true" tabindex="-1"></a>        <span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">2</span>))</span>
<span id="cb8-30"><a href="#cb8-30" aria-hidden="true" tabindex="-1"></a>        <span class="fu">par</span>(<span class="at">oma =</span> <span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">3</span>))</span>
<span id="cb8-31"><a href="#cb8-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-32"><a href="#cb8-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-33"><a href="#cb8-33" aria-hidden="true" tabindex="-1"></a>    <span class="do">## PLOT 1: all SNPs</span></span>
<span id="cb8-34"><a href="#cb8-34" aria-hidden="true" tabindex="-1"></a>    <span class="do">### Calculate Genomic Inflation factor</span></span>
<span id="cb8-35"><a href="#cb8-35" aria-hidden="true" tabindex="-1"></a>    chisq <span class="ot">&lt;-</span> <span class="fu">qchisq</span>(<span class="dv">1</span><span class="sc">-</span>dat<span class="sc">$</span>P,<span class="dv">1</span>)</span>
<span id="cb8-36"><a href="#cb8-36" aria-hidden="true" tabindex="-1"></a>    lambda <span class="ot">&lt;-</span> <span class="fu">median</span>(chisq)<span class="sc">/</span><span class="fu">qchisq</span>(<span class="fl">0.5</span>,<span class="dv">1</span>)</span>
<span id="cb8-37"><a href="#cb8-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-38"><a href="#cb8-38" aria-hidden="true" tabindex="-1"></a>    <span class="co"># make qq plot</span></span>
<span id="cb8-39"><a href="#cb8-39" aria-hidden="true" tabindex="-1"></a>    main <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"All SNPs: "</span>, pretty_name, <span class="st">" (Lambda = "</span>, <span class="fu">round</span>(lambda, <span class="at">digits =</span> <span class="dv">3</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>, <span class="fu">nrow</span>(dat), <span class="st">" SNPs"</span>)</span>
<span id="cb8-40"><a href="#cb8-40" aria-hidden="true" tabindex="-1"></a>    p1 <span class="ot">&lt;-</span> <span class="fu">qq</span>(dat<span class="sc">$</span>P, <span class="at">main =</span> main, <span class="at">cex =</span> <span class="dv">1</span>, <span class="at">cex.axis =</span> <span class="fl">1.5</span>)</span>
<span id="cb8-41"><a href="#cb8-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-42"><a href="#cb8-42" aria-hidden="true" tabindex="-1"></a>    <span class="do">###############################</span></span>
<span id="cb8-43"><a href="#cb8-43" aria-hidden="true" tabindex="-1"></a>    <span class="do">## PLOT 2: Apply INFO and MAF filter</span></span>
<span id="cb8-44"><a href="#cb8-44" aria-hidden="true" tabindex="-1"></a>    dat2 <span class="ot">=</span> dat[dat<span class="sc">$</span>MAF <span class="sc">&gt;</span> <span class="fl">0.01</span>,]</span>
<span id="cb8-45"><a href="#cb8-45" aria-hidden="true" tabindex="-1"></a>    dat2 <span class="ot">=</span> dat2[dat2<span class="sc">$</span>INFO <span class="sc">&gt;</span> <span class="fl">0.8</span>,]</span>
<span id="cb8-46"><a href="#cb8-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-47"><a href="#cb8-47" aria-hidden="true" tabindex="-1"></a>    <span class="do">### Calculate Genomic Inflation factor</span></span>
<span id="cb8-48"><a href="#cb8-48" aria-hidden="true" tabindex="-1"></a>    chisq <span class="ot">&lt;-</span> <span class="fu">qchisq</span>(<span class="dv">1</span><span class="sc">-</span>dat2<span class="sc">$</span>P,<span class="dv">1</span>)</span>
<span id="cb8-49"><a href="#cb8-49" aria-hidden="true" tabindex="-1"></a>    lambda <span class="ot">&lt;-</span> <span class="fu">median</span>(chisq)<span class="sc">/</span><span class="fu">qchisq</span>(<span class="fl">0.5</span>,<span class="dv">1</span>)</span>
<span id="cb8-50"><a href="#cb8-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-51"><a href="#cb8-51" aria-hidden="true" tabindex="-1"></a>    <span class="co"># make qq plot</span></span>
<span id="cb8-52"><a href="#cb8-52" aria-hidden="true" tabindex="-1"></a>    main <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"SNPs passing INFO &gt; 0.9 &amp; MAF &gt; 0.01 filters:</span><span class="sc">\n</span><span class="st">"</span>, pretty_name, <span class="st">" (Lambda = "</span>, <span class="fu">round</span>(lambda, <span class="at">digits =</span> <span class="dv">3</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>, <span class="fu">nrow</span>(dat2), <span class="st">" SNPs"</span>)</span>
<span id="cb8-53"><a href="#cb8-53" aria-hidden="true" tabindex="-1"></a>    p2 <span class="ot">&lt;-</span> <span class="fu">qq</span>(dat2<span class="sc">$</span>P, <span class="at">main =</span> main, <span class="at">cex =</span> <span class="dv">1</span>, <span class="at">cex.axis =</span> <span class="fl">1.5</span>)</span>
<span id="cb8-54"><a href="#cb8-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-55"><a href="#cb8-55" aria-hidden="true" tabindex="-1"></a>    <span class="do">###############################</span></span>
<span id="cb8-56"><a href="#cb8-56" aria-hidden="true" tabindex="-1"></a>    <span class="do">## PLOT 3: HapMap3 SNPs only</span></span>
<span id="cb8-57"><a href="#cb8-57" aria-hidden="true" tabindex="-1"></a>    hm3 <span class="ot">&lt;-</span> <span class="st">"eur_w_ld_chr/w_hm3.snplist"</span></span>
<span id="cb8-58"><a href="#cb8-58" aria-hidden="true" tabindex="-1"></a>    hm3SNPs <span class="ot">&lt;-</span> <span class="fu">fread</span>(hm3)</span>
<span id="cb8-59"><a href="#cb8-59" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(hm3SNPs) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ID"</span>, <span class="st">"ALLELE1"</span>, <span class="st">"ALLELE0"</span>)</span>
<span id="cb8-60"><a href="#cb8-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-61"><a href="#cb8-61" aria-hidden="true" tabindex="-1"></a>    <span class="co"># merge the two data sets and only keep overlapping SNPS that agree in A1 and A2</span></span>
<span id="cb8-62"><a href="#cb8-62" aria-hidden="true" tabindex="-1"></a>    dat2 <span class="ot">&lt;-</span> <span class="fu">merge</span>(dat, hm3SNPs, <span class="at">by =</span> <span class="st">"ID"</span>)</span>
<span id="cb8-63"><a href="#cb8-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-64"><a href="#cb8-64" aria-hidden="true" tabindex="-1"></a>    <span class="do">### Calculate Genomic Inflation factor</span></span>
<span id="cb8-65"><a href="#cb8-65" aria-hidden="true" tabindex="-1"></a>    chisq <span class="ot">&lt;-</span> <span class="fu">qchisq</span>(<span class="dv">1</span><span class="sc">-</span>dat2<span class="sc">$</span>P,<span class="dv">1</span>)</span>
<span id="cb8-66"><a href="#cb8-66" aria-hidden="true" tabindex="-1"></a>    lambda <span class="ot">&lt;-</span> <span class="fu">median</span>(chisq)<span class="sc">/</span><span class="fu">qchisq</span>(<span class="fl">0.5</span>,<span class="dv">1</span>)</span>
<span id="cb8-67"><a href="#cb8-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-68"><a href="#cb8-68" aria-hidden="true" tabindex="-1"></a>    <span class="co"># make qq plot</span></span>
<span id="cb8-69"><a href="#cb8-69" aria-hidden="true" tabindex="-1"></a>    main <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"HapMap3 SNPs: "</span>, pretty_name, <span class="st">" (Lambda = "</span>, <span class="fu">round</span>(lambda, <span class="at">digits =</span> <span class="dv">3</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>, <span class="fu">nrow</span>(dat2), <span class="st">" SNPs"</span>)</span>
<span id="cb8-70"><a href="#cb8-70" aria-hidden="true" tabindex="-1"></a>    p3 <span class="ot">&lt;-</span> <span class="fu">qq</span>(dat2<span class="sc">$</span>P, <span class="at">main =</span> main, <span class="at">cex =</span> <span class="dv">1</span>, <span class="at">cex.axis =</span> <span class="fl">1.5</span>)</span>
<span id="cb8-71"><a href="#cb8-71" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb8-72"><a href="#cb8-72" aria-hidden="true" tabindex="-1"></a>    <span class="do">###############################</span></span>
<span id="cb8-73"><a href="#cb8-73" aria-hidden="true" tabindex="-1"></a>    <span class="do">## PLOT 4: 1000 Genomes SNPs only</span></span>
<span id="cb8-74"><a href="#cb8-74" aria-hidden="true" tabindex="-1"></a>    <span class="co"># different reference files: this one is from FUMA website</span></span>
<span id="cb8-75"><a href="#cb8-75" aria-hidden="true" tabindex="-1"></a>    ref <span class="ot">=</span> <span class="st">"1KGphase3EURvariants.txt.gz"</span></span>
<span id="cb8-76"><a href="#cb8-76" aria-hidden="true" tabindex="-1"></a>    ref1000 <span class="ot">&lt;-</span> <span class="fu">fread</span>(ref, <span class="at">select =</span> <span class="fu">c</span>(<span class="st">"SNP"</span>, <span class="st">"A1"</span>, <span class="st">"A2"</span>))</span>
<span id="cb8-77"><a href="#cb8-77" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(ref1000) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ID"</span>, <span class="st">"ALLELE1"</span>, <span class="st">"ALLELE0"</span>)</span>
<span id="cb8-78"><a href="#cb8-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-79"><a href="#cb8-79" aria-hidden="true" tabindex="-1"></a>    <span class="co"># merge the two data sets and only keep overlapping SNPS that agree in A1 and A2</span></span>
<span id="cb8-80"><a href="#cb8-80" aria-hidden="true" tabindex="-1"></a>    dat2 <span class="ot">&lt;-</span> <span class="fu">merge</span>(dat, ref1000, <span class="at">by =</span> <span class="st">"ID"</span>)</span>
<span id="cb8-81"><a href="#cb8-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-82"><a href="#cb8-82" aria-hidden="true" tabindex="-1"></a>    <span class="do">### Calculate Genomic Inflation factor</span></span>
<span id="cb8-83"><a href="#cb8-83" aria-hidden="true" tabindex="-1"></a>    chisq <span class="ot">&lt;-</span> <span class="fu">qchisq</span>(<span class="dv">1</span><span class="sc">-</span>dat2<span class="sc">$</span>P,<span class="dv">1</span>)</span>
<span id="cb8-84"><a href="#cb8-84" aria-hidden="true" tabindex="-1"></a>    lambda <span class="ot">&lt;-</span> <span class="fu">median</span>(chisq)<span class="sc">/</span><span class="fu">qchisq</span>(<span class="fl">0.5</span>,<span class="dv">1</span>)</span>
<span id="cb8-85"><a href="#cb8-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-86"><a href="#cb8-86" aria-hidden="true" tabindex="-1"></a>    <span class="co"># make qq plot</span></span>
<span id="cb8-87"><a href="#cb8-87" aria-hidden="true" tabindex="-1"></a>    main <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"1000 Genomes SNPs: "</span>, pretty_name, <span class="st">" (Lambda = "</span>, <span class="fu">round</span>(lambda, <span class="at">digits =</span> <span class="dv">3</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>, <span class="fu">nrow</span>(dat2), <span class="st">" SNPs"</span>)</span>
<span id="cb8-88"><a href="#cb8-88" aria-hidden="true" tabindex="-1"></a>    p4 <span class="ot">&lt;-</span> <span class="fu">qq</span>(dat2<span class="sc">$</span>P, <span class="at">main =</span> main, <span class="at">cex =</span> <span class="dv">1</span>, <span class="at">cex.axis =</span> <span class="fl">1.5</span>)</span>
<span id="cb8-89"><a href="#cb8-89" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb8-90"><a href="#cb8-90" aria-hidden="true" tabindex="-1"></a>    <span class="do">###############################</span></span>
<span id="cb8-91"><a href="#cb8-91" aria-hidden="true" tabindex="-1"></a>    <span class="do">## PLOT 5: HapMap3 SNPs removed</span></span>
<span id="cb8-92"><a href="#cb8-92" aria-hidden="true" tabindex="-1"></a>    <span class="co"># isolate SNPs not shared between data and hm3</span></span>
<span id="cb8-93"><a href="#cb8-93" aria-hidden="true" tabindex="-1"></a>    notHm3 <span class="ot">=</span> <span class="fu">setdiff</span>(dat<span class="sc">$</span>ID, hm3SNPs<span class="sc">$</span>ID)</span>
<span id="cb8-94"><a href="#cb8-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-95"><a href="#cb8-95" aria-hidden="true" tabindex="-1"></a>    <span class="co"># only retain those </span></span>
<span id="cb8-96"><a href="#cb8-96" aria-hidden="true" tabindex="-1"></a>    dat2 <span class="ot">&lt;-</span> dat[dat<span class="sc">$</span>ID <span class="sc">%in%</span> notHm3,]</span>
<span id="cb8-97"><a href="#cb8-97" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb8-98"><a href="#cb8-98" aria-hidden="true" tabindex="-1"></a>    <span class="do">### Calculate Genomic Inflation factor</span></span>
<span id="cb8-99"><a href="#cb8-99" aria-hidden="true" tabindex="-1"></a>    chisq <span class="ot">&lt;-</span> <span class="fu">qchisq</span>(<span class="dv">1</span><span class="sc">-</span>dat2<span class="sc">$</span>P,<span class="dv">1</span>)</span>
<span id="cb8-100"><a href="#cb8-100" aria-hidden="true" tabindex="-1"></a>    lambda <span class="ot">&lt;-</span> <span class="fu">median</span>(chisq)<span class="sc">/</span><span class="fu">qchisq</span>(<span class="fl">0.5</span>,<span class="dv">1</span>)</span>
<span id="cb8-101"><a href="#cb8-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-102"><a href="#cb8-102" aria-hidden="true" tabindex="-1"></a>    <span class="co"># make qq plot</span></span>
<span id="cb8-103"><a href="#cb8-103" aria-hidden="true" tabindex="-1"></a>    main <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"All SNPs that are not HapMap3: "</span>, pretty_name, <span class="st">" (Lambda = "</span>, <span class="fu">round</span>(lambda, <span class="at">digits =</span> <span class="dv">3</span>), <span class="st">")</span><span class="sc">\n</span><span class="st">"</span>, <span class="fu">nrow</span>(dat2), <span class="st">" SNPs"</span>)</span>
<span id="cb8-104"><a href="#cb8-104" aria-hidden="true" tabindex="-1"></a>    p5 <span class="ot">&lt;-</span> <span class="fu">qq</span>(dat2<span class="sc">$</span>P, <span class="at">main =</span> main, <span class="at">cex =</span> <span class="dv">1</span>, <span class="at">cex.axis =</span> <span class="fl">1.5</span>)</span>
<span id="cb8-105"><a href="#cb8-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-106"><a href="#cb8-106" aria-hidden="true" tabindex="-1"></a>    <span class="co"># save</span></span>
<span id="cb8-107"><a href="#cb8-107" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dev.off</span>()</span>
<span id="cb8-108"><a href="#cb8-108" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/QQeval_resid_stand.png" class="img-fluid figure-img"></p>
</figure>
</div>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>