<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.189">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Anna Elisabeth Furtjes">
<meta name="dcterms.date" content="2024-11-14">

<title>Brain Atrophy Genetics - Genetic correlations</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Brain Atrophy Genetics</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html">Overview</a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#genetic-correlations-in-genomicsem" id="toc-genetic-correlations-in-genomicsem" class="nav-link active" data-scroll-target="#genetic-correlations-in-genomicsem">Genetic correlations in GenomicSEM</a>
  <ul class="collapse">
  <li><a href="#munge" id="toc-munge" class="nav-link" data-scroll-target="#munge">Munge</a></li>
  <li><a href="#genetic-correlations" id="toc-genetic-correlations" class="nav-link" data-scroll-target="#genetic-correlations">Genetic correlations</a></li>
  <li><a href="#plot-results-full-sample" id="toc-plot-results-full-sample" class="nav-link" data-scroll-target="#plot-results-full-sample">Plot results (full sample)</a></li>
  <li><a href="#plot-sex-split-correlations" id="toc-plot-sex-split-correlations" class="nav-link" data-scroll-target="#plot-sex-split-correlations">Plot sex-split correlations</a></li>
  </ul></li>
  <li><a href="#gwas-by-subtraction-in-genomicsem" id="toc-gwas-by-subtraction-in-genomicsem" class="nav-link" data-scroll-target="#gwas-by-subtraction-in-genomicsem">GWAS-by-subtraction in GenomicSEM</a>
  <ul class="collapse">
  <li><a href="#snp-level-analysis" id="toc-snp-level-analysis" class="nav-link" data-scroll-target="#snp-level-analysis">SNP-level analysis</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Genetic correlations</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Anna Elisabeth Furtjes </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">November 14, 2024</p>
    </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Doi</div>
    <div class="quarto-title-meta-contents">
      <p class="doi">
        <a href="https://doi.org/10.1101/2024.11.06.622274">10.1101/2024.11.06.622274</a>
      </p>
    </div>
  </div>
  </div>
  

</header>

<hr>
<section id="genetic-correlations-in-genomicsem" class="level2">
<h2 class="anchored" data-anchor-id="genetic-correlations-in-genomicsem">Genetic correlations in GenomicSEM</h2>
<section id="munge" class="level3">
<h3 class="anchored" data-anchor-id="munge">Munge</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co"># install_github("GenomicSEM/GenomicSEM")</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomicSEM)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="do">#######################################################################</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="do">### MUNGE</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="do">#######################################################################</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">pattern =</span> <span class="st">"GWAS_brainAtrophy_"</span>)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="co">#files &lt;- files[grepl("males", files)]</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>hm3 <span class="ot">&lt;-</span> <span class="st">"/eur_w_ld_chr/w_hm3.snplist"</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>maf.filter <span class="ot">=</span> <span class="fl">0.01</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>info.filter <span class="ot">=</span> <span class="fl">0.9</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="co"># cycle through each of the files</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> files){</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>    <span class="co"># read in file first</span></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>    file <span class="ot">&lt;-</span> <span class="fu">fread</span>(i)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># if it's residual or ratio score, we flip the effect sizes because that was also done throughout the rest of manuscript to harmonise the scores (higher score, more atrophy)</span></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">grepl</span>(<span class="st">"ratio|resid"</span>, i)){</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>        file<span class="sc">$</span>BETA <span class="ot">&lt;-</span> file<span class="sc">$</span>BETA <span class="sc">*</span> (<span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>    <span class="co"># write temporarily because munge can't read in gz files</span></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">fwrite</span>(file, <span class="fu">paste0</span>(<span class="st">"temp"</span>,i,<span class="st">".txt"</span>), <span class="at">na =</span> <span class="st">"NA"</span>, <span class="at">quote =</span> F, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>    <span class="co"># isolate trait name</span></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>    trait.name <span class="ot">=</span> <span class="fu">str_remove</span>(i, <span class="st">"GWAS_brainAtrophy_"</span>)</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>    trait.name <span class="ot">=</span> <span class="fu">str_remove</span>(trait.name, <span class="st">"_N43110.gz"</span>)</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>    <span class="co"># munge this file</span></span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">munge</span>(<span class="at">files =</span> <span class="fu">paste0</span>(<span class="st">"temp"</span>,i,<span class="st">".txt"</span>), </span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>            <span class="at">trait.names =</span> trait.name,</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a>            <span class="at">hm3 =</span> hm3, </span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a>            <span class="at">maf.filter =</span> maf.filter, </span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a>            <span class="at">info.filter =</span> info.filter,</span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a>            <span class="at">log.name =</span> <span class="fu">paste0</span>(trait.name),</span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a>            <span class="at">column.names =</span> <span class="fu">list</span>(<span class="at">SNP =</span> <span class="st">"ID"</span>, </span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a>                                <span class="at">A1 =</span> <span class="st">"ALLELE1"</span>,</span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a>                                <span class="at">A2 =</span> <span class="st">"ALLELE0"</span>, </span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a>                                <span class="at">effect =</span> <span class="st">"BETA"</span>, </span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a>                                <span class="at">P =</span> <span class="st">"P"</span>, </span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a>                                <span class="at">N =</span> <span class="st">"N"</span>))</span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a>    <span class="co"># remove temp file again</span></span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a>    <span class="fu">file.remove</span>(<span class="fu">paste0</span>(<span class="st">"temp"</span>,i,<span class="st">".txt"</span>))</span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a><span class="do">############################################################</span></span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a><span class="do">#### Munging difficult sumstats</span></span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a><span class="do">############################################################</span></span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomicSEM)</span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"/CCACE_Shared/Anna_F/BrainAtrophy/data/GWASsumstats"</span>)</span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a>hm3 <span class="ot">&lt;-</span> <span class="st">"/CCACE_Shared/Anna_F/BrainAtrophy/eur_w_ld_chr/w_hm3.snplist"</span></span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a>maf.filter <span class="ot">=</span> <span class="fl">0.01</span></span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a>info.filter <span class="ot">=</span> <span class="fl">0.9</span></span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a>i<span class="ot">=</span><span class="fu">list.files</span>(<span class="at">pattern =</span> <span class="st">"CHARGE"</span>)</span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a>file <span class="ot">&lt;-</span> <span class="fu">fread</span>(i, <span class="at">fill=</span><span class="cn">TRUE</span>)</span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a><span class="co"># separate fused column</span></span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a>file <span class="ot">&lt;-</span> <span class="fu">separate</span>(file, <span class="at">col =</span> <span class="st">"N RSNUMBERS"</span>, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"N"</span>,<span class="st">"RSNUMBER"</span>), <span class="at">sep =</span> <span class="st">" "</span>)</span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a><span class="co"># some rows are empty and the entry has shifted to to column V13</span></span>
<span id="cb1-71"><a href="#cb1-71" aria-hidden="true" tabindex="-1"></a>file<span class="sc">$</span>RSNUMBER <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(file<span class="sc">$</span>RSNUMBER), file<span class="sc">$</span>V13, file<span class="sc">$</span>RSNUMBER)</span>
<span id="cb1-72"><a href="#cb1-72" aria-hidden="true" tabindex="-1"></a><span class="co"># problem with p-value column - rename</span></span>
<span id="cb1-73"><a href="#cb1-73" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(file)[<span class="fu">grep</span>(<span class="st">"P-"</span>, <span class="fu">names</span>(file))] <span class="ot">&lt;-</span> <span class="st">"P"</span></span>
<span id="cb1-74"><a href="#cb1-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-75"><a href="#cb1-75" aria-hidden="true" tabindex="-1"></a><span class="co"># write temporarily because munge can't read in gz files</span></span>
<span id="cb1-76"><a href="#cb1-76" aria-hidden="true" tabindex="-1"></a><span class="fu">fwrite</span>(file, <span class="fu">paste0</span>(<span class="st">"temp"</span>,i,<span class="st">".txt"</span>), <span class="at">na =</span> <span class="st">"NA"</span>, <span class="at">quote =</span> F, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-77"><a href="#cb1-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-78"><a href="#cb1-78" aria-hidden="true" tabindex="-1"></a><span class="co"># isolate trait name</span></span>
<span id="cb1-79"><a href="#cb1-79" aria-hidden="true" tabindex="-1"></a>trait.name <span class="ot">=</span> <span class="st">"ICV_Adams.et.al"</span></span>
<span id="cb1-80"><a href="#cb1-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-81"><a href="#cb1-81" aria-hidden="true" tabindex="-1"></a><span class="co"># munge this file</span></span>
<span id="cb1-82"><a href="#cb1-82" aria-hidden="true" tabindex="-1"></a><span class="fu">munge</span>(<span class="at">files =</span> <span class="fu">paste0</span>(<span class="st">"temp"</span>,i,<span class="st">".txt"</span>), </span>
<span id="cb1-83"><a href="#cb1-83" aria-hidden="true" tabindex="-1"></a>        <span class="at">trait.names =</span> trait.name,</span>
<span id="cb1-84"><a href="#cb1-84" aria-hidden="true" tabindex="-1"></a>        <span class="at">hm3 =</span> hm3, </span>
<span id="cb1-85"><a href="#cb1-85" aria-hidden="true" tabindex="-1"></a>        <span class="at">maf.filter =</span> maf.filter, </span>
<span id="cb1-86"><a href="#cb1-86" aria-hidden="true" tabindex="-1"></a>        <span class="at">info.filter =</span> info.filter,</span>
<span id="cb1-87"><a href="#cb1-87" aria-hidden="true" tabindex="-1"></a>        <span class="at">log.name =</span> <span class="fu">paste0</span>(trait.name),</span>
<span id="cb1-88"><a href="#cb1-88" aria-hidden="true" tabindex="-1"></a>        <span class="at">column.names =</span> <span class="fu">list</span>(<span class="at">SNP =</span> <span class="st">"RSNUMBER"</span>, </span>
<span id="cb1-89"><a href="#cb1-89" aria-hidden="true" tabindex="-1"></a>                            <span class="at">A1 =</span> <span class="st">"Allele1"</span>,</span>
<span id="cb1-90"><a href="#cb1-90" aria-hidden="true" tabindex="-1"></a>                            <span class="at">A2 =</span> <span class="st">"Allele2"</span>, </span>
<span id="cb1-91"><a href="#cb1-91" aria-hidden="true" tabindex="-1"></a>                            <span class="at">Z =</span> <span class="st">"Zscore"</span>, </span>
<span id="cb1-92"><a href="#cb1-92" aria-hidden="true" tabindex="-1"></a>                            <span class="at">P =</span> <span class="st">"P"</span>, </span>
<span id="cb1-93"><a href="#cb1-93" aria-hidden="true" tabindex="-1"></a>                            <span class="at">N =</span> <span class="st">"Weight"</span>))</span>
<span id="cb1-94"><a href="#cb1-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-95"><a href="#cb1-95" aria-hidden="true" tabindex="-1"></a><span class="co"># remove temp file again</span></span>
<span id="cb1-96"><a href="#cb1-96" aria-hidden="true" tabindex="-1"></a><span class="fu">file.remove</span>(<span class="fu">paste0</span>(<span class="st">"temp"</span>,i,<span class="st">".txt"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="genetic-correlations" class="level3">
<h3 class="anchored" data-anchor-id="genetic-correlations">Genetic correlations</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="do">#######################################################################</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="do">### LDSC</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="do">#######################################################################</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomicSEM)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="co"># vector of munged sumstats</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>traits <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">pattern =</span> <span class="st">".sumstats.gz"</span>)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="co"># remove log files</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>traits <span class="ot">&lt;-</span> traits[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">".log"</span>, traits)]</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="co"># remove BrainChange GWAS because they produce negative h2 estimate</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>traits <span class="ot">=</span> traits[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"BrainChange"</span>, traits)]</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a><span class="co"># remove male and female specific sumstats</span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>traits <span class="ot">=</span> traits[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"males"</span>, traits)]</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a><span class="co"># remove SNP-by-age analyses</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>traits <span class="ot">=</span> traits[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"SNPxage"</span>, traits)]</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a><span class="co"># trait.names</span></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>trait.names <span class="ot">&lt;-</span> <span class="fu">str_remove</span>(traits, <span class="at">pattern =</span> <span class="st">".sumstats.gz"</span>)</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a><span class="co"># folder with ld scores</span></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>ld <span class="ot">=</span> <span class="st">"eur_w_ld_chr/"</span></span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a><span class="co"># folder with ld weights</span></span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>wld <span class="ot">=</span> <span class="st">"eur_w_ld_chr/"</span></span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a><span class="co"># run ldsc</span></span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>LDSCoutput <span class="ot">&lt;-</span> <span class="fu">ldsc</span>(<span class="at">traits =</span> traits, </span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ld =</span> ld, <span class="at">wld =</span> wld, </span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>                    <span class="at">sample.prev =</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(traits)),</span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>                    <span class="at">population.prev =</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(traits)),</span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>                    <span class="at">trait.names =</span> trait.names,</span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>                    <span class="at">stand =</span> T)</span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a><span class="co"># dimnames</span></span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(LDSCoutput<span class="sc">$</span>S)[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="fu">dimnames</span>(LDSCoutput<span class="sc">$</span>S)[[<span class="dv">2</span>]]</span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(LDSCoutput<span class="sc">$</span>S_Stand)[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="fu">dimnames</span>(LDSCoutput<span class="sc">$</span>S)[[<span class="dv">2</span>]]</span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(LDSCoutput<span class="sc">$</span>S_Stand)[[<span class="dv">2</span>]] <span class="ot">&lt;-</span> <span class="fu">dimnames</span>(LDSCoutput<span class="sc">$</span>S)[[<span class="dv">2</span>]]</span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(LDSCoutput<span class="sc">$</span>I)[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="fu">dimnames</span>(LDSCoutput<span class="sc">$</span>S)[[<span class="dv">2</span>]]</span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(LDSCoutput<span class="sc">$</span>I)[[<span class="dv">2</span>]] <span class="ot">&lt;-</span> <span class="fu">dimnames</span>(LDSCoutput<span class="sc">$</span>S)[[<span class="dv">2</span>]]</span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(LDSCoutput, <span class="at">file =</span> <span class="st">"LDSCoutput_neurodegenrative.RData"</span>)</span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a><span class="co">#save(LDSCoutput, file = "LDSCoutput_sexsplit.RData")</span></span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a><span class="do">###########################################################################</span></span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a><span class="do">#### Calculate associated matrix of Z statistics for this rG matrix</span></span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a><span class="do">###########################################################################</span></span>
<span id="cb2-47"><a href="#cb2-47" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gdata)</span>
<span id="cb2-48"><a href="#cb2-48" aria-hidden="true" tabindex="-1"></a>S_LD<span class="ot">&lt;-</span>LDSCoutput<span class="sc">$</span>S</span>
<span id="cb2-49"><a href="#cb2-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-50"><a href="#cb2-50" aria-hidden="true" tabindex="-1"></a><span class="co">#pull V: the sampling covariance matrix</span></span>
<span id="cb2-51"><a href="#cb2-51" aria-hidden="true" tabindex="-1"></a>V_LD<span class="ot">&lt;-</span>LDSCoutput<span class="sc">$</span>V</span>
<span id="cb2-52"><a href="#cb2-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-53"><a href="#cb2-53" aria-hidden="true" tabindex="-1"></a><span class="co">#standardize S [equivalent of cov2cor]</span></span>
<span id="cb2-54"><a href="#cb2-54" aria-hidden="true" tabindex="-1"></a>D<span class="ot">=</span><span class="fu">sqrt</span>(<span class="fu">diag</span>(<span class="fu">diag</span>(S_LD)))</span>
<span id="cb2-55"><a href="#cb2-55" aria-hidden="true" tabindex="-1"></a>S_Stand<span class="ot">=</span><span class="fu">solve</span>(D)<span class="sc">%*%</span>S_LD<span class="sc">%*%</span><span class="fu">solve</span>(D)</span>
<span id="cb2-56"><a href="#cb2-56" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(S_Stand)<span class="ot">&lt;-</span><span class="fu">rownames</span>(S_LD)</span>
<span id="cb2-57"><a href="#cb2-57" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(S_Stand)<span class="ot">&lt;-</span><span class="fu">colnames</span>(S_Stand)</span>
<span id="cb2-58"><a href="#cb2-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-59"><a href="#cb2-59" aria-hidden="true" tabindex="-1"></a><span class="co">#obtain diagonals of the original V matrix and take their sqrt to get SE's</span></span>
<span id="cb2-60"><a href="#cb2-60" aria-hidden="true" tabindex="-1"></a>Dvcov<span class="ot">&lt;-</span><span class="fu">sqrt</span>(<span class="fu">diag</span>(V_LD))</span>
<span id="cb2-61"><a href="#cb2-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-62"><a href="#cb2-62" aria-hidden="true" tabindex="-1"></a><span class="co">#calculate the ratio of the rescaled and original S matrices</span></span>
<span id="cb2-63"><a href="#cb2-63" aria-hidden="true" tabindex="-1"></a>scaleO<span class="ot">=</span><span class="fu">as.vector</span>(<span class="fu">lowerTriangle</span>((S_Stand<span class="sc">/</span>S_LD),<span class="at">diag=</span>T))</span>
<span id="cb2-64"><a href="#cb2-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-65"><a href="#cb2-65" aria-hidden="true" tabindex="-1"></a><span class="do">## Make sure that if ratio in NaN (devision by zero) we put the zero back in</span></span>
<span id="cb2-66"><a href="#cb2-66" aria-hidden="true" tabindex="-1"></a>scaleO[<span class="fu">is.nan</span>(scaleO)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb2-67"><a href="#cb2-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-68"><a href="#cb2-68" aria-hidden="true" tabindex="-1"></a><span class="co">#rescale the SEs by the same multiples that the S matrix was rescaled by</span></span>
<span id="cb2-69"><a href="#cb2-69" aria-hidden="true" tabindex="-1"></a>Dvcovl<span class="ot">&lt;-</span><span class="fu">as.vector</span>(Dvcov<span class="sc">*</span><span class="fu">t</span>(scaleO))</span>
<span id="cb2-70"><a href="#cb2-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-71"><a href="#cb2-71" aria-hidden="true" tabindex="-1"></a><span class="co">#obtain the sampling correlation matrix by standardizing the original V matrix</span></span>
<span id="cb2-72"><a href="#cb2-72" aria-hidden="true" tabindex="-1"></a>Vcor<span class="ot">&lt;-</span><span class="fu">cov2cor</span>(V_LD)</span>
<span id="cb2-73"><a href="#cb2-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-74"><a href="#cb2-74" aria-hidden="true" tabindex="-1"></a><span class="co">#rescale the sampling correlation matrix by the appropriate diagonals</span></span>
<span id="cb2-75"><a href="#cb2-75" aria-hidden="true" tabindex="-1"></a>V_stand<span class="ot">&lt;-</span><span class="fu">diag</span>(Dvcovl)<span class="sc">%*%</span>Vcor<span class="sc">%*%</span><span class="fu">diag</span>(Dvcovl)</span>
<span id="cb2-76"><a href="#cb2-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-77"><a href="#cb2-77" aria-hidden="true" tabindex="-1"></a><span class="co">#calculate SEs of rg</span></span>
<span id="cb2-78"><a href="#cb2-78" aria-hidden="true" tabindex="-1"></a>k<span class="ot">&lt;-</span><span class="fu">nrow</span>(LDSCoutput<span class="sc">$</span>S)</span>
<span id="cb2-79"><a href="#cb2-79" aria-hidden="true" tabindex="-1"></a>SE_stand<span class="ot">&lt;-</span><span class="fu">matrix</span>(<span class="dv">0</span>, k, k)</span>
<span id="cb2-80"><a href="#cb2-80" aria-hidden="true" tabindex="-1"></a>SE_stand[<span class="fu">lower.tri</span>(SE_stand,<span class="at">diag=</span><span class="cn">TRUE</span>)] <span class="ot">&lt;-</span><span class="fu">sqrt</span>(<span class="fu">diag</span>(V_stand))</span>
<span id="cb2-81"><a href="#cb2-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-82"><a href="#cb2-82" aria-hidden="true" tabindex="-1"></a><span class="co">#calculate Z of rg matrix</span></span>
<span id="cb2-83"><a href="#cb2-83" aria-hidden="true" tabindex="-1"></a>Z_cor<span class="ot">&lt;-</span>S_Stand<span class="sc">/</span>SE_stand</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="plot-results-full-sample" class="level3">
<h3 class="anchored" data-anchor-id="plot-results-full-sample">Plot results (full sample)</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="do">###########################################################################</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="do">#### Genetic correlation matrix</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="do">###########################################################################</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="do">### INCLUDE NEURODEGENERATIVE DISEASES</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"LDSCoutput_neurodegenrative.RData"</span>)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="co"># pull out correlation matrix</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>cormat <span class="ot">&lt;-</span> LDSCoutput<span class="sc">$</span>S_Stand</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="co"># re-roder</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>cormat <span class="ot">=</span> cormat[<span class="fu">c</span>(<span class="st">"ALZ"</span>, <span class="st">"ALZ_2019"</span>, <span class="st">"PD"</span>, <span class="st">"ALS"</span>, <span class="st">"ICV_Adams.et.al"</span>, <span class="st">"TBV_Zhao.et.al"</span>, <span class="st">"TBV_Smith.et.al"</span>, <span class="st">"PC1_whole_brain_greymatter"</span>, <span class="st">"brainage"</span>, <span class="st">"ICVstand"</span>, <span class="st">"TBVstand"</span>, <span class="st">"CSFstand"</span>, <span class="st">"diff_stand"</span>, <span class="st">"ratio_stand"</span>, <span class="st">"resid_stand"</span>), <span class="fu">c</span>(<span class="st">"ALZ"</span>, <span class="st">"ALZ_2019"</span>, <span class="st">"PD"</span>, <span class="st">"ALS"</span>, <span class="st">"ICV_Adams.et.al"</span>, <span class="st">"TBV_Zhao.et.al"</span>, <span class="st">"TBV_Smith.et.al"</span>, <span class="st">"PC1_whole_brain_greymatter"</span>, <span class="st">"brainage"</span>, <span class="st">"ICVstand"</span>, <span class="st">"TBVstand"</span>, <span class="st">"CSFstand"</span>, <span class="st">"diff_stand"</span>, <span class="st">"ratio_stand"</span>, <span class="st">"resid_stand"</span>)]</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a><span class="co"># re-name</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(cormat)[[<span class="dv">1</span>]] <span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">"Alzheimer's &amp; related dementias</span><span class="sc">\n</span><span class="st">(Bellenguez et al., 2021)"</span>, <span class="st">"Alzheimer's</span><span class="sc">\n</span><span class="st">(Kunkle et al., 2019)"</span>, <span class="st">"Parkinson's</span><span class="sc">\n</span><span class="st">(Nalls et al., 2019)"</span>, <span class="st">"Amyotrophic lateral sclerosis</span><span class="sc">\n</span><span class="st">(van Rheenen et al. 2021)"</span>, <span class="st">"ICV</span><span class="sc">\n</span><span class="st">(Adams et al., 2016)"</span>, <span class="st">"TBV</span><span class="sc">\n</span><span class="st">(Zhao et al., 2018)"</span>, <span class="st">"TBV</span><span class="sc">\n</span><span class="st">(Smith et al., 2021)"</span>, <span class="st">"Grey matter whole brain PC1</span><span class="sc">\n</span><span class="st">(Furtjes et al., 2023)"</span>, <span class="st">"Brain age</span><span class="sc">\n</span><span class="st">(Kaufmann et al., 2019)"</span>,<span class="st">"ICV"</span>, <span class="st">"TBV"</span>, <span class="st">"CSF"</span>, <span class="st">"Difference score</span><span class="sc">\n</span><span class="st">(cross)"</span>, <span class="st">"Ratio score</span><span class="sc">\n</span><span class="st">(cross)"</span>, <span class="st">"Residual score</span><span class="sc">\n</span><span class="st">(cross)"</span>)</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(cormat)[[<span class="dv">2</span>]] <span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">"Alzheimer's &amp; related dementias</span><span class="sc">\n</span><span class="st">(Bellenguez et al., 2021)"</span>, <span class="st">"Alzheimer's</span><span class="sc">\n</span><span class="st">(Kunkle et al., 2019)"</span>, <span class="st">"Parkinson's</span><span class="sc">\n</span><span class="st">(Nalls et al., 2019)"</span>, <span class="st">"Amyotrophic lateral sclerosis</span><span class="sc">\n</span><span class="st">(van Rheenen et al. 2021)"</span>, <span class="st">"ICV</span><span class="sc">\n</span><span class="st">(Adams et al., 2016)"</span>, <span class="st">"TBV</span><span class="sc">\n</span><span class="st">(Zhao et al., 2018)"</span>, <span class="st">"TBV</span><span class="sc">\n</span><span class="st">(Smith et al., 2021)"</span>, <span class="st">"Grey matter whole brain PC1</span><span class="sc">\n</span><span class="st">(Furtjes et al., 2023)"</span>, <span class="st">"Brain age</span><span class="sc">\n</span><span class="st">(Kaufmann et al., 2019)"</span>,<span class="st">"ICV"</span>, <span class="st">"TBV"</span>, <span class="st">"CSF"</span>, <span class="st">"Difference score</span><span class="sc">\n</span><span class="st">(cross)"</span>, <span class="st">"Ratio score</span><span class="sc">\n</span><span class="st">(cross)"</span>, <span class="st">"Residual score</span><span class="sc">\n</span><span class="st">(cross)"</span>)</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a><span class="co"># define function to obtain lower triangle</span></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>  get_lower_tri<span class="ot">&lt;-</span><span class="cf">function</span>(cormat){</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>    cormat[<span class="fu">upper.tri</span>(cormat)] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(cormat)</span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get correlation matrix for both samples together</span></span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>  cor <span class="ot">=</span> <span class="fu">get_lower_tri</span>(cormat)</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># melt matrix</span></span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>  melted <span class="ot">=</span> reshape2<span class="sc">::</span><span class="fu">melt</span>(cor)</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>    <span class="co"># one value is &gt; 1 which can happen sometimes</span></span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>    melted<span class="sc">$</span>value[melted<span class="sc">$</span>value <span class="sc">&gt;</span> <span class="fl">1.01</span>] <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a>    <span class="co"># get rounded value</span></span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a>  melted<span class="sc">$</span>value_round <span class="ot">=</span> <span class="fu">round</span>(melted<span class="sc">$</span>value, <span class="at">digit =</span> <span class="dv">2</span>)</span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>  melted<span class="sc">$</span>distance0 <span class="ot">=</span> <span class="fu">abs</span>(melted<span class="sc">$</span>value)</span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a>    <span class="co"># for some reason one corr over 1 doesn't get colored</span></span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a>    melted<span class="sc">$</span>value <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(melted<span class="sc">$</span>value <span class="sc">&gt;</span> <span class="dv">1</span>, <span class="dv">1</span>, melted<span class="sc">$</span>value)</span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a>    <span class="co"># pull out standard errors</span></span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>    <span class="co"># SEs will be listed in the same order as they are listed in the genetic covariance matrix  </span></span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a>    r<span class="ot">&lt;-</span><span class="fu">nrow</span>(LDSCoutput<span class="sc">$</span>S)</span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>    SE_Stand<span class="ot">&lt;-</span><span class="fu">matrix</span>(<span class="dv">0</span>, r, r)</span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>    SE_Stand[<span class="fu">lower.tri</span>(SE_Stand,<span class="at">diag=</span><span class="cn">TRUE</span>)] <span class="ot">&lt;-</span><span class="fu">sqrt</span>(<span class="fu">diag</span>(LDSCoutput<span class="sc">$</span>V_Stand))</span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dimnames</span>(SE_Stand)<span class="ot">&lt;-</span> <span class="fu">dimnames</span>(LDSCoutput<span class="sc">$</span>S)</span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a>    <span class="co"># re-roder</span></span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a>    SE_Stand <span class="ot">=</span> SE_Stand[<span class="fu">c</span>(<span class="st">"ALZ"</span>,<span class="st">"ALZ_2019"</span>,<span class="st">"PD"</span>,<span class="st">"ALS"</span>,<span class="st">"ICV_Adams.et.al"</span>,<span class="st">"TBV_Zhao.et.al"</span>, <span class="st">"TBV_Smith.et.al"</span>, <span class="st">"PC1_whole_brain_greymatter"</span>,<span class="st">"brainage"</span>, <span class="st">"ICVstand"</span>, <span class="st">"TBVstand"</span>, <span class="st">"CSFstand"</span>, <span class="st">"diff_stand"</span>, <span class="st">"ratio_stand"</span>, <span class="st">"resid_stand"</span>), <span class="fu">c</span>(<span class="st">"ALZ"</span>,<span class="st">"ALZ_2019"</span>,<span class="st">"PD"</span>,<span class="st">"ALS"</span>,<span class="st">"ICV_Adams.et.al"</span>, <span class="st">"TBV_Zhao.et.al"</span>, <span class="st">"TBV_Smith.et.al"</span>, <span class="st">"PC1_whole_brain_greymatter"</span>,<span class="st">"brainage"</span>, <span class="st">"ICVstand"</span>, <span class="st">"TBVstand"</span>, <span class="st">"CSFstand"</span>, <span class="st">"diff_stand"</span>, <span class="st">"ratio_stand"</span>, <span class="st">"resid_stand"</span>)]</span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a>    <span class="co"># re-name</span></span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dimnames</span>(SE_Stand)[[<span class="dv">1</span>]] <span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">"Alzheimer's &amp; related dementias</span><span class="sc">\n</span><span class="st">(Bellenguez et al., 2021)"</span>, <span class="st">"Alzheimer's</span><span class="sc">\n</span><span class="st">(Kunkle et al., 2019)"</span>, <span class="st">"Parkinson's</span><span class="sc">\n</span><span class="st">(Nalls et al., 2019)"</span>, <span class="st">"Amyotrophic lateral sclerosis</span><span class="sc">\n</span><span class="st">(van Rheenen et al. 2021)"</span>, <span class="st">"ICV</span><span class="sc">\n</span><span class="st">(Adams et al., 2016)"</span>, <span class="st">"TBV</span><span class="sc">\n</span><span class="st">(Zhao et al., 2018)"</span>, <span class="st">"TBV</span><span class="sc">\n</span><span class="st">(Smith et al., 2021)"</span>, <span class="st">"Grey matter whole brain PC1</span><span class="sc">\n</span><span class="st">(Furtjes et al., 2023)"</span>, <span class="st">"Brain age</span><span class="sc">\n</span><span class="st">(Kaufmann et al., 2019)"</span>,<span class="st">"ICV"</span>, <span class="st">"TBV"</span>, <span class="st">"CSF"</span>, <span class="st">"Difference score</span><span class="sc">\n</span><span class="st">(cross)"</span>, <span class="st">"Ratio score</span><span class="sc">\n</span><span class="st">(cross)"</span>, <span class="st">"Residual score</span><span class="sc">\n</span><span class="st">(cross)"</span>)</span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dimnames</span>(SE_Stand)[[<span class="dv">2</span>]] <span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">"Alzheimer's &amp; related dementias</span><span class="sc">\n</span><span class="st">(Bellenguez et al., 2021)"</span>, <span class="st">"Alzheimer's</span><span class="sc">\n</span><span class="st">(Kunkle et al., 2019)"</span>, <span class="st">"Parkinson's</span><span class="sc">\n</span><span class="st">(Nalls et al., 2019)"</span>, <span class="st">"Amyotrophic lateral sclerosis</span><span class="sc">\n</span><span class="st">(van Rheenen et al. 2021)"</span>, <span class="st">"ICV</span><span class="sc">\n</span><span class="st">(Adams et al., 2016)"</span>, <span class="st">"TBV</span><span class="sc">\n</span><span class="st">(Zhao et al., 2018)"</span>, <span class="st">"TBV</span><span class="sc">\n</span><span class="st">(Smith et al., 2021)"</span>, <span class="st">"Grey matter whole brain PC1</span><span class="sc">\n</span><span class="st">(Furtjes et al., 2023)"</span>, <span class="st">"Brain age</span><span class="sc">\n</span><span class="st">(Kaufmann et al., 2019)"</span>,<span class="st">"ICV"</span>, <span class="st">"TBV"</span>, <span class="st">"CSF"</span>, <span class="st">"Difference score</span><span class="sc">\n</span><span class="st">(cross)"</span>, <span class="st">"Ratio score</span><span class="sc">\n</span><span class="st">(cross)"</span>, <span class="st">"Residual score</span><span class="sc">\n</span><span class="st">(cross)"</span>)</span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a>    <span class="co">#melt data</span></span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a>    SEmelted <span class="ot">&lt;-</span> reshape2<span class="sc">::</span><span class="fu">melt</span>(SE_Stand)</span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(SEmelted)[<span class="fu">grepl</span>(<span class="st">"value"</span>, <span class="fu">names</span>(SEmelted))] <span class="ot">&lt;-</span> <span class="st">"SE"</span></span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a>    <span class="co"># merge with corr data</span></span>
<span id="cb3-56"><a href="#cb3-56" aria-hidden="true" tabindex="-1"></a>    both <span class="ot">=</span> <span class="fu">merge</span>(melted, SEmelted, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Var1"</span>, <span class="st">"Var2"</span>), <span class="at">all.x =</span> T)</span>
<span id="cb3-57"><a href="#cb3-57" aria-hidden="true" tabindex="-1"></a>    <span class="co"># work out confidence intervals</span></span>
<span id="cb3-58"><a href="#cb3-58" aria-hidden="true" tabindex="-1"></a>    both<span class="sc">$</span>ci_u <span class="ot">&lt;-</span> both<span class="sc">$</span>value <span class="sc">+</span> (<span class="fl">1.96</span><span class="sc">*</span>both<span class="sc">$</span>SE)</span>
<span id="cb3-59"><a href="#cb3-59" aria-hidden="true" tabindex="-1"></a>    both<span class="sc">$</span>ci_l <span class="ot">&lt;-</span> both<span class="sc">$</span>value <span class="sc">-</span> (<span class="fl">1.96</span><span class="sc">*</span>both<span class="sc">$</span>SE)</span>
<span id="cb3-60"><a href="#cb3-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-61"><a href="#cb3-61" aria-hidden="true" tabindex="-1"></a>    <span class="co"># get sig assocs</span></span>
<span id="cb3-62"><a href="#cb3-62" aria-hidden="true" tabindex="-1"></a>    both<span class="sc">$</span>sig <span class="ot">&lt;-</span> <span class="st">""</span></span>
<span id="cb3-63"><a href="#cb3-63" aria-hidden="true" tabindex="-1"></a>    both<span class="sc">$</span>sig[<span class="fu">which</span>(both<span class="sc">$</span>ci_u <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> both<span class="sc">$</span>ci_l <span class="sc">&lt;</span> <span class="dv">0</span>)] <span class="ot">&lt;-</span> <span class="st">"</span><span class="sc">\n</span><span class="st">(ns.)"</span></span>
<span id="cb3-64"><a href="#cb3-64" aria-hidden="true" tabindex="-1"></a>    both<span class="sc">$</span>sig[<span class="fu">which</span>(both<span class="sc">$</span>ci_u <span class="sc">&lt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> both<span class="sc">$</span>ci_l <span class="sc">&gt;</span> <span class="dv">0</span>)] <span class="ot">&lt;-</span> <span class="st">"</span><span class="sc">\n</span><span class="st">(ns.)"</span></span>
<span id="cb3-65"><a href="#cb3-65" aria-hidden="true" tabindex="-1"></a>    both<span class="sc">$</span>sig[<span class="fu">is.na</span>(both<span class="sc">$</span>value)] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb3-66"><a href="#cb3-66" aria-hidden="true" tabindex="-1"></a>    both<span class="sc">$</span>value_round <span class="ot">&lt;-</span> <span class="fu">paste0</span>(both<span class="sc">$</span>value_round, <span class="st">" "</span>, both<span class="sc">$</span>sig)</span>
<span id="cb3-67"><a href="#cb3-67" aria-hidden="true" tabindex="-1"></a>    both<span class="sc">$</span>value_round[<span class="fu">is.na</span>(both<span class="sc">$</span>value)] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb3-68"><a href="#cb3-68" aria-hidden="true" tabindex="-1"></a>    <span class="co">#both$value_round[grep("NA", both$value_round)] &lt;- str_remove(both$value_round[grep("NA", both$value_round)], pattern = " NA")</span></span>
<span id="cb3-69"><a href="#cb3-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-70"><a href="#cb3-70" aria-hidden="true" tabindex="-1"></a>  <span class="co"># plot</span></span>
<span id="cb3-71"><a href="#cb3-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ggplot2)</span>
<span id="cb3-72"><a href="#cb3-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-73"><a href="#cb3-73" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">=</span> <span class="fu">ggplot</span>()<span class="sc">+</span></span>
<span id="cb3-74"><a href="#cb3-74" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">data =</span> both, <span class="fu">aes</span>(<span class="at">x =</span> Var1, <span class="at">y =</span> Var2, <span class="at">shape =</span> value, <span class="at">fill =</span> value, <span class="at">size =</span> distance0), <span class="at">shape =</span> <span class="dv">21</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>, <span class="at">colour =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb3-75"><a href="#cb3-75" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_gradient2</span>(<span class="at">low =</span> <span class="st">"grey45"</span>, <span class="at">mid =</span> <span class="st">"white"</span>, <span class="at">high =</span> <span class="st">"grey45"</span>,<span class="at">limit =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>), <span class="at">space =</span> <span class="st">"Lab"</span> ,<span class="at">name=</span><span class="st">"Correlation"</span>, <span class="at">guide =</span> <span class="st">"legend"</span>)<span class="sc">+</span></span>
<span id="cb3-76"><a href="#cb3-76" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_size_continuous</span>(<span class="at">range =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">15</span>), <span class="at">guide =</span> <span class="st">"none"</span>)<span class="sc">+</span></span>
<span id="cb3-77"><a href="#cb3-77" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="at">data =</span> both, <span class="fu">aes</span>(Var1, Var2, <span class="at">label =</span> value_round), <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">3</span>)<span class="sc">+</span></span>
<span id="cb3-78"><a href="#cb3-78" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_rect</span>(<span class="fu">aes</span>(<span class="at">xmax =</span> <span class="fu">stage</span>(<span class="st">"Residual score</span><span class="sc">\n</span><span class="st">(cross)"</span>, <span class="fu">after_scale</span>(xmax<span class="fl">+0.5</span>)), </span>
<span id="cb3-79"><a href="#cb3-79" aria-hidden="true" tabindex="-1"></a>                    <span class="at">xmin =</span> <span class="fu">stage</span>(<span class="st">"Residual score</span><span class="sc">\n</span><span class="st">(cross)"</span>, <span class="fu">after_scale</span>(xmin<span class="fl">-0.5</span>)), </span>
<span id="cb3-80"><a href="#cb3-80" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ymax =</span> <span class="fu">stage</span>(<span class="st">"Residual score</span><span class="sc">\n</span><span class="st">(cross)"</span>, <span class="fu">after_scale</span>(ymax<span class="fl">+0.5</span>)), </span>
<span id="cb3-81"><a href="#cb3-81" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ymin =</span> <span class="fu">stage</span>(<span class="st">"Alzheimer's &amp; related dementias</span><span class="sc">\n</span><span class="st">(Bellenguez et al., 2021)"</span>, <span class="fu">after_scale</span>(ymin<span class="fl">-0.5</span>))), <span class="at">fill =</span> <span class="st">"#004D40"</span>, <span class="at">alpha =</span> <span class="fl">0.3</span>)<span class="sc">+</span></span>
<span id="cb3-82"><a href="#cb3-82" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_rect</span>(<span class="fu">aes</span>(<span class="at">xmax =</span> <span class="fu">stage</span>(<span class="st">"Ratio score</span><span class="sc">\n</span><span class="st">(cross)"</span>, <span class="fu">after_scale</span>(xmax<span class="fl">+0.5</span>)), </span>
<span id="cb3-83"><a href="#cb3-83" aria-hidden="true" tabindex="-1"></a>                    <span class="at">xmin =</span> <span class="fu">stage</span>(<span class="st">"Ratio score</span><span class="sc">\n</span><span class="st">(cross)"</span>, <span class="fu">after_scale</span>(xmin<span class="fl">-0.5</span>)), </span>
<span id="cb3-84"><a href="#cb3-84" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ymax =</span> <span class="fu">stage</span>(<span class="st">"Ratio score</span><span class="sc">\n</span><span class="st">(cross)"</span>, <span class="fu">after_scale</span>(ymax<span class="fl">+0.5</span>)), </span>
<span id="cb3-85"><a href="#cb3-85" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ymin =</span> <span class="fu">stage</span>(<span class="st">"Alzheimer's &amp; related dementias</span><span class="sc">\n</span><span class="st">(Bellenguez et al., 2021)"</span>, <span class="fu">after_scale</span>(ymin<span class="fl">-0.5</span>))), <span class="at">fill =</span> <span class="st">"#FFC107"</span>, <span class="at">alpha =</span> <span class="fl">0.3</span>)<span class="sc">+</span></span>
<span id="cb3-86"><a href="#cb3-86" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_rect</span>(<span class="fu">aes</span>(<span class="at">xmax =</span> <span class="fu">stage</span>(<span class="st">"Difference score</span><span class="sc">\n</span><span class="st">(cross)"</span>, <span class="fu">after_scale</span>(xmax<span class="fl">+0.5</span>)), </span>
<span id="cb3-87"><a href="#cb3-87" aria-hidden="true" tabindex="-1"></a>                    <span class="at">xmin =</span> <span class="fu">stage</span>(<span class="st">"Difference score</span><span class="sc">\n</span><span class="st">(cross)"</span>, <span class="fu">after_scale</span>(xmin<span class="fl">-0.5</span>)), </span>
<span id="cb3-88"><a href="#cb3-88" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ymax =</span> <span class="fu">stage</span>(<span class="st">"Difference score</span><span class="sc">\n</span><span class="st">(cross)"</span>, <span class="fu">after_scale</span>(ymax<span class="fl">+0.5</span>)), </span>
<span id="cb3-89"><a href="#cb3-89" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ymin =</span> <span class="fu">stage</span>(<span class="st">"Alzheimer's &amp; related dementias</span><span class="sc">\n</span><span class="st">(Bellenguez et al., 2021)"</span>, <span class="fu">after_scale</span>(ymin<span class="fl">-0.5</span>))), <span class="at">fill =</span> <span class="st">"#D81B60"</span>, <span class="at">alpha =</span> <span class="fl">0.3</span>)<span class="sc">+</span></span>
<span id="cb3-90"><a href="#cb3-90" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">""</span>)<span class="sc">+</span></span>
<span id="cb3-91"><a href="#cb3-91" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">""</span>)<span class="sc">+</span></span>
<span id="cb3-92"><a href="#cb3-92" aria-hidden="true" tabindex="-1"></a>    <span class="co">#scale_x_discrete(labels = axisNames)+</span></span>
<span id="cb3-93"><a href="#cb3-93" aria-hidden="true" tabindex="-1"></a>    <span class="co">#scale_y_discrete(labels = axisNames)+</span></span>
<span id="cb3-94"><a href="#cb3-94" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">"none"</span>)<span class="sc">+</span></span>
<span id="cb3-95"><a href="#cb3-95" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb3-96"><a href="#cb3-96" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb3-97"><a href="#cb3-97" aria-hidden="true" tabindex="-1"></a>            <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">270</span>, <span class="at">vjust =</span> <span class="dv">0</span>, <span class="at">hjust =</span><span class="dv">0</span>))<span class="sc">+</span></span>
<span id="cb3-98"><a href="#cb3-98" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"Genetic correlations"</span>)</span>
<span id="cb3-99"><a href="#cb3-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-100"><a href="#cb3-100" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"genetic_corr_neurodeg.png"</span>, <span class="at">plot =</span> p, <span class="at">width =</span> <span class="dv">22</span>, <span class="at">height =</span> <span class="dv">22</span>, <span class="at">units =</span> <span class="st">"cm"</span>, <span class="at">dpi =</span> <span class="dv">600</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><img src="images/genetic_corr_neurodeg.png" class="img-fluid"></p>
</section>
<section id="plot-sex-split-correlations" class="level3">
<h3 class="anchored" data-anchor-id="plot-sex-split-correlations">Plot sex-split correlations</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="do">###########################################################################</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="do">#### Genetic correlation matrix</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="do">#### Sex split</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="do">###########################################################################</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"LDSCoutput_sexsplit.RData"</span>)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="co"># pull out correlation matrix</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>cormat <span class="ot">&lt;-</span> LDSCoutput<span class="sc">$</span>S_Stand</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="co"># re-roder</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>cormat <span class="ot">=</span> cormat[<span class="fu">c</span>(<span class="st">"ICVstand"</span>,<span class="st">"TBVstand"</span>,<span class="st">"CSFstand"</span>,<span class="st">"diff_stand"</span>, <span class="st">"ratio_stand"</span>, <span class="st">"resid_stand"</span>, <span class="st">"diff_stand_males"</span>, <span class="st">"ratio_stand_males"</span>,<span class="st">"resid_stand_males"</span>,<span class="st">"diff_stand_females"</span>, <span class="st">"ratio_stand_females"</span>, <span class="st">"resid_stand_females"</span>), <span class="fu">c</span>(<span class="st">"ICVstand"</span>,<span class="st">"TBVstand"</span>,<span class="st">"CSFstand"</span>,<span class="st">"diff_stand"</span>, <span class="st">"ratio_stand"</span>, <span class="st">"resid_stand"</span>, <span class="st">"diff_stand_males"</span>, <span class="st">"ratio_stand_males"</span>,<span class="st">"resid_stand_males"</span>,<span class="st">"diff_stand_females"</span>, <span class="st">"ratio_stand_females"</span>, <span class="st">"resid_stand_females"</span>)]</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="co"># re-name</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(cormat)[[<span class="dv">1</span>]] <span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">"ICV (all)"</span>,<span class="st">"TBV (all)"</span>,<span class="st">"CSF (all)"</span>,<span class="st">"Difference Score (all)"</span>, <span class="st">"Ratio score (all)"</span>, <span class="st">"Residual score (all)"</span>, <span class="st">"Difference Score (males)"</span>, <span class="st">"Ratio score (males)"</span>,<span class="st">"Residual score (males)"</span>,<span class="st">"Difference Score (females)"</span>, <span class="st">"Ratio score (females)"</span>, <span class="st">"Residual score (females)"</span>)</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(cormat)[[<span class="dv">2</span>]] <span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">"ICV (all)"</span>,<span class="st">"TBV (all)"</span>,<span class="st">"CSF (all)"</span>,<span class="st">"Difference Score (all)"</span>, <span class="st">"Ratio score (all)"</span>, <span class="st">"Residual score (all)"</span>, <span class="st">"Difference Score (males)"</span>, <span class="st">"Ratio score (males)"</span>,<span class="st">"Residual score (males)"</span>,<span class="st">"Difference Score (females)"</span>, <span class="st">"Ratio score (females)"</span>, <span class="st">"Residual score (females)"</span>)</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a><span class="co"># define function to obtain lower triangle</span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>  get_lower_tri<span class="ot">&lt;-</span><span class="cf">function</span>(cormat){</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>    cormat[<span class="fu">upper.tri</span>(cormat)] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(cormat)</span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get correlation matrix for both samples together</span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>  cor <span class="ot">=</span> <span class="fu">get_lower_tri</span>(cormat)</span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># melt matrix</span></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>  melted <span class="ot">=</span> reshape2<span class="sc">::</span><span class="fu">melt</span>(cor)</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>    <span class="co"># one value is &gt; 1 which can happen sometimes</span></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>    melted<span class="sc">$</span>value[melted<span class="sc">$</span>value <span class="sc">&gt;</span> <span class="fl">1.01</span>] <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>    <span class="co"># get rounded value</span></span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>  melted<span class="sc">$</span>value_round <span class="ot">=</span> <span class="fu">round</span>(melted<span class="sc">$</span>value, <span class="at">digit =</span> <span class="dv">2</span>)</span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a>  melted<span class="sc">$</span>distance0 <span class="ot">=</span> <span class="fu">abs</span>(melted<span class="sc">$</span>value)</span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a>    <span class="co"># for some reason one corr over 1 doesn't get colored</span></span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a>    melted<span class="sc">$</span>value <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(melted<span class="sc">$</span>value <span class="sc">&gt;</span> <span class="dv">1</span>, <span class="dv">1</span>, melted<span class="sc">$</span>value)</span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a>    <span class="co"># pull out standard errors</span></span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a>    <span class="co"># SEs will be listed in the same order as they are listed in the genetic covariance matrix  </span></span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a>    r<span class="ot">&lt;-</span><span class="fu">nrow</span>(LDSCoutput<span class="sc">$</span>S)</span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a>    SE_Stand<span class="ot">&lt;-</span><span class="fu">matrix</span>(<span class="dv">0</span>, r, r)</span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a>    SE_Stand[<span class="fu">lower.tri</span>(SE_Stand,<span class="at">diag=</span><span class="cn">TRUE</span>)] <span class="ot">&lt;-</span><span class="fu">sqrt</span>(<span class="fu">diag</span>(LDSCoutput<span class="sc">$</span>V_Stand))</span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dimnames</span>(SE_Stand)<span class="ot">&lt;-</span> <span class="fu">dimnames</span>(LDSCoutput<span class="sc">$</span>S)</span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a>    <span class="co"># re-roder</span></span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a>    SE_Stand <span class="ot">=</span> SE_Stand[<span class="fu">c</span>(<span class="st">"ICVstand"</span>,<span class="st">"TBVstand"</span>,<span class="st">"CSFstand"</span>,<span class="st">"diff_stand"</span>, <span class="st">"ratio_stand"</span>, <span class="st">"resid_stand"</span>, <span class="st">"diff_stand_males"</span>, <span class="st">"ratio_stand_males"</span>,<span class="st">"resid_stand_males"</span>,<span class="st">"diff_stand_females"</span>, <span class="st">"ratio_stand_females"</span>, <span class="st">"resid_stand_females"</span>) ,<span class="fu">c</span>(<span class="st">"ICVstand"</span>,<span class="st">"TBVstand"</span>,<span class="st">"CSFstand"</span>,<span class="st">"diff_stand"</span>, <span class="st">"ratio_stand"</span>, <span class="st">"resid_stand"</span>, <span class="st">"diff_stand_males"</span>, <span class="st">"ratio_stand_males"</span>,<span class="st">"resid_stand_males"</span>,<span class="st">"diff_stand_females"</span>, <span class="st">"ratio_stand_females"</span>, <span class="st">"resid_stand_females"</span>)]</span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a>    <span class="co"># re-name</span></span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dimnames</span>(SE_Stand)[[<span class="dv">1</span>]] <span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">"ICV (all)"</span>,<span class="st">"TBV (all)"</span>,<span class="st">"CSF (all)"</span>,<span class="st">"Difference Score (all)"</span>, <span class="st">"Ratio score (all)"</span>, <span class="st">"Residual score (all)"</span>, <span class="st">"Difference Score (males)"</span>, <span class="st">"Ratio score (males)"</span>,<span class="st">"Residual score (males)"</span>,<span class="st">"Difference Score (females)"</span>, <span class="st">"Ratio score (females)"</span>, <span class="st">"Residual score (females)"</span>)</span>
<span id="cb4-48"><a href="#cb4-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dimnames</span>(SE_Stand)[[<span class="dv">2</span>]] <span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">"ICV (all)"</span>,<span class="st">"TBV (all)"</span>,<span class="st">"CSF (all)"</span>,<span class="st">"Difference Score (all)"</span>, <span class="st">"Ratio score (all)"</span>, <span class="st">"Residual score (all)"</span>, <span class="st">"Difference Score (males)"</span>, <span class="st">"Ratio score (males)"</span>,<span class="st">"Residual score (males)"</span>,<span class="st">"Difference Score (females)"</span>, <span class="st">"Ratio score (females)"</span>, <span class="st">"Residual score (females)"</span>)</span>
<span id="cb4-49"><a href="#cb4-49" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-50"><a href="#cb4-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-51"><a href="#cb4-51" aria-hidden="true" tabindex="-1"></a>    <span class="co">#melt data</span></span>
<span id="cb4-52"><a href="#cb4-52" aria-hidden="true" tabindex="-1"></a>    SEmelted <span class="ot">&lt;-</span> reshape2<span class="sc">::</span><span class="fu">melt</span>(SE_Stand)</span>
<span id="cb4-53"><a href="#cb4-53" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(SEmelted)[<span class="fu">grepl</span>(<span class="st">"value"</span>, <span class="fu">names</span>(SEmelted))] <span class="ot">&lt;-</span> <span class="st">"SE"</span></span>
<span id="cb4-54"><a href="#cb4-54" aria-hidden="true" tabindex="-1"></a>    <span class="co"># merge with corr data</span></span>
<span id="cb4-55"><a href="#cb4-55" aria-hidden="true" tabindex="-1"></a>    both <span class="ot">=</span> <span class="fu">merge</span>(melted, SEmelted, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Var1"</span>, <span class="st">"Var2"</span>), <span class="at">all.x =</span> T)</span>
<span id="cb4-56"><a href="#cb4-56" aria-hidden="true" tabindex="-1"></a>    <span class="co"># work out confidence intervals</span></span>
<span id="cb4-57"><a href="#cb4-57" aria-hidden="true" tabindex="-1"></a>    both<span class="sc">$</span>ci_u <span class="ot">&lt;-</span> both<span class="sc">$</span>value <span class="sc">+</span> (<span class="fl">1.96</span><span class="sc">*</span>both<span class="sc">$</span>SE)</span>
<span id="cb4-58"><a href="#cb4-58" aria-hidden="true" tabindex="-1"></a>    both<span class="sc">$</span>ci_l <span class="ot">&lt;-</span> both<span class="sc">$</span>value <span class="sc">-</span> (<span class="fl">1.96</span><span class="sc">*</span>both<span class="sc">$</span>SE)</span>
<span id="cb4-59"><a href="#cb4-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-60"><a href="#cb4-60" aria-hidden="true" tabindex="-1"></a>    <span class="co"># get sig assocs</span></span>
<span id="cb4-61"><a href="#cb4-61" aria-hidden="true" tabindex="-1"></a>    both<span class="sc">$</span>sig <span class="ot">&lt;-</span> <span class="st">""</span></span>
<span id="cb4-62"><a href="#cb4-62" aria-hidden="true" tabindex="-1"></a>    both<span class="sc">$</span>sig[<span class="fu">which</span>(both<span class="sc">$</span>ci_u <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> both<span class="sc">$</span>ci_l <span class="sc">&lt;</span> <span class="dv">0</span>)] <span class="ot">&lt;-</span> <span class="st">"</span><span class="sc">\n</span><span class="st">(ns.)"</span></span>
<span id="cb4-63"><a href="#cb4-63" aria-hidden="true" tabindex="-1"></a>    both<span class="sc">$</span>sig[<span class="fu">which</span>(both<span class="sc">$</span>ci_u <span class="sc">&lt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> both<span class="sc">$</span>ci_l <span class="sc">&gt;</span> <span class="dv">0</span>)] <span class="ot">&lt;-</span> <span class="st">"</span><span class="sc">\n</span><span class="st">(ns.)"</span></span>
<span id="cb4-64"><a href="#cb4-64" aria-hidden="true" tabindex="-1"></a>    both<span class="sc">$</span>sig[<span class="fu">is.na</span>(both<span class="sc">$</span>value)] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb4-65"><a href="#cb4-65" aria-hidden="true" tabindex="-1"></a>    both<span class="sc">$</span>value_round <span class="ot">&lt;-</span> <span class="fu">paste0</span>(both<span class="sc">$</span>value_round, <span class="st">" "</span>, both<span class="sc">$</span>sig)</span>
<span id="cb4-66"><a href="#cb4-66" aria-hidden="true" tabindex="-1"></a>    both<span class="sc">$</span>value_round[<span class="fu">is.na</span>(both<span class="sc">$</span>value)] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb4-67"><a href="#cb4-67" aria-hidden="true" tabindex="-1"></a>    <span class="co">#both$value_round[grep("NA", both$value_round)] &lt;- str_remove(both$value_round[grep("NA", both$value_round)], pattern = " NA")</span></span>
<span id="cb4-68"><a href="#cb4-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-69"><a href="#cb4-69" aria-hidden="true" tabindex="-1"></a>  <span class="co"># plot</span></span>
<span id="cb4-70"><a href="#cb4-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ggplot2)</span>
<span id="cb4-71"><a href="#cb4-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-72"><a href="#cb4-72" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">=</span> <span class="fu">ggplot</span>()<span class="sc">+</span></span>
<span id="cb4-73"><a href="#cb4-73" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">data =</span> both, <span class="fu">aes</span>(<span class="at">x =</span> Var1, <span class="at">y =</span> Var2, <span class="at">shape =</span> value, <span class="at">fill =</span> value, <span class="at">size =</span> distance0), <span class="at">shape =</span> <span class="dv">21</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>, <span class="at">colour =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb4-74"><a href="#cb4-74" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_gradient2</span>(<span class="at">low =</span> <span class="st">"#91D2D9"</span>, <span class="at">mid =</span> <span class="st">"white"</span>, <span class="at">high =</span> <span class="st">"#FFC4C4"</span>,<span class="at">limit =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>), <span class="at">space =</span> <span class="st">"Lab"</span> ,<span class="at">name=</span><span class="st">"Correlation"</span>, <span class="at">guide =</span> <span class="st">"legend"</span>)<span class="sc">+</span></span>
<span id="cb4-75"><a href="#cb4-75" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_size_continuous</span>(<span class="at">range =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">15</span>), <span class="at">guide =</span> <span class="st">"none"</span>)<span class="sc">+</span></span>
<span id="cb4-76"><a href="#cb4-76" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="at">data =</span> both, <span class="fu">aes</span>(Var1, Var2, <span class="at">label =</span> value_round), <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">3</span>)<span class="sc">+</span></span>
<span id="cb4-77"><a href="#cb4-77" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">""</span>)<span class="sc">+</span></span>
<span id="cb4-78"><a href="#cb4-78" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">""</span>)<span class="sc">+</span></span>
<span id="cb4-79"><a href="#cb4-79" aria-hidden="true" tabindex="-1"></a>    <span class="co">#scale_x_discrete(labels = axisNames)+</span></span>
<span id="cb4-80"><a href="#cb4-80" aria-hidden="true" tabindex="-1"></a>    <span class="co">#scale_y_discrete(labels = axisNames)+</span></span>
<span id="cb4-81"><a href="#cb4-81" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill =</span> <span class="st">"none"</span>)<span class="sc">+</span></span>
<span id="cb4-82"><a href="#cb4-82" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb4-83"><a href="#cb4-83" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb4-84"><a href="#cb4-84" aria-hidden="true" tabindex="-1"></a>            <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">270</span>, <span class="at">vjust =</span> <span class="dv">0</span>, <span class="at">hjust =</span><span class="dv">0</span>))<span class="sc">+</span></span>
<span id="cb4-85"><a href="#cb4-85" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"Genetic correlations"</span>)</span>
<span id="cb4-86"><a href="#cb4-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-87"><a href="#cb4-87" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"genetic_corr_sexsplit.png"</span>, <span class="at">plot =</span> p, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">20</span>, <span class="at">units =</span> <span class="st">"cm"</span>, <span class="at">dpi =</span> <span class="dv">600</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/genetic_corr_sexsplit.png" class="img-fluid figure-img"></p>
</figure>
</div>
</section>
</section>
<section id="gwas-by-subtraction-in-genomicsem" class="level2">
<h2 class="anchored" data-anchor-id="gwas-by-subtraction-in-genomicsem">GWAS-by-subtraction in GenomicSEM</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(wd)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomicSEM)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="do">#### Step 2</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>traits <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"TBVstand.sumstats.gz"</span>,<span class="st">"ICVstand.sumstats.gz"</span>,<span class="st">"resid_stand.sumstats.gz"</span>)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>sample.prev <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(traits))</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>population.prev <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(traits))</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>ld<span class="ot">&lt;-</span><span class="st">"eur_w_ld_chr/"</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>wld <span class="ot">&lt;-</span> <span class="st">"eur_w_ld_chr/"</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>trait.names<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">"TBV"</span>, <span class="st">"ICV"</span>, <span class="st">"resid"</span>)</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>LDSCoutput <span class="ot">&lt;-</span> <span class="fu">ldsc</span>(<span class="at">traits =</span> traits, </span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>                   <span class="at">sample.prev =</span> sample.prev, </span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>                   <span class="at">population.prev =</span> population.prev, </span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>                   <span class="at">ld=</span>ld, </span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>                   <span class="at">wld=</span>wld, </span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>                   <span class="at">trait.names=</span>trait.names,</span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">stand =</span> T)</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(LDSCoutput, <span class="at">file=</span><span class="st">"LDSCoutput_TBV_ICV.RData"</span>)</span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a><span class="do">###### Step 3</span></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="at">file=</span><span class="fu">paste0</span>(wd, <span class="st">"/LDSCoutput_TBV_ICV.RData"</span>))</span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a><span class="co"># BaseICV is captured by both diff atrophy score and ICV</span></span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Atrophy however should only be indicated by diff atrophy score and should be independent of ICV</span></span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a>model<span class="ot">&lt;-</span><span class="st">'</span></span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a><span class="st">        BaseICV=~NA*TBV + ICV</span></span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a><span class="st">        Atrophy=~NA*TBV</span></span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a><span class="st">         </span></span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a><span class="st">        # make sure all variances and covariances are captured by the latent factors</span></span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a><span class="st">         Atrophy ~~ 1*Atrophy</span></span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a><span class="st">         BaseICV ~~ 1*BaseICV</span></span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a><span class="st">        # forcing atrophy and BaseICV to be uncorrelated</span></span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a><span class="st">         Atrophy~~0*BaseICV</span></span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true" tabindex="-1"></a><span class="st">         ICV ~~ 0*TBV</span></span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true" tabindex="-1"></a><span class="st">         ICV ~~ 0*ICV</span></span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true" tabindex="-1"></a><span class="st">         TBV ~~ 0*TBV</span></span>
<span id="cb5-42"><a href="#cb5-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-43"><a href="#cb5-43" aria-hidden="true" tabindex="-1"></a><span class="st">        # add in resid to how correlated it is with Atrophy</span></span>
<span id="cb5-44"><a href="#cb5-44" aria-hidden="true" tabindex="-1"></a><span class="st">        Atrophy ~~ resid</span></span>
<span id="cb5-45"><a href="#cb5-45" aria-hidden="true" tabindex="-1"></a><span class="st">        BaseICV ~~ resid</span></span>
<span id="cb5-46"><a href="#cb5-46" aria-hidden="true" tabindex="-1"></a><span class="st">'</span></span>
<span id="cb5-47"><a href="#cb5-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-48"><a href="#cb5-48" aria-hidden="true" tabindex="-1"></a><span class="co"># this is a fully saturated model (which is why no fit statistics are printed)</span></span>
<span id="cb5-49"><a href="#cb5-49" aria-hidden="true" tabindex="-1"></a>output<span class="ot">&lt;-</span><span class="fu">usermodel</span>(LDSCoutput,<span class="at">estimation=</span><span class="st">"DWLS"</span>,<span class="at">model=</span>model)</span>
<span id="cb5-50"><a href="#cb5-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-51"><a href="#cb5-51" aria-hidden="true" tabindex="-1"></a><span class="do">## Step 4</span></span>
<span id="cb5-52"><a href="#cb5-52" aria-hidden="true" tabindex="-1"></a><span class="co"># files to prepare</span></span>
<span id="cb5-53"><a href="#cb5-53" aria-hidden="true" tabindex="-1"></a>files <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">paste0</span>(GWASwd, <span class="st">"/GWAS_brainAtrophy_TBVstand_N43110.gz"</span>), </span>
<span id="cb5-54"><a href="#cb5-54" aria-hidden="true" tabindex="-1"></a>        <span class="fu">paste0</span>(GWASwd, <span class="st">"/GWAS_brainAtrophy_ICVstand_N43110.gz"</span>))</span>
<span id="cb5-55"><a href="#cb5-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-56"><a href="#cb5-56" aria-hidden="true" tabindex="-1"></a><span class="fu">file.exists</span>(files)</span>
<span id="cb5-57"><a href="#cb5-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-58"><a href="#cb5-58" aria-hidden="true" tabindex="-1"></a>ref <span class="ot">=</span> <span class="st">"reference.1000G.maf.0.005.txt.gz"</span></span>
<span id="cb5-59"><a href="#cb5-59" aria-hidden="true" tabindex="-1"></a>trait.names <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"TBV"</span>,<span class="st">"ICV"</span>)</span>
<span id="cb5-60"><a href="#cb5-60" aria-hidden="true" tabindex="-1"></a><span class="co"># SEs are not on a logistic scale</span></span>
<span id="cb5-61"><a href="#cb5-61" aria-hidden="true" tabindex="-1"></a>se.logit <span class="ot">=</span> <span class="fu">c</span>(F,F)</span>
<span id="cb5-62"><a href="#cb5-62" aria-hidden="true" tabindex="-1"></a>info.filter <span class="ot">=</span> <span class="fl">0.6</span></span>
<span id="cb5-63"><a href="#cb5-63" aria-hidden="true" tabindex="-1"></a>maf.filter <span class="ot">=</span> <span class="fl">0.01</span></span>
<span id="cb5-64"><a href="#cb5-64" aria-hidden="true" tabindex="-1"></a><span class="co">#betas = c("BETA","BETA")</span></span>
<span id="cb5-65"><a href="#cb5-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-66"><a href="#cb5-66" aria-hidden="true" tabindex="-1"></a>sumstats <span class="ot">&lt;-</span> <span class="fu">sumstats</span>(<span class="at">files =</span> files, </span>
<span id="cb5-67"><a href="#cb5-67" aria-hidden="true" tabindex="-1"></a>                        <span class="at">ref =</span> ref, </span>
<span id="cb5-68"><a href="#cb5-68" aria-hidden="true" tabindex="-1"></a>                        <span class="at">trait.names =</span> trait.names, </span>
<span id="cb5-69"><a href="#cb5-69" aria-hidden="true" tabindex="-1"></a>                        <span class="at">se.logit =</span> se.logit, </span>
<span id="cb5-70"><a href="#cb5-70" aria-hidden="true" tabindex="-1"></a>                        <span class="at">info.filter =</span> info.filter, </span>
<span id="cb5-71"><a href="#cb5-71" aria-hidden="true" tabindex="-1"></a>                        <span class="at">maf.filter =</span> maf.filter, </span>
<span id="cb5-72"><a href="#cb5-72" aria-hidden="true" tabindex="-1"></a>                        <span class="co">#betas = betas,</span></span>
<span id="cb5-73"><a href="#cb5-73" aria-hidden="true" tabindex="-1"></a>                        <span class="at">OLS=</span><span class="fu">c</span>(T,T),</span>
<span id="cb5-74"><a href="#cb5-74" aria-hidden="true" tabindex="-1"></a>                        <span class="at">linprob=</span><span class="cn">NULL</span>,<span class="at">N=</span><span class="fu">c</span>(<span class="dv">43110</span>,<span class="dv">43110</span>), </span>
<span id="cb5-75"><a href="#cb5-75" aria-hidden="true" tabindex="-1"></a>                        <span class="at">parallel =</span> F)</span>
<span id="cb5-76"><a href="#cb5-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-77"><a href="#cb5-77" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(sumstats, <span class="at">file=</span><span class="st">"Sumstats_TBV_ICV.RData"</span>)</span>
<span id="cb5-78"><a href="#cb5-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-79"><a href="#cb5-79" aria-hidden="true" tabindex="-1"></a><span class="fu">file.exists</span>(<span class="st">"Sumstats_TBV_ICV.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="snp-level-analysis" class="level3">
<h3 class="anchored" data-anchor-id="snp-level-analysis">SNP-level analysis</h3>
<p>The script below makes it easier to parallelise SNP-level GWAS-by-subtraction analyses in GenomicSEM (make sure you change the model in the script when using this). Run this with: <code>Rscript GWASbySubtractionPARALLEL.R --chr 3 --RData Sumstats_TBV_ICV.RData --LDSCout LDSCoutput_TBV_ICV.RData --outPath GWASbySubtraction</code></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co">#!/usr/bin/Rscript</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Run the 5th step of GWAS-by-subtraction to parallelise each chromosome </span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"optparse"</span>)) <span class="fu">install.packages</span>(<span class="st">"optparse"</span>, <span class="at">repos=</span><span class="st">"https://cran.rstudio.com/"</span>)</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(<span class="st">"optparse"</span>))</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="co"># SET OPTIONS</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>option_list <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">make_option</span>(<span class="st">"--chr"</span>, <span class="at">action =</span> <span class="st">"store"</span>, <span class="at">default =</span> <span class="cn">NA</span>, <span class="at">type =</span> <span class="st">"character"</span>,</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>              <span class="at">help =</span> <span class="st">"Chrosomome to be processed [required]"</span>),</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">make_option</span>(<span class="st">"--trait1"</span>, <span class="at">action =</span> <span class="st">"store"</span>, <span class="at">default =</span> <span class="cn">NA</span>, <span class="at">type =</span> <span class="st">"character"</span>,</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>              <span class="at">help =</span> <span class="st">"Trait 1 [required]"</span>),</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">make_option</span>(<span class="st">"--trait2"</span>, <span class="at">action =</span> <span class="st">"store"</span>, <span class="at">default =</span> <span class="cn">NA</span>, <span class="at">type =</span> <span class="st">"character"</span>,</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>              <span class="at">help =</span> <span class="st">"Trait 2 [required]"</span>),</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">make_option</span>(<span class="st">"--RData"</span>, <span class="at">action =</span> <span class="st">"store"</span>, <span class="at">default =</span> <span class="cn">NA</span>, <span class="at">type =</span> <span class="st">"character"</span>,</span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>              <span class="at">help =</span> <span class="st">"RData containing cleaned sumstats [required]"</span>),</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">make_option</span>(<span class="st">"--LDSCout"</span>, <span class="at">action =</span> <span class="st">"store"</span>, <span class="at">default =</span> <span class="cn">NA</span>, <span class="at">type =</span> <span class="st">"character"</span>,</span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>              <span class="at">help =</span> <span class="st">"LDSC output [required]"</span>),</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">make_option</span>(<span class="st">"--outPath"</span>, <span class="at">action =</span> <span class="st">"store"</span>, <span class="at">default =</span> <span class="cn">NA</span>, <span class="at">type =</span> <span class="st">"character"</span>,</span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>              <span class="at">help =</span> <span class="st">"Output file name [required]"</span>)</span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>opt <span class="ot">=</span> <span class="fu">parse_args</span>(<span class="fu">OptionParser</span>(<span class="at">option_list=</span>option_list))</span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomicSEM)</span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a><span class="co"># FEEDBACK ON OPTIONS</span></span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">Sys.time</span>())</span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"GWAS-by-subtraction CHR = "</span>, opt<span class="sc">$</span>chr))</span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(opt<span class="sc">$</span>RData)</span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(opt<span class="sc">$</span>outPath,<span class="st">"/Atrophy_GWASbySubtraction"</span>, opt<span class="sc">$</span>chr,<span class="st">".RData"</span>))</span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a><span class="do">## Step 5</span></span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a><span class="co"># load LDSC output</span></span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="at">file=</span>opt<span class="sc">$</span>LDSCout)</span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-40"><a href="#cb6-40" aria-hidden="true" tabindex="-1"></a><span class="co"># load in RData file with cleaned SNP data</span></span>
<span id="cb6-41"><a href="#cb6-41" aria-hidden="true" tabindex="-1"></a>file <span class="ot">=</span> opt<span class="sc">$</span>RData</span>
<span id="cb6-42"><a href="#cb6-42" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(file)</span>
<span id="cb6-43"><a href="#cb6-43" aria-hidden="true" tabindex="-1"></a><span class="co"># keep only one chromosome</span></span>
<span id="cb6-44"><a href="#cb6-44" aria-hidden="true" tabindex="-1"></a>sumstats <span class="ot">=</span> sumstats[<span class="fu">which</span>(sumstats<span class="sc">$</span>CHR <span class="sc">==</span> opt<span class="sc">$</span>chr),]</span>
<span id="cb6-45"><a href="#cb6-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-46"><a href="#cb6-46" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">table</span>(sumstats<span class="sc">$</span>CHR))</span>
<span id="cb6-47"><a href="#cb6-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-48"><a href="#cb6-48" aria-hidden="true" tabindex="-1"></a><span class="co"># SNP model</span></span>
<span id="cb6-49"><a href="#cb6-49" aria-hidden="true" tabindex="-1"></a>modelSNP<span class="ot">&lt;-</span><span class="st">'</span></span>
<span id="cb6-50"><a href="#cb6-50" aria-hidden="true" tabindex="-1"></a><span class="st">        BaseICV=~NA*TBV + start(0.8)*ICV</span></span>
<span id="cb6-51"><a href="#cb6-51" aria-hidden="true" tabindex="-1"></a><span class="st">        Atrophy=~NA*TBV</span></span>
<span id="cb6-52"><a href="#cb6-52" aria-hidden="true" tabindex="-1"></a><span class="st">         </span></span>
<span id="cb6-53"><a href="#cb6-53" aria-hidden="true" tabindex="-1"></a><span class="st">        # make sure all variances and covariances are captured by the latent factors</span></span>
<span id="cb6-54"><a href="#cb6-54" aria-hidden="true" tabindex="-1"></a><span class="st">         Atrophy ~~ 1*Atrophy</span></span>
<span id="cb6-55"><a href="#cb6-55" aria-hidden="true" tabindex="-1"></a><span class="st">         BaseICV ~~ 1*BaseICV</span></span>
<span id="cb6-56"><a href="#cb6-56" aria-hidden="true" tabindex="-1"></a><span class="st">         ICV ~~ 0*TBV</span></span>
<span id="cb6-57"><a href="#cb6-57" aria-hidden="true" tabindex="-1"></a><span class="st">         ICV ~~ 0*ICV</span></span>
<span id="cb6-58"><a href="#cb6-58" aria-hidden="true" tabindex="-1"></a><span class="st">         TBV ~~ 0*TBV</span></span>
<span id="cb6-59"><a href="#cb6-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-60"><a href="#cb6-60" aria-hidden="true" tabindex="-1"></a><span class="st">        # forcing atrophy and BaseICV to be uncorrelated</span></span>
<span id="cb6-61"><a href="#cb6-61" aria-hidden="true" tabindex="-1"></a><span class="st">         Atrophy~~0*BaseICV</span></span>
<span id="cb6-62"><a href="#cb6-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-63"><a href="#cb6-63" aria-hidden="true" tabindex="-1"></a><span class="st">        # add in SNP effect</span></span>
<span id="cb6-64"><a href="#cb6-64" aria-hidden="true" tabindex="-1"></a><span class="st">        Atrophy ~ SNP</span></span>
<span id="cb6-65"><a href="#cb6-65" aria-hidden="true" tabindex="-1"></a><span class="st">        BaseICV ~ SNP</span></span>
<span id="cb6-66"><a href="#cb6-66" aria-hidden="true" tabindex="-1"></a><span class="st">        SNP ~~ SNP</span></span>
<span id="cb6-67"><a href="#cb6-67" aria-hidden="true" tabindex="-1"></a><span class="st">'</span></span>
<span id="cb6-68"><a href="#cb6-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-69"><a href="#cb6-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-70"><a href="#cb6-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-71"><a href="#cb6-71" aria-hidden="true" tabindex="-1"></a><span class="co">#Run the Genomic SEM GWAS</span></span>
<span id="cb6-72"><a href="#cb6-72" aria-hidden="true" tabindex="-1"></a>outputGWAS<span class="ot">&lt;-</span><span class="fu">userGWAS</span>(<span class="at">covstruc=</span>LDSCoutput,</span>
<span id="cb6-73"><a href="#cb6-73" aria-hidden="true" tabindex="-1"></a>                    <span class="at">SNPs=</span>sumstats,</span>
<span id="cb6-74"><a href="#cb6-74" aria-hidden="true" tabindex="-1"></a>                    <span class="at">estimation=</span><span class="st">"ML"</span>,</span>
<span id="cb6-75"><a href="#cb6-75" aria-hidden="true" tabindex="-1"></a>                    <span class="at">sub =</span><span class="fu">c</span>(<span class="st">"Atrophy ~ SNP"</span>),</span>
<span id="cb6-76"><a href="#cb6-76" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Q_SNP =</span> T,</span>
<span id="cb6-77"><a href="#cb6-77" aria-hidden="true" tabindex="-1"></a>                    <span class="at">smooth_check =</span> T, </span>
<span id="cb6-78"><a href="#cb6-78" aria-hidden="true" tabindex="-1"></a>                    <span class="at">parallel =</span> T,</span>
<span id="cb6-79"><a href="#cb6-79" aria-hidden="true" tabindex="-1"></a>                    <span class="at">GC =</span> <span class="st">"standard"</span>,</span>
<span id="cb6-80"><a href="#cb6-80" aria-hidden="true" tabindex="-1"></a>                    <span class="at">MPI =</span> F,</span>
<span id="cb6-81"><a href="#cb6-81" aria-hidden="true" tabindex="-1"></a>                    <span class="at">fix_measurement =</span> T,</span>
<span id="cb6-82"><a href="#cb6-82" aria-hidden="true" tabindex="-1"></a>                    <span class="at">model=</span>modelSNP,</span>
<span id="cb6-83"><a href="#cb6-83" aria-hidden="true" tabindex="-1"></a>                    <span class="at">cores =</span> <span class="dv">5</span>,</span>
<span id="cb6-84"><a href="#cb6-84" aria-hidden="true" tabindex="-1"></a>                    <span class="at">toler =</span> F, </span>
<span id="cb6-85"><a href="#cb6-85" aria-hidden="true" tabindex="-1"></a>                    <span class="at">SNPSE =</span> F) <span class="co"># printwarn = FALSE</span></span>
<span id="cb6-86"><a href="#cb6-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-87"><a href="#cb6-87" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(outputGWAS, <span class="at">file=</span><span class="fu">paste0</span>(opt<span class="sc">$</span>outPath,<span class="st">"/Atrophy_GWASbySubtraction_"</span>, opt<span class="sc">$</span>chr,<span class="st">".RData"</span>))</span>
<span id="cb6-88"><a href="#cb6-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-89"><a href="#cb6-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-90"><a href="#cb6-90" aria-hidden="true" tabindex="-1"></a><span class="do">##### Run with </span></span>
<span id="cb6-91"><a href="#cb6-91" aria-hidden="true" tabindex="-1"></a><span class="co"># Rscript GWASbySubtractionPARALLEL.R --chr 3 --RData Sumstats_TBV_ICV.RData --LDSCout LDSCoutput_TBV_ICV.RData --outPath GWASbySubtraction</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>