<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.189">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Anna Elisabeth Furtjes">
<meta name="dcterms.date" content="2025-03-20">

<title>Brain Atrophy Genetics - UK Biobank: Genetic data cleaning</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Brain Atrophy Genetics</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html">Overview</a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#initial-sample-level-quality-control-qc" id="toc-initial-sample-level-quality-control-qc" class="nav-link active" data-scroll-target="#initial-sample-level-quality-control-qc">Initial sample-level quality control (QC)</a></li>
  <li><a href="#format-genetic-data" id="toc-format-genetic-data" class="nav-link" data-scroll-target="#format-genetic-data">Format genetic data</a></li>
  <li><a href="#filter-for-missing-genotype-data-and-minor-allele-frequency" id="toc-filter-for-missing-genotype-data-and-minor-allele-frequency" class="nav-link" data-scroll-target="#filter-for-missing-genotype-data-and-minor-allele-frequency">Filter for missing genotype data and minor allele frequency</a></li>
  <li><a href="#remove-related-individuals" id="toc-remove-related-individuals" class="nav-link" data-scroll-target="#remove-related-individuals">Remove related individuals</a></li>
  <li><a href="#perform-sex-check" id="toc-perform-sex-check" class="nav-link" data-scroll-target="#perform-sex-check">Perform sex-check</a>
  <ul class="collapse">
  <li><a href="#remove-individuals-that-did-not-pass-sex-check" id="toc-remove-individuals-that-did-not-pass-sex-check" class="nav-link" data-scroll-target="#remove-individuals-that-did-not-pass-sex-check">Remove individuals that did not pass sex-check</a></li>
  </ul></li>
  <li><a href="#generate-final-.fam-and-.snplist-files" id="toc-generate-final-.fam-and-.snplist-files" class="nav-link" data-scroll-target="#generate-final-.fam-and-.snplist-files">Generate final .fam and .snplist files</a></li>
  <li><a href="#generate-final-bed-file" id="toc-generate-final-bed-file" class="nav-link" data-scroll-target="#generate-final-bed-file">Generate final bed file</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">UK Biobank: Genetic data cleaning</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Anna Elisabeth Furtjes </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">March 20, 2025</p>
    </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Doi</div>
    <div class="quarto-title-meta-contents">
      <p class="doi">
        <a href="https://doi.org/10.1101/2024.11.06.622274">10.1101/2024.11.06.622274</a>
      </p>
    </div>
  </div>
  </div>
  

</header>

<hr>
<section id="initial-sample-level-quality-control-qc" class="level2">
<h2 class="anchored" data-anchor-id="initial-sample-level-quality-control-qc">Initial sample-level quality control (QC)</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Initial sample QC (R code)</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="do">## 1. Do 4 means clustering to get europeans only</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="do">## 2. Get variable with het &amp; miss info</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="do">## 3. Generate a keep list </span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="co"># load packages</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="co"># get all participant IDs that have neuroimaging data (this is the phenotypic data prepared as GWAS input)</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>neuro <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">"UKB_CrossNeuroIDP.txt"</span>, <span class="at">select =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="co"># get PC1:40 data using ukbtools package</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>ukb_sqc<span class="ot">=</span><span class="fu">fread</span>(<span class="st">"UKBB_500K/Decrypted/ukb_sqc_v2.txt"</span>)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="co"># https://kenhanscombe.github.io/ukbtools/articles/explore-ukb-data.html</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="co"># With ukb_sqc_v2.txt read into the dataframe my_sqc_data</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>ukb_sqc <span class="ot">&lt;-</span> <span class="fu">ukb_gen_sqc_names</span>(ukb_sqc)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="co"># genetic file is organised in the same order as application-specific ref fam file</span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>ref <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">"UKBB_500K/Decrypted/ukb1027_cal_chr1_v2_s488374.fam"</span>)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="co"># merge ref file to get IDs</span></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>ukb_sqc <span class="ot">=</span> <span class="fu">cbind</span>(ref[,<span class="fu">c</span>(<span class="st">"V1"</span>)], ukb_sqc)</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="co"># re-name ID column that comes from the fam file</span></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ukb_sqc)[<span class="fu">which</span>(<span class="fu">names</span>(ukb_sqc) <span class="sc">==</span> <span class="st">"V1"</span>)] <span class="ot">=</span> <span class="st">"IID"</span></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Get variable with het &amp; miss info</span></span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a><span class="do">###############################</span></span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(ukb_sqc<span class="sc">$</span>het_missing_outliers)</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">"We lose "</span>,<span class="fu">nrow</span>(neuro)<span class="sc">-</span><span class="fu">sum</span>(neuro<span class="sc">$</span>IID <span class="sc">%in%</span> ukb_sqc<span class="sc">$</span>IID), <span class="st">" participants from the neuroimaging data set because these IDs are not available in the genetic data. (N = "</span>, <span class="fu">sum</span>(neuro<span class="sc">$</span>IID <span class="sc">%in%</span> ukb_sqc<span class="sc">$</span>IID),<span class="st">")"</span>))</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] "We lose  1220  participants from the neuroimaging data set because these IDs are not available in the genetic data. (N =  45598 )"</span></span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a>neuro <span class="ot">=</span> <span class="fu">merge</span>(neuro, ukb_sqc[,<span class="fu">c</span>(<span class="st">"IID"</span>,<span class="st">"het_missing_outliers"</span>)], <span class="at">by =</span> <span class="st">"IID"</span>)</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">"We lose "</span>, <span class="fu">sum</span>(neuro<span class="sc">$</span>het_missing_outliers <span class="sc">==</span> <span class="dv">1</span>),<span class="st">" participants because they were outliers in heterozygosity and missinggness. N = "</span>, <span class="fu">sum</span>(neuro<span class="sc">$</span>het_missing_outliers <span class="sc">==</span> <span class="dv">0</span>), <span class="st">""</span>))</span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] "We lose  83  participants because they were outliers in heterozygosity and missinggness. N =  45515 "</span></span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a><span class="co"># exclude particpants that were labeled as outliers by UKB core team</span></span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a>neuro <span class="ot">=</span> neuro[<span class="fu">which</span>(neuro<span class="sc">$</span>het_missing_outlier <span class="sc">==</span> <span class="dv">0</span>),]</span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a><span class="do">################################</span></span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a><span class="co"># do 4 means clustering</span></span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a><span class="do">################################</span></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a><span class="do">##Read in PCs, remove NAs, rename</span></span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a><span class="co">#PCs&lt;-fread(args[1], data.table=F)</span></span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a>colNames <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"V1"</span>, <span class="fu">paste0</span>(<span class="st">"pc"</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">40</span>))</span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a>PCs <span class="ot">&lt;-</span> ukb_sqc[, ..colNames]</span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a>PCs<span class="ot">&lt;-</span><span class="fu">na.omit</span>(PCs)</span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(PCs)<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">"ID"</span>,<span class="st">"PC.1"</span>,<span class="st">"PC.2"</span>,<span class="st">"PC.3"</span>,<span class="st">"PC.4"</span>,<span class="st">"PC.5"</span>,<span class="st">"PC.6"</span>,<span class="st">"PC.7"</span>,<span class="st">"PC.8"</span>,<span class="st">"PC.9"</span>,<span class="st">"PC.10"</span>,<span class="st">"PC.11"</span>,<span class="st">"PC.12"</span>,<span class="st">"PC.13"</span>,<span class="st">"PC.14"</span>,<span class="st">"PC.15"</span>,<span class="st">"PC.16"</span>,<span class="st">"PC.17"</span>,<span class="st">"PC.18"</span>,<span class="st">"PC.19"</span>,<span class="st">"PC.20"</span>,<span class="st">"PC.21"</span>,<span class="st">"PC.22"</span>,<span class="st">"PC.23"</span>,<span class="st">"PC.24"</span>,<span class="st">"PC.25"</span>,<span class="st">"PC.26"</span>,<span class="st">"PC.27"</span>,<span class="st">"PC.28"</span>,<span class="st">"PC.29"</span>,<span class="st">"PC.30"</span>,<span class="st">"PC.31"</span>,<span class="st">"PC.32"</span>,<span class="st">"PC.33"</span>,<span class="st">"PC.34"</span>,<span class="st">"PC.35"</span>,<span class="st">"PC.36"</span>,<span class="st">"PC.37"</span>,<span class="st">"PC.38"</span>,<span class="st">"PC.39"</span>,<span class="st">"PC.40"</span>)</span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a><span class="do">##Set seed</span></span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1204688</span>)</span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a><span class="do">##K means clustering on each PC</span></span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a>K_MEAN <span class="ot">&lt;-</span> <span class="dv">4</span></span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a>PC1_K<span class="ot">&lt;-</span><span class="fu">kmeans</span>(PCs<span class="sc">$</span>PC<span class="fl">.1</span>, K_MEAN)</span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a>PC2_K<span class="ot">&lt;-</span><span class="fu">kmeans</span>(PCs<span class="sc">$</span>PC<span class="fl">.2</span>, K_MEAN)</span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a><span class="do">##Add clusters to PC dataframe</span></span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a>PCs<span class="sc">$</span>PC1.Cluster<span class="ot">&lt;-</span>PC1_K<span class="sc">$</span>cluster</span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a>PCs<span class="sc">$</span>PC2.Cluster<span class="ot">&lt;-</span>PC2_K<span class="sc">$</span>cluster</span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a>PCs<span class="sc">$</span>Clusters<span class="ot">&lt;-</span><span class="fu">as.factor</span>(<span class="fu">paste</span>(PC1_K<span class="sc">$</span>cluster,PC2_K<span class="sc">$</span>cluster,<span class="at">sep=</span><span class="st">"."</span>))</span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a><span class="do">##WWE group is the majority</span></span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a>MAX_PC1<span class="ot">&lt;-</span><span class="fu">ifelse</span>(<span class="fu">match</span>(<span class="fu">max</span>(<span class="fu">table</span>(PCs<span class="sc">$</span>PC1.Cluster, PCs<span class="sc">$</span>PC2.Cluster)), <span class="fu">table</span>(PCs<span class="sc">$</span>PC1.Cluster, PCs<span class="sc">$</span>PC2.Cluster)) <span class="sc">%%</span> K_MEAN <span class="sc">==</span> <span class="dv">0</span>, K_MEAN, <span class="fu">match</span>(<span class="fu">max</span>(<span class="fu">table</span>(PCs<span class="sc">$</span>PC1.Cluster, PCs<span class="sc">$</span>PC2.Cluster)), <span class="fu">table</span>(PCs<span class="sc">$</span>PC1.Cluster, PCs<span class="sc">$</span>PC2.Cluster)) <span class="sc">%%</span> K_MEAN)</span>
<span id="cb1-71"><a href="#cb1-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-72"><a href="#cb1-72" aria-hidden="true" tabindex="-1"></a>MAX_PC2<span class="ot">&lt;-</span><span class="fu">ceiling</span>(<span class="fu">match</span>(<span class="fu">max</span>(<span class="fu">table</span>(PCs<span class="sc">$</span>PC1.Cluster, PCs<span class="sc">$</span>PC2.Cluster)), <span class="fu">table</span>(PCs<span class="sc">$</span>PC1.Cluster, PCs<span class="sc">$</span>PC2.Cluster))<span class="sc">/</span>K_MEAN)</span>
<span id="cb1-73"><a href="#cb1-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-74"><a href="#cb1-74" aria-hidden="true" tabindex="-1"></a><span class="do">##Make lists of WWE IDs</span></span>
<span id="cb1-75"><a href="#cb1-75" aria-hidden="true" tabindex="-1"></a>WWE<span class="ot">&lt;-</span><span class="fu">as.data.frame</span>(PCs[PCs<span class="sc">$</span>PC1.Cluster <span class="sc">==</span> MAX_PC1 <span class="sc">&amp;</span> PCs<span class="sc">$</span>PC2.Cluster <span class="sc">==</span> MAX_PC2,<span class="dv">1</span>])</span>
<span id="cb1-76"><a href="#cb1-76" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(WWE)<span class="ot">&lt;-</span><span class="st">"ID"</span></span>
<span id="cb1-77"><a href="#cb1-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-78"><a href="#cb1-78" aria-hidden="true" tabindex="-1"></a><span class="do">## Get overlap with participants of interest</span></span>
<span id="cb1-79"><a href="#cb1-79" aria-hidden="true" tabindex="-1"></a>WWE <span class="ot">=</span> WWE[WWE<span class="sc">$</span>ID <span class="sc">%in%</span> neuro<span class="sc">$</span>FID,]</span>
<span id="cb1-80"><a href="#cb1-80" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">"We excluded "</span>, <span class="fu">nrow</span>(neuro)<span class="sc">-</span><span class="fu">length</span>(WWE),<span class="st">"participants based on 4-means clustering. N = "</span>, <span class="fu">length</span>(WWE), <span class="st">""</span>))</span>
<span id="cb1-81"><a href="#cb1-81" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] "We excluded  1332 participants based on 4-means clustering. N =  44183 "</span></span>
<span id="cb1-82"><a href="#cb1-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-83"><a href="#cb1-83" aria-hidden="true" tabindex="-1"></a>WWE_PLINK<span class="ot">&lt;-</span><span class="fu">as.data.frame</span>(<span class="fu">cbind</span>(WWE,WWE))</span>
<span id="cb1-84"><a href="#cb1-84" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(WWE_PLINK)<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">"FID"</span>,<span class="st">"IID"</span>)</span>
<span id="cb1-85"><a href="#cb1-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-86"><a href="#cb1-86" aria-hidden="true" tabindex="-1"></a><span class="do">##Write to file</span></span>
<span id="cb1-87"><a href="#cb1-87" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(WWE_PLINK, <span class="at">file=</span><span class="fu">paste0</span>(out,<span class="st">"/ukb_neuroimaging_4MeansClustering_excl_het_miss.txt"</span>), <span class="at">row.names=</span>F, <span class="at">col.names=</span>T, <span class="at">quote =</span> F)</span>
<span id="cb1-88"><a href="#cb1-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-89"><a href="#cb1-89" aria-hidden="true" tabindex="-1"></a><span class="do">##Plot</span></span>
<span id="cb1-90"><a href="#cb1-90" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="fu">paste0</span>(out,<span class="st">"/ukb_neuroimaging_4MeansClustering"</span>, <span class="st">".pdf"</span>))</span>
<span id="cb1-91"><a href="#cb1-91" aria-hidden="true" tabindex="-1"></a><span class="fu">with</span>(PCs, <span class="fu">print</span>(<span class="fu">qplot</span>(PC<span class="fl">.1</span>, PC<span class="fl">.2</span>, <span class="at">colour=</span>Clusters)))</span>
<span id="cb1-92"><a href="#cb1-92" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb1-93"><a href="#cb1-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-94"><a href="#cb1-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-95"><a href="#cb1-95" aria-hidden="true" tabindex="-1"></a><span class="do">#############################</span></span>
<span id="cb1-96"><a href="#cb1-96" aria-hidden="true" tabindex="-1"></a><span class="do">### check how many of those participants now have self-reported different ancestry</span></span>
<span id="cb1-97"><a href="#cb1-97" aria-hidden="true" tabindex="-1"></a><span class="do">#############################</span></span>
<span id="cb1-98"><a href="#cb1-98" aria-hidden="true" tabindex="-1"></a><span class="co"># this function identifies the path to the most recent download file on our server</span></span>
<span id="cb1-99"><a href="#cb1-99" aria-hidden="true" tabindex="-1"></a>getFieldLoc <span class="ot">=</span> <span class="cf">function</span>(<span class="at">path =</span> path, <span class="at">fileName =</span> fileName, <span class="at">fieldID =</span> fieldID){</span>
<span id="cb1-100"><a href="#cb1-100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(stringr)</span>
<span id="cb1-101"><a href="#cb1-101" aria-hidden="true" tabindex="-1"></a>  <span class="co"># read all the field.ukb files</span></span>
<span id="cb1-102"><a href="#cb1-102" aria-hidden="true" tabindex="-1"></a>  files_to_read <span class="ot">=</span> <span class="fu">list.files</span>(</span>
<span id="cb1-103"><a href="#cb1-103" aria-hidden="true" tabindex="-1"></a>    <span class="at">path =</span> path,</span>
<span id="cb1-104"><a href="#cb1-104" aria-hidden="true" tabindex="-1"></a>    <span class="at">pattern =</span> fileName,</span>
<span id="cb1-105"><a href="#cb1-105" aria-hidden="true" tabindex="-1"></a>    <span class="at">recursive =</span> T,</span>
<span id="cb1-106"><a href="#cb1-106" aria-hidden="true" tabindex="-1"></a>    <span class="at">full.names =</span> T</span>
<span id="cb1-107"><a href="#cb1-107" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb1-108"><a href="#cb1-108" aria-hidden="true" tabindex="-1"></a>  <span class="co"># read all files</span></span>
<span id="cb1-109"><a href="#cb1-109" aria-hidden="true" tabindex="-1"></a>  dat <span class="ot">=</span> <span class="fu">lapply</span>(files_to_read, fread)</span>
<span id="cb1-110"><a href="#cb1-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(dat) <span class="ot">=</span> files_to_read</span>
<span id="cb1-111"><a href="#cb1-111" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-112"><a href="#cb1-112" aria-hidden="true" tabindex="-1"></a>  <span class="co"># search for field ID of interest </span></span>
<span id="cb1-113"><a href="#cb1-113" aria-hidden="true" tabindex="-1"></a>  candidates <span class="ot">=</span> <span class="fu">names</span>(dat)[<span class="fu">grep</span>(fieldID, dat)]</span>
<span id="cb1-114"><a href="#cb1-114" aria-hidden="true" tabindex="-1"></a>  <span class="co"># figure out which one is from the most recent file (i.e., highest number)</span></span>
<span id="cb1-115"><a href="#cb1-115" aria-hidden="true" tabindex="-1"></a>  candidates <span class="ot">=</span> <span class="fu">str_remove</span>(candidates, <span class="fu">paste0</span>(path, <span class="st">"/"</span>))</span>
<span id="cb1-116"><a href="#cb1-116" aria-hidden="true" tabindex="-1"></a>  candidates <span class="ot">=</span> <span class="fu">str_remove</span>(candidates, <span class="fu">paste0</span>(<span class="st">"/"</span>, fileName))</span>
<span id="cb1-117"><a href="#cb1-117" aria-hidden="true" tabindex="-1"></a>  candidates <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">as.numeric</span>(<span class="fu">sapply</span>(<span class="fu">str_extract_all</span>(candidates, <span class="st">"</span><span class="sc">\\</span><span class="st">d+"</span>), tail , <span class="dv">1</span>)))</span>
<span id="cb1-118"><a href="#cb1-118" aria-hidden="true" tabindex="-1"></a>  most_recent <span class="ot">=</span> <span class="fu">max</span>(candidates, <span class="at">na.rm =</span>T)</span>
<span id="cb1-119"><a href="#cb1-119" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-120"><a href="#cb1-120" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">paste0</span>(path, most_recent))</span>
<span id="cb1-121"><a href="#cb1-121" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-122"><a href="#cb1-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-123"><a href="#cb1-123" aria-hidden="true" tabindex="-1"></a>path1 <span class="ot">=</span> <span class="fu">getFieldLoc</span>(<span class="at">path =</span> path, </span>
<span id="cb1-124"><a href="#cb1-124" aria-hidden="true" tabindex="-1"></a>            <span class="at">fileName =</span> <span class="st">"fields.ukb"</span>, </span>
<span id="cb1-125"><a href="#cb1-125" aria-hidden="true" tabindex="-1"></a>            <span class="at">fieldID =</span> <span class="dv">21000</span>)</span>
<span id="cb1-126"><a href="#cb1-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-127"><a href="#cb1-127" aria-hidden="true" tabindex="-1"></a>fileID <span class="ot">=</span> <span class="fu">list.files</span>(<span class="at">pat=</span>path1,<span class="at">pattern=</span><span class="st">"csv"</span>)</span>
<span id="cb1-128"><a href="#cb1-128" aria-hidden="true" tabindex="-1"></a><span class="co"># read in file</span></span>
<span id="cb1-129"><a href="#cb1-129" aria-hidden="true" tabindex="-1"></a>file <span class="ot">=</span> <span class="fu">fread</span>(<span class="fu">paste0</span>(path1, <span class="st">"/"</span>, fileID))</span>
<span id="cb1-130"><a href="#cb1-130" aria-hidden="true" tabindex="-1"></a><span class="co"># file doesnt like column names that start with number and it doesnt like -</span></span>
<span id="cb1-131"><a href="#cb1-131" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(file) <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">"f."</span>,<span class="fu">names</span>(file))</span>
<span id="cb1-132"><a href="#cb1-132" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(file) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">"-"</span>, <span class="st">"_"</span>, <span class="fu">names</span>(file), <span class="at">fixed =</span> T)</span>
<span id="cb1-133"><a href="#cb1-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-134"><a href="#cb1-134" aria-hidden="true" tabindex="-1"></a><span class="co"># list columns of interest</span></span>
<span id="cb1-135"><a href="#cb1-135" aria-hidden="true" tabindex="-1"></a>id<span class="ot">=</span><span class="fu">which</span>(<span class="fu">names</span>(file) <span class="sc">==</span> <span class="st">"f.eid"</span>)</span>
<span id="cb1-136"><a href="#cb1-136" aria-hidden="true" tabindex="-1"></a>Cols <span class="ot">=</span> <span class="fu">grep</span>(<span class="st">"f.21000"</span>, <span class="fu">names</span>(file))</span>
<span id="cb1-137"><a href="#cb1-137" aria-hidden="true" tabindex="-1"></a>file <span class="ot">=</span> file[, <span class="fu">c</span>(..id, ..Cols)]</span>
<span id="cb1-138"><a href="#cb1-138" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(file)[<span class="fu">which</span>(<span class="fu">names</span>(file) <span class="sc">==</span> <span class="st">"f.eid"</span>)] <span class="ot">=</span> <span class="st">"IID"</span></span>
<span id="cb1-139"><a href="#cb1-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-140"><a href="#cb1-140" aria-hidden="true" tabindex="-1"></a><span class="co"># merge file with neuro but only keep the IDs available in neuro</span></span>
<span id="cb1-141"><a href="#cb1-141" aria-hidden="true" tabindex="-1"></a>neuro1 <span class="ot">=</span> <span class="fu">merge</span>(neuro, file, <span class="at">by =</span> <span class="st">"IID"</span>, <span class="at">all.x=</span>T)</span>
<span id="cb1-142"><a href="#cb1-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-143"><a href="#cb1-143" aria-hidden="true" tabindex="-1"></a><span class="co"># make a table for the remaining particpants to see what ethnicity they reported</span></span>
<span id="cb1-144"><a href="#cb1-144" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(neuro1<span class="sc">$</span>f<span class="fl">.21000</span>_0<span class="fl">.0</span>)</span>
<span id="cb1-145"><a href="#cb1-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-146"><a href="#cb1-146" aria-hidden="true" tabindex="-1"></a><span class="co"># make variable that encodes European ancestry</span></span>
<span id="cb1-147"><a href="#cb1-147" aria-hidden="true" tabindex="-1"></a>neuro1<span class="sc">$</span>European <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(neuro1<span class="sc">$</span>f<span class="fl">.21000</span>_0<span class="fl">.0</span> <span class="sc">==</span> <span class="dv">1</span> <span class="sc">|</span> neuro1<span class="sc">$</span>f<span class="fl">.21000</span>_0<span class="fl">.0</span> <span class="sc">==</span> <span class="dv">1001</span> <span class="sc">|</span> neuro1<span class="sc">$</span>f<span class="fl">.21000</span>_0<span class="fl">.0</span> <span class="sc">==</span> <span class="dv">1002</span> <span class="sc">|</span> neuro1<span class="sc">$</span>f<span class="fl">.21000</span>_0<span class="fl">.0</span> <span class="sc">==</span> <span class="dv">1003</span>,<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb1-148"><a href="#cb1-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-149"><a href="#cb1-149" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">"We're losing "</span>, <span class="fu">sum</span>(neuro1<span class="sc">$</span>European <span class="sc">==</span> <span class="dv">0</span>, <span class="at">na.rm=</span>T),<span class="st">" participants because they self-reported to be non-European."</span>,<span class="fu">sum</span>(<span class="fu">is.na</span>(neuro1<span class="sc">$</span>European)),<span class="st">"are missing this info and will also be excluded. N remaining = "</span>, <span class="fu">sum</span>(neuro1<span class="sc">$</span>European <span class="sc">==</span><span class="dv">1</span>, <span class="at">na.rm=</span>T)))</span>
<span id="cb1-150"><a href="#cb1-150" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] "We're losing  1470  participants because they self-reported to be non-European. 9 are missing this info and will also be excluded. N remaining =  44036"</span></span>
<span id="cb1-151"><a href="#cb1-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-152"><a href="#cb1-152" aria-hidden="true" tabindex="-1"></a><span class="co"># delete non-European</span></span>
<span id="cb1-153"><a href="#cb1-153" aria-hidden="true" tabindex="-1"></a>neuro1 <span class="ot">=</span> neuro1[<span class="fu">which</span>(neuro1<span class="sc">$</span>European <span class="sc">==</span> <span class="dv">1</span>),]</span>
<span id="cb1-154"><a href="#cb1-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-155"><a href="#cb1-155" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(neuro1, <span class="at">file=</span><span class="fu">paste0</span>(out,<span class="st">"/ukb_neuroimaging_4MeansClustering_excl_het_missEUR.txt"</span>), <span class="at">row.names=</span>F, <span class="at">col.names=</span>T, <span class="at">quote =</span> F)</span>
<span id="cb1-156"><a href="#cb1-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-157"><a href="#cb1-157" aria-hidden="true" tabindex="-1"></a><span class="do">##### At this point, we now have a list of IDs saved in ukb_neuroimaging_4MeansClustering_excl_het_miss.txt that are:</span></span>
<span id="cb1-158"><a href="#cb1-158" aria-hidden="true" tabindex="-1"></a><span class="co"># People with neuroimaging data</span></span>
<span id="cb1-159"><a href="#cb1-159" aria-hidden="true" tabindex="-1"></a><span class="co"># Have genetic data</span></span>
<span id="cb1-160"><a href="#cb1-160" aria-hidden="true" tabindex="-1"></a><span class="co"># Have not been labeled unusual by the UKB core team for missingness and heterozygosity</span></span>
<span id="cb1-161"><a href="#cb1-161" aria-hidden="true" tabindex="-1"></a><span class="co"># Have been included based on 4 means clustering</span></span>
<span id="cb1-162"><a href="#cb1-162" aria-hidden="true" tabindex="-1"></a><span class="co"># Have self-reported that they are of European descent</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="format-genetic-data" class="level2">
<h2 class="anchored" data-anchor-id="format-genetic-data">Format genetic data</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co">##########################################################################################################</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 0: Only include participants with neuroimaging data </span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="co">## this set of individuals will already exclude non-Europeans (PCs and self-report)</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co">## extreme scores on hetezygosity and missingness (see above)</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="co"># copy bim bed and fam files to get consistent naming across bim and bed files</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="co">## bed files</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> CHR <span class="kw">in</span> <span class="dt">{</span><span class="dv">1</span><span class="dt">..</span><span class="dv">22</span><span class="dt">}</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> <span class="va">${sourceBED}</span>/ukb_cal_chr<span class="va">${CHR}</span>_v2.bed <span class="va">${target}</span>/ukb_chr<span class="va">${CHR}</span>_v2.bed</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="co">## bim files</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> CHR <span class="kw">in</span> <span class="dt">{</span><span class="dv">1</span><span class="dt">..</span><span class="dv">22</span><span class="dt">}</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> <span class="va">${sourceBIM}</span>/ukb_snp_chr<span class="va">${CHR}</span>_v2.bim <span class="va">${target}</span>/ukb_chr<span class="va">${CHR}</span>_v2.bim</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a><span class="co">## fam file</span></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> /GWAS_Source/UB_BB/UKBB_500K/Decrypted/ukb1027_cal_chr1_v2_s488374.fam <span class="va">${target}</span>/ukb_v2.fam</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a><span class="co"># do fam file for each chromosome</span></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> CHR <span class="kw">in</span> <span class="dt">{</span><span class="dv">1</span><span class="dt">..</span><span class="dv">22</span><span class="dt">}</span></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> <span class="va">${target}</span>/ukb_v2.fam <span class="va">${target}</span>/ukb_chr<span class="va">${CHR}</span>_v2.fam</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a><span class="co">## put data from each chromosome together into one bim, bed and fam file</span></span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a><span class="co">## and filter participants of interest</span></span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="va">$target</span></span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a><span class="va">phenoIDs</span><span class="op">=</span><span class="st">"/CCACE_Shared/Anna_F/BrainAtrophy/data/geneticQC/ukb_neuroimaging_4MeansClustering_excl_het_missEUR.txt"</span></span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a><span class="ex">plink19</span> <span class="dt">\</span></span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>--merge-list <span class="va">$target</span>/allGenoFiles.txt <span class="dt">\</span></span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>--make-bed <span class="dt">\</span></span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>--keep <span class="va">$phenoIDs</span> <span class="dt">\</span></span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>--out ukb_neuroimaging_reQC</span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a><span class="co">## starting out with 784256 variants and 488377 participants, that are getting cut down to the 44036 participants that survived cleaning above</span></span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a><span class="co">#################################################</span></span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a><span class="co"># I realised later that this is not including the sex-chromosomes which are needed for sex-check later (this merging back and forth could have been done in one step, but I am saving resources to not to it from scratch again)</span></span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a><span class="co"># Copy X, Y and XY chromosome info first</span></span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a><span class="co">## bed files</span></span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> CHR <span class="kw">in</span> X XY Y</span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> <span class="va">${sourceBED}</span>/ukb_cal_chr<span class="va">${CHR}</span>_v2.bed <span class="va">${target}</span>/ukb_chr<span class="va">${CHR}</span>_v2.bed</span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb2-47"><a href="#cb2-47" aria-hidden="true" tabindex="-1"></a><span class="co">## bim files</span></span>
<span id="cb2-48"><a href="#cb2-48" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> CHR <span class="kw">in</span> X XY Y</span>
<span id="cb2-49"><a href="#cb2-49" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb2-50"><a href="#cb2-50" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> <span class="va">${sourceBIM}</span>/ukb_snp_chr<span class="va">${CHR}</span>_v2.bim <span class="va">${target}</span>/ukb_chr<span class="va">${CHR}</span>_v2.bim</span>
<span id="cb2-51"><a href="#cb2-51" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb2-52"><a href="#cb2-52" aria-hidden="true" tabindex="-1"></a><span class="co">## fam file for each chromosome</span></span>
<span id="cb2-53"><a href="#cb2-53" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> CHR <span class="kw">in</span> X XY Y</span>
<span id="cb2-54"><a href="#cb2-54" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb2-55"><a href="#cb2-55" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> <span class="va">${target}</span>/ukb_v2.fam <span class="va">${target}</span>/ukb_chr<span class="va">${CHR}</span>_v2.fam</span>
<span id="cb2-56"><a href="#cb2-56" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb2-57"><a href="#cb2-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-58"><a href="#cb2-58" aria-hidden="true" tabindex="-1"></a><span class="co">## merge X, Y, XY files together</span></span>
<span id="cb2-59"><a href="#cb2-59" aria-hidden="true" tabindex="-1"></a><span class="ex">plink19</span> <span class="dt">\</span></span>
<span id="cb2-60"><a href="#cb2-60" aria-hidden="true" tabindex="-1"></a>--merge-list <span class="va">${target}</span>/allXYfiles.txt <span class="dt">\</span></span>
<span id="cb2-61"><a href="#cb2-61" aria-hidden="true" tabindex="-1"></a>--make-bed <span class="dt">\</span></span>
<span id="cb2-62"><a href="#cb2-62" aria-hidden="true" tabindex="-1"></a>--keep <span class="va">$phenoIDs</span> <span class="dt">\</span></span>
<span id="cb2-63"><a href="#cb2-63" aria-hidden="true" tabindex="-1"></a>--out ukb_neuroimaging_sexchrom</span>
<span id="cb2-64"><a href="#cb2-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-65"><a href="#cb2-65" aria-hidden="true" tabindex="-1"></a><span class="co"># now merge sex chromosomes with other data</span></span>
<span id="cb2-66"><a href="#cb2-66" aria-hidden="true" tabindex="-1"></a><span class="ex">plink19</span> <span class="dt">\</span></span>
<span id="cb2-67"><a href="#cb2-67" aria-hidden="true" tabindex="-1"></a>--merge-list <span class="va">${target}</span>/allAuto_and_Sex.txt <span class="dt">\</span></span>
<span id="cb2-68"><a href="#cb2-68" aria-hidden="true" tabindex="-1"></a>--make-bed <span class="dt">\</span></span>
<span id="cb2-69"><a href="#cb2-69" aria-hidden="true" tabindex="-1"></a>--keep <span class="va">$phenoIDs</span> <span class="dt">\</span></span>
<span id="cb2-70"><a href="#cb2-70" aria-hidden="true" tabindex="-1"></a>--out ukb_neuroimaging_autosomal_sex_preQC</span>
<span id="cb2-71"><a href="#cb2-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-72"><a href="#cb2-72" aria-hidden="true" tabindex="-1"></a><span class="co">## now altogether, we have 44036 partcicipants left, and 805161 variants </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="filter-for-missing-genotype-data-and-minor-allele-frequency" class="level2">
<h2 class="anchored" data-anchor-id="filter-for-missing-genotype-data-and-minor-allele-frequency">Filter for missing genotype data and minor allele frequency</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 1: Get SNP-list based on geno filter</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="ex">plink19</span> <span class="dt">\</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>--bed <span class="va">${target}</span>/ukb_neuroimaging_autosomal_sex_preQC.bed <span class="dt">\</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>--bim <span class="va">${target}</span>/ukb_neuroimaging_autosomal_sex_preQC.bim <span class="dt">\</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>--fam <span class="va">${target}</span>/ukb_neuroimaging_autosomal_sex_preQC.fam <span class="dt">\</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>--geno 0.02 <span class="dt">\</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>--write-snplist <span class="dt">\</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>--out <span class="va">${target}</span>/ukb_neuroimaging_GENO0.02</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="co"># removed 104462 variants, 700699 remaining</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="co"># no participants removed (44036 remaining)</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="co">## this creates a snplist that feeds into next step</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 2: apply various cleaning steps to get ID list (.fam) to feed into relatedness analysis</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a><span class="ex">plink19</span> <span class="dt">\</span></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>--bed <span class="va">${target}</span>/ukb_neuroimaging_autosomal_sex_preQC.bed <span class="dt">\</span></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>--bim <span class="va">${target}</span>/ukb_neuroimaging_autosomal_sex_preQC.bim <span class="dt">\</span></span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>--fam <span class="va">${target}</span>/ukb_neuroimaging_autosomal_sex_preQC.fam <span class="dt">\</span></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>--geno 0.02 <span class="dt">\</span></span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>--extract ukb_neuroimaging_GENO0.02.snplist <span class="dt">\</span></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>--write-snplist <span class="dt">\</span></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>--make-just-fam <span class="dt">\</span></span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>--freq <span class="at">--maf</span> 0.01 <span class="dt">\</span></span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>--out ukb_neuroimaging_MAF0.01_GENO0.02_QA_EUR</span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a><span class="co"># this fam file still contains 44036 participants</span></span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a><span class="co"># MAF filter removed 103158 variants, 597541 remaining</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="remove-related-individuals" class="level2">
<h2 class="anchored" data-anchor-id="remove-related-individuals">Remove related individuals</h2>
<p>Our server was incompatible with the greedyRelated software, which is why I use R package here instead that runs the same analysis.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ukbtools)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co"># read in ukb relatedness file</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>rel <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">"/UKBB_500K/Decrypted/ukb1027_rel_s488374.dat"</span>)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>neuro <span class="ot">=</span> <span class="fu">fread</span>(<span class="st">"ukb_neuroimaging_MAF0.01_GENO0.02_QA_EUR.fam"</span>)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(neuro<span class="sc">$</span>V1 <span class="sc">%in%</span> rel<span class="sc">$</span>ID1)</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 7509</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(neuro<span class="sc">$</span>V1 <span class="sc">%in%</span> rel<span class="sc">$</span>ID2)</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="co"># 7748</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="co"># identify degree of relatedness </span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="fu">ukb_gen_rel_count</span>(rel)</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="fu">ukb_gen_rel_count</span>(rel, <span class="at">plot =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="co"># generate a list of participants to exclude based on relatedness</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a><span class="co"># default cut-off &gt; 0.0884 King coefficient corresponding to 3rd degree relatedness</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>IDtoRemove <span class="ot">=</span> <span class="fu">ukb_gen_samples_to_remove</span>(rel, <span class="at">ukb_with_data =</span> neuro<span class="sc">$</span>V1)</span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a><span class="co"># get PLINK format</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>IDtoRemove<span class="ot">&lt;-</span><span class="fu">as.data.frame</span>(<span class="fu">cbind</span>(IDtoRemove,IDtoRemove))</span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(IDtoRemove)<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">"FID"</span>,<span class="st">"IID"</span>)</span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a><span class="co"># write table </span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(IDtoRemove, <span class="at">file=</span><span class="fu">paste0</span>(out,<span class="st">"/IDstoRemove_related.txt"</span>), <span class="at">row.names=</span>F, <span class="at">col.names=</span>T, <span class="at">quote =</span> F)</span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a><span class="co"># identified 622 IDs to remove due to relatedness</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>This next step removes IDs of related individuals just identified in the previous step.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 5: Make fam ID list excluding related individuals</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="va">relatedIDs</span><span class="op">=</span><span class="st">"/CCACE_Shared/Anna_F/BrainAtrophy/data/geneticQC/sourceCopy/IDstoRemove_related.txt"</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="ex">plink19</span> <span class="dt">\</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>--bed <span class="va">${target}</span>/ukb_neuroimaging_autosomal_sex_preQC.bed <span class="dt">\</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>--bim <span class="va">${target}</span>/ukb_neuroimaging_autosomal_sex_preQC.bim <span class="dt">\</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>--fam <span class="va">${target}</span>/ukb_neuroimaging_autosomal_sex_preQC.fam <span class="dt">\</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>--geno 0.02 <span class="dt">\</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>--extract ukb_neuroimaging_MAF0.01_GENO0.02_QA_EUR.snplist <span class="dt">\</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>--keep <span class="va">${target}</span>/ukb_neuroimaging_MAF0.01_GENO0.02_QA_EUR.fam <span class="dt">\</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>--write-snplist <span class="dt">\</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>--make-just-fam <span class="dt">\</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>--remove <span class="va">$relatedIDs</span> <span class="dt">\</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>--hardy <span class="at">--hwe</span> 0.00000001 <span class="dt">\</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>--out <span class="va">${target}</span>/ukb_neuroimaging_MAF0.01_GENO0.02_QA_EUR_HWE0.00000001</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a><span class="co"># after removing 622 IDs due to relatedness, we have 43414 participants remaining</span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a><span class="co"># 84 variants removed due to missing genotype data</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a><span class="co"># 11398 variants removed due to Hardy-Weinberg exact test (running this analysis throws a warning that "--hwe observation counts vary by more than 10%, due to the X chromosome.  You may want to use a less stringent --hwe p-value threshold for X chromosome variants" - however, without the X chromsome, this filter removes 11169 variants which is about the same so will leave this</span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a><span class="co"># 586058 variants remaining (which is about the same as in my previous study in another application)</span></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a><span class="co"># the resulting .fam and .snplist files could now be fed into regenie, but first we'll also do sex-check</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="perform-sex-check" class="level2">
<h2 class="anchored" data-anchor-id="perform-sex-check">Perform sex-check</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 6: Prune SNPs for sex-check</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co"># to do sex-check, we prune SNP data to be independent and exclude high LD regions</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co"># this is because patterns of LD will impair chromosome-specific tests of homozygosity</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="fu">awk</span> <span class="at">-f</span> <span class="va">${target}</span>/highLDregions4bim_b37.awk <span class="va">${target}</span>/ukb_neuroimaging_reQC.bim <span class="op">&gt;</span> <span class="va">${target}</span>/ukb_neuroimaging_High_LD_Regions_To_Exclude.txt</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="co"># this file contains 28923 SNPs to exclude from sex-check analysis</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="ex">plink19</span> <span class="dt">\</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>--bed <span class="va">${target}</span>/ukb_neuroimaging_autosomal_sex_preQC.bed <span class="dt">\</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>--bim <span class="va">${target}</span>/ukb_neuroimaging_autosomal_sex_preQC.bim <span class="dt">\</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>--fam <span class="va">${target}</span>/ukb_neuroimaging_autosomal_sex_preQC.fam <span class="dt">\</span></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>--maf 0.05 <span class="dt">\</span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>--hwe 0.001 <span class="dt">\</span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>--geno 0.02 <span class="dt">\</span></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>--thin-indiv-count 300 <span class="dt">\</span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>--indep-pairphase 200 100 0.2 <span class="dt">\</span></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>--seed 1204688 <span class="dt">\</span></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>--exclude <span class="va">${target}</span>/ukb_neuroimaging_High_LD_Regions_To_Exclude.txt <span class="dt">\</span></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>--extract <span class="va">${target}</span>/ukb_neuroimaging_MAF0.01_GENO0.02_QA_EUR_HWE0.00000001.snplist <span class="dt">\</span></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>--keep <span class="va">${target}</span>/ukb_neuroimaging_MAF0.01_GENO0.02_QA_EUR_HWE0.00000001.fam <span class="dt">\</span></span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>--out <span class="va">${target}</span>/ukb_neuroimaging_MAF0.01_GENO0.02_QA_EUR_HWE0.00000001_LD1_ALL</span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 7: Perform sex-check</span></span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a><span class="co"># F statistic used here is a function of teh deviation of the observed number of heterozygote variants from that expected under Hardy-Weinberg equilibrium</span></span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a><span class="co"># males should have F ~1 because all X chromosome variants are hemizygous and no heterozygotes can be observed</span></span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a><span class="co"># females should have lower values of F, distributed around 0 (but females with very high F stats have been observed)</span></span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a><span class="co"># this script shouldn't need a seperate --split-x analysis because the X chromosome's pseudoautosomal region is already presented as XY chromosome</span></span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a><span class="ex">plink19</span> <span class="dt">\</span></span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a>--bed <span class="va">${target}</span>/ukb_neuroimaging_autosomal_sex_preQC.bed <span class="dt">\</span></span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a>--bim <span class="va">${target}</span>/ukb_neuroimaging_autosomal_sex_preQC.bim <span class="dt">\</span></span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a>--fam <span class="va">${target}</span>/ukb_neuroimaging_autosomal_sex_preQC.fam <span class="dt">\</span></span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a>--check-sex <span class="dt">\</span></span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a>--keep <span class="va">${target}</span>/ukb_neuroimaging_MAF0.01_GENO0.02_QA_EUR_HWE0.00000001.fam <span class="dt">\</span></span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a>--extract <span class="va">${target}</span>/ukb_neuroimaging_MAF0.01_GENO0.02_QA_EUR_HWE0.00000001_LD1_ALL.prune.in <span class="dt">\</span></span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a>--out <span class="va">${target}</span>/ukb_neuroimaging_MAF0.01_GENO0.02_QA_EUR_HWE0.00000001_PRUNE_SEX</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="remove-individuals-that-did-not-pass-sex-check" class="level3">
<h3 class="anchored" data-anchor-id="remove-individuals-that-did-not-pass-sex-check">Remove individuals that did not pass sex-check</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="do">### Extract participants that have been labeled as PROBLEM</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>R</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>dat<span class="ot">=</span><span class="fu">fread</span>(<span class="st">"ukb_neuroimaging_MAF0.01_GENO0.02_QA_EUR_HWE0.00000001_PRUNE_SEX.sexcheck"</span>)</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>IDstoRemove<span class="ot">=</span>dat[<span class="fu">which</span>(dat<span class="sc">$</span>STATUS <span class="sc">==</span> <span class="st">"PROBLEM"</span>),<span class="fu">c</span>(<span class="st">"FID"</span>,<span class="st">"IID"</span>)]</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 22 </span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(IDstoRemove, <span class="at">file=</span><span class="fu">paste0</span>(out,<span class="st">"/sexCheck_toRemove.txt"</span>), <span class="at">row.names=</span>F, <span class="at">col.names=</span>T, <span class="at">quote =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="generate-final-.fam-and-.snplist-files" class="level2">
<h2 class="anchored" data-anchor-id="generate-final-.fam-and-.snplist-files">Generate final .fam and .snplist files</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co">## Step 8: Generate final .fam and .snplist to feed into regenie (excluding failed sex checks)</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="ex">plink19</span> <span class="dt">\</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>--bed <span class="va">${target}</span>/ukb_neuroimaging_autosomal_sex_preQC.bed <span class="dt">\</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>--bim <span class="va">${target}</span>/ukb_neuroimaging_autosomal_sex_preQC.bim <span class="dt">\</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>--fam <span class="va">${target}</span>/ukb_neuroimaging_autosomal_sex_preQC.fam <span class="dt">\</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>--extract <span class="va">${target}</span>/ukb_neuroimaging_MAF0.01_GENO0.02_QA_EUR_HWE0.00000001.snplist <span class="dt">\</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>--keep <span class="va">${target}</span>/ukb_neuroimaging_MAF0.01_GENO0.02_QA_EUR_HWE0.00000001.fam <span class="dt">\</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>--write-snplist <span class="dt">\</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>--make-just-fam <span class="dt">\</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>--remove <span class="va">${target}</span>/sexCheck_toRemove.txt <span class="dt">\</span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>--out <span class="va">${target}</span>/ukb_neuroimaging_MAF0.01_GENO0.02_QA_EUR_HWE0.00000001_sexcheck</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="generate-final-bed-file" class="level2">
<h2 class="anchored" data-anchor-id="generate-final-bed-file">Generate final bed file</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 9: Generate final bed files that will be used as input</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="va">final</span><span class="op">=</span><span class="st">"/CCACE_Shared/Anna_F/BrainAtrophy/data/geneticQC"</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="ex">plink19</span> <span class="dt">\</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>--bed <span class="va">${target}</span>/ukb_neuroimaging_autosomal_sex_preQC.bed <span class="dt">\</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>--bim <span class="va">${target}</span>/ukb_neuroimaging_autosomal_sex_preQC.bim <span class="dt">\</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>--fam <span class="va">${target}</span>/ukb_neuroimaging_autosomal_sex_preQC.fam <span class="dt">\</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>--extract <span class="va">${target}</span>/ukb_neuroimaging_MAF0.01_GENO0.02_QA_EUR_HWE0.00000001_sexcheck.snplist <span class="dt">\</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>--keep <span class="va">${target}</span>/ukb_neuroimaging_MAF0.01_GENO0.02_QA_EUR_HWE0.00000001_sexcheck.fam <span class="dt">\</span></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>--chr 1-22 <span class="dt">\</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>--make-bed <span class="dt">\</span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>--out <span class="va">${final}</span>/ukb_neuroimaging_brainAtrophy_GWASinput</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>